---
title: "DXR_Project"
author: "Emma M Pfortmiller"
date: "2025-04-30"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
    toc_depth: 2
    toc_float: true
    theme: journal
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
  
}
```

```{r Necessary Libraries, include=FALSE}
library(tidyverse)
library(readxl)
library(ggVennDiagram)
library(gprofiler2)
library(ggrepel)
library(reshape2)
library(hrbrthemes)
library(edgebundleR)
library(edgeR)
library(pheatmap)
library(ggfortify)
library(PCAtools)
#library(Cormotif)
library(RColorBrewer)
library(biomaRt)
library(RUVSeq)
library(Rfast)
library(cowplot)
```

```{r Import Counts Files, include=FALSE}
#these counts files are from featureCounts, all saved as RDS objects

####Individual 1 - 84-1####
# Counts_84_DOX_24 <- readRDS("data/counts/Counts_84_DOX_24.RDS")
# Counts_84_DMSO_24 <- readRDS("data/counts/Counts_84_DMSO_24.RDS")
# Counts_84_DOX_24rec <- readRDS("data/counts/Counts_84_DOX_24rec.RDS")
# Counts_84_DMSO_24rec <- readRDS("data/counts/Counts_84_DMSO_24rec.RDS")
# Counts_84_DOX_144rec <- readRDS("data/counts/Counts_84_DOX_144rec.RDS")
# Counts_84_DMSO_144rec <- readRDS("data/counts/Counts_84_DMSO_144rec.RDS")

####Individual 2 - 87-1####
# Counts_87_DOX_24 <- readRDS("data/counts/Counts_87_DOX_24.RDS")
# Counts_87_DMSO_24 <- readRDS("data/counts/Counts_87_DMSO_24.RDS")
# Counts_87_DOX_24rec <- readRDS("data/counts/Counts_87_DOX_24rec.RDS")
# Counts_87_DMSO_24rec <- readRDS("data/counts/Counts_87_DMSO_24rec.RDS")
# Counts_87_DOX_144rec <- readRDS("data/counts/Counts_87_DOX_144rec.RDS")
# Counts_87_DMSO_144rec <- readRDS("data/counts/Counts_87_DMSO_144rec.RDS")

####Individual 3 - 78-1####
# Counts_78_DOX_24 <- readRDS("data/counts/Counts_78_DOX_24.RDS")
# Counts_78_DMSO_24 <- readRDS("data/counts/Counts_78_DMSO_24.RDS")
# Counts_78_DOX_24rec <- readRDS("data/counts/Counts_78_DOX_24rec.RDS")
# Counts_78_DMSO_24rec <- readRDS("data/counts/Counts_78_DMSO_24rec.RDS")
# Counts_78_DOX_144rec <- readRDS("data/counts/Counts_78_DOX_144rec.RDS")
# Counts_78_DMSO_144rec <- readRDS("data/counts/Counts_78_DMSO_144rec.RDS")

####Individual 4 - 75-1####
# Counts_75_DOX_24 <- readRDS("data/counts/Counts_75_DOX_24.RDS")
# Counts_75_DMSO_24 <- readRDS("data/counts/Counts_75_DMSO_24.RDS")
# Counts_75_DOX_24rec <- readRDS("data/counts/Counts_75_DOX_24rec.RDS")
# Counts_75_DMSO_24rec <- readRDS("data/counts/Counts_75_DMSO_24rec.RDS")
# Counts_75_DOX_144rec <- readRDS("data/counts/Counts_75_DOX_144rec.RDS")
# Counts_75_DMSO_144rec <- readRDS("data/counts/Counts_75_DMSO_144rec.RDS")

####Individual 5 - 17-3####
# Counts_17_DOX_24 <- readRDS("data/counts/Counts_17_DOX_24.RDS")
# Counts_17_DMSO_24 <- readRDS("data/counts/Counts_17_DMSO_24.RDS")
# Counts_17_DOX_24rec <- readRDS("data/counts/Counts_17_DOX_24rec.RDS")
# Counts_17_DMSO_24rec <- readRDS("data/counts/Counts_17_DMSO_24rec.RDS")
# Counts_17_DOX_144rec <- readRDS("data/counts/Counts_17_DOX_144rec.RDS")
# Counts_17_DMSO_144rec <- readRDS("data/counts/Counts_17_DMSO_144rec.RDS")

####Individual 6 - 90-1####
# Counts_90_DOX_24 <- readRDS("data/counts/Counts_90_DOX_24.RDS")
# Counts_90_DMSO_24 <- readRDS("data/counts/Counts_90_DMSO_24.RDS")
# Counts_90_DOX_24rec <- readRDS("data/counts/Counts_90_DOX_24rec.RDS")
# Counts_90_DMSO_24rec <- readRDS("data/counts/Counts_90_DMSO_24rec.RDS")
# Counts_90_DOX_144rec <- readRDS("data/counts/Counts_90_DOX_144rec.RDS")
# Counts_90_DMSO_144rec <- readRDS("data/counts/Counts_90_DMSO_144rec.RDS")

####Individual 7 - 90-1REP####
# Counts_90REP_DOX_24 <- readRDS("data/counts/Counts_90REP_DOX_24.RDS")
# Counts_90REP_DMSO_24 <- readRDS("data/counts/Counts_90REP_DMSO_24.RDS")
# Counts_90REP_DOX_24rec <- readRDS("data/counts/Counts_90REP_DOX_24rec.RDS")
# Counts_90REP_DMSO_24rec <- readRDS("data/counts/Counts_90REP_DMSO_24rec.RDS")
# Counts_90REP_DOX_144rec <- readRDS("data/counts/Counts_90REP_DOX_144rec.RDS")
# Counts_90REP_DMSO_144rec <- readRDS("data/counts/Counts_90REP_DMSO_144rec.RDS")

```


```{r Create Data Matrix}
#input your counts data for the raw featureCounts files + make a matrix with all DOX + DMSO samples

counts_raw_df <-
  data.frame(
    Counts_84_DOX_24,
    Counts_84_DMSO_24$MCW_EMP_JT_R29_R1.bam,
    Counts_84_DOX_24rec$MCW_EMP_JT_R30_R1.bam,
    Counts_84_DMSO_24rec$MCW_EMP_JT_R32_R1.bam,
    Counts_84_DOX_144rec$MCW_EMP_JT_R33_R1.bam,
    Counts_84_DMSO_144rec$MCW_EMP_JT_R35_R1.bam,
    Counts_87_DOX_24$MCW_EMP_JT_R36_R1.bam,
    Counts_87_DMSO_24$MCW_EMP_JT_R38_R1.bam,
    Counts_87_DOX_24rec$MCW_EMP_JT_R39_R1.bam,
    Counts_87_DMSO_24rec$MCW_EMP_JT_R41_R1.bam,
    Counts_87_DOX_144rec$MCW_EMP_JT_R42_R1.bam,
    Counts_87_DMSO_144rec$MCW_EMP_JT_R44_R1.bam,
    Counts_78_DOX_24$MCW_EMP_JT_R45_R1.bam,
    Counts_78_DMSO_24$MCW_EMP_JT_R47_R1.bam,
    Counts_78_DOX_24rec$MCW_EMP_JT_R48_R1.bam,
    Counts_78_DMSO_24rec$MCW_EMP_JT_R50_R1.bam,
    Counts_78_DOX_144rec$MCW_EMP_JT_R51_R1.bam,
    Counts_78_DMSO_144rec$MCW_EMP_JT_R53_R1.bam,
    Counts_75_DOX_24$MCW_EMP_JT_R54_R1.bam,
    Counts_75_DMSO_24$MCW_EMP_JT_R56_R1.bam,
    Counts_75_DOX_24rec$MCW_EMP_JT_R57_R1.bam,
    Counts_75_DMSO_24rec$MCW_EMP_JT_R59_R1.bam,
    Counts_75_DOX_144rec$MCW_EMP_JT_R60_R1.bam,
    Counts_75_DMSO_144rec$MCW_EMP_JT_R62_R1.bam,
    Counts_17_DOX_24$MCW_EMP_JT_R63_R1.bam,
    Counts_17_DMSO_24$MCW_EMP_JT_R65_R1.bam,
    Counts_17_DOX_24rec$MCW_EMP_JT_R66_R1.bam,
    Counts_17_DMSO_24rec$MCW_EMP_JT_R68_R1.bam,
    Counts_17_DOX_144rec$MCW_EMP_JT_R69_R1.bam,
    Counts_17_DMSO_144rec$MCW_EMP_JT_R71_R1.bam,
    Counts_90_DOX_24$MCW_EMP_JT_R72_R1.bam,
    Counts_90_DMSO_24$MCW_EMP_JT_R74_R1.bam,
    Counts_90_DOX_24rec$MCW_EMP_JT_R75_R1.bam,
    Counts_90_DMSO_24rec$MCW_EMP_JT_R77_R1.bam,
    Counts_90_DOX_144rec$MCW_EMP_JT_R78_R1.bam,
    Counts_90_DMSO_144rec$MCW_EMP_JT_R80_R1.bam,
    Counts_90REP_DOX_24$MCW_EMP_JT_R81_R1.bam,
    Counts_90REP_DMSO_24$MCW_EMP_JT_R83_R1.bam,
    Counts_90REP_DOX_24rec$MCW_EMP_JT_R84_R1.bam,
    Counts_90REP_DMSO_24rec$MCW_EMP_JT_R86_R1.bam,
    Counts_90REP_DOX_144rec$MCW_EMP_JT_R87_R1.bam,
    Counts_90REP_DMSO_144rec$MCW_EMP_JT_R89_R1.bam
  )

# colnames(counts_raw_df) <- c("entrezgene_id", 
#                                 "DOX_24_Ind1",
#                                 "DMSO_24_Ind1",
#                                 "DOX_24rec_Ind1",
#                                 "DMSO_24rec_Ind1",
#                                 "DOX_144rec_Ind1",
#                                 "DMSO_144rec_Ind1",
#                                 "DOX_24_Ind2",
#                                 "DMSO_24_Ind2",
#                                 "DOX_24rec_Ind2",
#                                 "DMSO_24rec_Ind2",
#                                 "DOX_144rec_Ind2",
#                                 "DMSO_144rec_Ind2",
#                                 "DOX_24_Ind3",
#                                 "DMSO_24_Ind3",
#                                 "DOX_24rec_Ind3",
#                                 "DMSO_24rec_Ind3",
#                                 "DOX_144rec_Ind3",
#                                 "DMSO_144rec_Ind3",
#                                 "DOX_24_Ind4",
#                                 "DMSO_24_Ind4",
#                                 "DOX_24rec_Ind4",
#                                 "DMSO_24rec_Ind4",
#                                 "DOX_144rec_Ind4",
#                                 "DMSO_144rec_Ind4",
#                                 "DOX_24_Ind5",
#                                 "DMSO_24_Ind5",
#                                 "DOX_24rec_Ind5",
#                                 "DMSO_24rec_Ind5",
#                                 "DOX_144rec_Ind5",
#                                 "DMSO_144rec_Ind5",
#                                 "DOX_24_Ind6",
#                                 "DMSO_24_Ind6",
#                                 "DOX_24rec_Ind6",
#                                 "DMSO_24rec_Ind6",
#                                 "DOX_144rec_Ind6",
#                                 "DMSO_144rec_Ind6",
#                                 "DOX_24_Ind6REP",
#                                 "DMSO_24_Ind6REP",
#                                 "DOX_24rec_Ind6REP",
#                                 "DMSO_24rec_Ind6REP",
#                                 "DOX_144rec_Ind6REP",
#                                 "DMSO_144rec_Ind6REP")

counts_raw_matrix <- counts_raw_df %>% column_to_rownames(var = "X") %>% as.matrix()


#write this one to an R object so I can hold onto it
#write.csv(counts_raw_matrix, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/featureCounts_Concat_Matrix_DOXSamples_EMP_250430.csv")

#saveRDS(counts_raw_matrix, "data/counts_raw_matrix.RDS")
counts_raw_matrix <- readRDS("data/counts_raw_matrix.RDS")
dim(counts_raw_matrix)
#begins with 28395 genes across 42 conditions

boxplot(counts_raw_matrix,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 1000), main = "Boxplot of Raw Counts All Individuals")

hist(counts_raw_matrix,  xlab = "Unfiltered Counts", xlim = c(-1000,125000), ylim = c(-100, 6000000), main = "Histogram of Raw Counts All Individuals")


```

```{r Data Sets}
#Raw Counts
counts_raw_matrix

#counts_raw_df <- data.frame(counts_raw_matrix)

#for QC checks, I need to convert this into cpm as well
#DO NOT USE FOR FILTERING

counts_cpm_unfilt <- cpm(counts_raw_matrix, log = TRUE)

hist(counts_cpm_unfilt,  
     main = "Histogram of total counts (unfiltered)", 
     xlab = expression("Log"[2]*" counts-per-million"), 
     col = 4)

#do this one instead now
###filter my data by rowMeans > 0

filcpm_matrix <- subset(counts_cpm_unfilt, (rowMeans(counts_cpm_unfilt) > 0))
dim(filcpm_matrix)

#now let's make a histogram of this to check the difference
hist(filcpm_matrix,  
     main = "Histogram of filtered counts using rowMeans > 0 method", 
     xlab = expression("Log"[2]*" counts-per-million"), 
     col = 2)

#export this as a csv
#write.csv(filcpm_matrix, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/filcpm_matrix.csv")

#Now I need to filter my dataset by rowmeans > 0 and pull those genes for filtering

# counts_raw_filt <- counts_raw_matrix[rowMeans(cpm(counts_raw_matrix, log = TRUE)) > 0, ]
dim(counts_raw_filt)
#14319 total genes

#also make a cpm version of this one for QC
#DO NOT USE FOR FILTERING OR ANALYSIS
counts_cpm_filt <- cpm(counts_raw_filt, log = TRUE)

#this file was filtered based on the log2cpm values, but is still in counts
#save this counts file and filter my counts based on it

filter_gene_list <- row.names(counts_raw_filt)
length(filter_gene_list)
#14319 - same number of genes as above

#saveRDS(filter_gene_list, "data/filter_gene_list_final.RDS")

#now filter my raw counts file based on this gene list

filt_counts_dxr <- counts_raw_matrix[(rownames(counts_raw_matrix) %in% filter_gene_list), ]

dim(filt_counts_dxr)
#14319 genes - matching the above

#now I'll convert these counts to cpm to do QC checks
filt_cpm_dxr <- cpm(filt_counts_dxr, log = TRUE)
#14319 genes and 42 columns as expected

#here is the sheet that has all of the mapping percentages made in excel
# fC_AllCounts <- read_csv("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/Recovery_MappingStats_Subread_EMP_241104_250211.csv") 
# 
# fC_DOXCounts <- fC_AllCounts %>% 
#   dplyr::filter(Treatment != "FLUO")

#saveRDS(fC_DOXCounts, "data/fC_DOXCounts.RDS")
fC_DOXCounts <- readRDS("data/fC_DOXCounts.RDS")

```

```{r Colors + Factors}

col_tx_large <- rep(c("#499FBD" , "#63666D"), 21)
col_tx_large_2 <- c(rep("#499FBD" , 3), rep("#63666D", 3), 21)


ind_col <- c("#003F5C", "#45AE91",  "#58508D", "#8B3E9B", "#FF6361", "#BC4169", "#FF2362")

tx_col <- c("#499FBD","#63666D")

time_col <- c("#fbb4b9", "#f768a1", "#ae017e")

ind_names <- c(rep("Ind1", 6), rep("Ind2", 6), rep("Ind3", 6), rep("Ind4", 6), rep("Ind5", 6), rep("Ind6", 6), rep("Ind6REP", 6))
time_names <- c(rep("24", 2), rep("24rec", 2), rep("144rec", 2))
time_names2 <- c("24", "24rec", "144rec")
time_names <- c(rep(time_names, 7))
time_names2 <- c(rep(time_names2, 7))
tx_names <- c("DOX", "DMSO")
tx_names <- c(rep(tx_names, 21))
tx_names2 <- c(rep("DOX", 3), rep("DMSO", 3))
tx_names2 <- c(rep(tx_names2, 21))
txtime_names <- c("DOX_24", "DMSO_24", "DOX_24rec", "DMSO_24rec", "DOX_144rec", "DMSO_144rec")
txtime_names <- c(rep(txtime_names, 7))
txtime_names2 <- c("DOX_24", "DOX_24rec", "DOX_144rec", "DMSO_24", "DMSO_24rec", "DMSO_144rec")
txtime_names2 <- c(rep(txtime_names2, 7))



####DE genes factors####
ind_names_de <- c(rep("Ind1", 6), rep("Ind2", 6), rep("Ind3", 6), rep("Ind4", 6), rep("Ind5", 6), rep("Ind6", 6), rep("Ind6REP", 6))
time_names_de <- c(rep("24", 2), rep("24rec", 2), rep("144rec", 2))
time_names2_de <- c("24", "24rec", "144rec")
time_names_de <- c(rep(time_names_de, 7))
time_names2_de <- c(rep(time_names2_de, 7))
tx_names_de <- c("DOX", "DMSO")
tx_names_de <- c(rep(tx_names_de, 21))
txtime_names_de <- c("DOX_24", "DMSO_24", "DOX_24rec", "DMSO_24rec", "DOX_144rec", "DMSO_144rec")
txtime_names_de <- c(rep(txtime_names_de, 7))

```



Since I've now made my filtered matrix of counts, I want to make sure everything looks right
I've named my filtered counts filt_counts_dxr
I've transformed this into log2cpm and named it filt_cpm_dxr

```{r QC Counts}
reads_by_sample <- c(tx_col)
fC_DOXCounts %>% 
  ggplot(., aes (x = Conditions, y = Total_Align, fill = Treatment, group_by = Line))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Read Counts by Treatment####
fC_DOXCounts %>% 
  ggplot(., aes (x =Treatment, y= Total_Align, fill = Treatment))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by treatment"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Reads Per Individual####
fC_DOXCounts %>% 
  ggplot(., aes (x =as.factor(Line), y=Total_Align))+
  geom_boxplot(aes(fill=as.factor(Line)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total number of reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Mapped Reads Per Drug####

reads_by_sample <- c(tx_col)
fC_DOXCounts %>% 
  ggplot(., aes (x = Conditions, y = Assigned_Align, fill = Treatment, group_by = Line))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Read Counts by Treatment####
fC_DOXCounts %>% 
  ggplot(., aes (x =Treatment, y= Assigned_Align, fill = Treatment))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by treatment"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Reads Per Individual####
fC_DOXCounts %>% 
  ggplot(., aes (x =as.factor(Line), y=Assigned_Align))+
  geom_boxplot(aes(fill=as.factor(Line)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total number of mapped reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


```


```{r QC Checks cpm}

####histogram of total counts, unfiltered####
hist(counts_raw_matrix, main = "Recovery All Individuals log2cpm Unfiltered",
     xlab = expression("log" [2]* "counts per million"))

####histogram of total counts, filtered####
hist(counts_raw_filt, 
     main = "Recovery All Individuals log2cpm Filtered rowMeans > 0", 
        xlab = expression("log"[2]*"counts per million"))

####boxplot log2cpm all samples####

boxplot(counts_cpm_unfilt, xlab = "Conditions", ylab = expression("log"[2]* "counts per million"), ylim= c(-10, 20), main = "Boxplots of log2cpm per sample")

####boxplot log2cpm filtered all samples####

boxplot(filt_cpm_dxr, xlab = "Conditions", ylab = expression("log"[2]* "counts per million"), ylim= c(-10, 20), main = "Boxplots of log2cpm Filtered rowMeans > 0 per sample")

```

```{r QC PCA Plots}
#We can use PCA plots to get an idea of how my samples are grouping together

PCA_data <- filt_cpm_dxr %>% 
  prcomp(.) %>% 
  t()

PCA_data_test <- (prcomp(t(filt_cpm_dxr), scale. = TRUE))


# ind_num_dxr <- c("1", "1", "1", "1", "1", "1", "2", "2", "2", "2", "2", "2", "3", "3", "3", "3", "3", "3", "4", "4", "4", "4", "4", "4", "5", "5", "5", "5", "5", "5", "6", "6", "6", "6", "6", "6", "6R", "6R", "6R", "6R", "6R", "6R")

# saveRDS(ind_num_dxr, "data/ind_num_dxr.RDS")

ind_num_dxr <- readRDS("data/ind_num_dxr.RDS")

####now make an annotation for my PCA####
annot <- data.frame("samples" = colnames(filt_cpm_dxr)) %>%
  separate_wider_delim(., 
                       cols = samples, 
                       names = c("Tx", "Time", "Ind"), 
                       delim = "_", cols_remove = FALSE)
  


#saveRDS(annot, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/annot.RDS")


#combine the prcomp matrix and annotation
annot_PCA_matrix <- PCA_data_test$x %>% cbind(., annot)


#now I can make a graph where I have filled values for individual! I have seven colors for seven individuals
# I have three fill values for three timepoints

#using annotation matrix above as well as annotated PCA matrix (annot_PCA_matrix)

#extra info for colors in the graph (fill parameter)
fill_col_ind <- c("#66C2A5", "#FC8D62", "#1F78B4", "#E78AC3", "#A6D854", "#FFD92A", "#8B3E9B")

fill_col_ind_dark <- c("#003F5C", "#45AE91",  "#58508D", "#BC4099", "#8B3E9B", "#FF6361", "#FF2362")

fill_col_tx <- c("#63666D", "#499FBD")

#I want the color parameter to be treatment, the shape as time, and individual as number

####PC1/PC2####
annot_PCA_matrix %>% ggplot(., aes(x=PC1, y=PC2, size=10)) +
  geom_point(aes(color = Tx, shape = Time)) +
  scale_color_manual(values = c(fill_col_tx))+
  scale_shape_manual(values = c(15, 19, 17))+
  geom_text_repel(aes(label = ind_num_dxr))+
  theme_bw(base_size = 10)+
  ggtitle(expression("PCA of filtered log"[2]*"cpm"))+
  guides(size="none")
  

####PC2/PC3####
annot_PCA_matrix %>% ggplot(., aes(x=PC2, y=PC3, size=10)) +
  geom_point(aes(color = Tx, shape = Time)) +
  scale_color_manual(values = c(fill_col_tx))+
  scale_shape_manual(values = c(15, 19, 17))+
  geom_text_repel(aes(label = ind_num_dxr))+
  theme_bw(base_size = 10)+
  ggtitle(expression("PCA of filtered log"[2]*"cpm"))+
  guides(size="none")

####PC3/PC4####
annot_PCA_matrix %>% ggplot(., aes(x=PC3, y=PC4, size=10)) +
  geom_point(aes(color = Tx, shape = Time)) +
  scale_color_manual(values = c(fill_col_tx))+
  scale_shape_manual(values = c(15, 19, 17))+
  geom_text_repel(aes(label = ind_num_dxr))+
  theme_bw(base_size = 10)+
  ggtitle(expression("PCA of filtered log"[2]*"cpm"))+
  guides(size="none")


```

```{r Correlation Heatmap, fig.width=10, fig.height=8}
#correlation heatmap for all samples

####PEARSON FILTERED####
fC_Matrix_Full_cpm_filter_pearsoncor <-
  cor(
    filt_cpm_dxr,
    y = NULL,
    use = "everything",
    method = "pearson"
  )

####SPEARMAN FILTERED####
fC_Matrix_Full_cpm_filter_spearmancor <-
  cor(
    filt_cpm_dxr,
    y = NULL,
    use = "everything",
    method = "spearman"
  )



#Now let's graph these correlations

pheatmap(fC_Matrix_Full_cpm_filter_pearsoncor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 12, fontsize_number = 9)

pheatmap(fC_Matrix_Full_cpm_filter_spearmancor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 12, fontsize_number = 9)



#now annotate these heatmaps
#Factor 1 - Individual
Individual <- as.factor(c(rep("Ind1", 6), rep("Ind2", 6), rep("Ind3", 6), rep("Ind4", 6), rep("Ind5", 6), rep("Ind6", 6), rep("Ind6REP", 6)))

#Factor 2 - Treatment
tx_factor <- c("DOX", "DMSO")
Tx <- as.factor(c(rep(tx_factor, 21)))
#view(Treatment)

#Factor 3 - Timepoint
time_factor <- c(rep("24", 2), rep("24rec", 2), rep("144rec", 2))
Time <- as.factor(c(rep(time_factor, 7)))

####annotation for colors####
annot_col_hm = list(Tx = c(DOX = "red", DMSO = "black"),
                             Ind = c(Ind1 = "#66C2A5", Ind2 = "#FC8D62", Ind3 = "#1F78B4", Ind4 = "#E78AC3", Ind5 = "#A6D854", Ind6 = "#FFD92A", Ind6REP = "#8B3E9B"),
                             Time = c("24" = "#046A38", "24rec" = "#0050B5", "144rec" = "#B725AD"))

####annotation for values####
annot_list_hm <- data.frame(Individual = as.factor(c(rep("Ind1", 6), rep("Ind2", 6), rep("Ind3", 6), rep("Ind4", 6), rep("Ind5", 6), rep("Ind6", 6), rep("Ind6REP", 6))),
                                               Tx = as.factor(c(rep(tx_factor, 21))), 
                                               Time = as.factor(c(rep(time_factor, 7))))


##add in the annotations from above into the dataframe
row.names(annot_list_hm) <- colnames(fC_Matrix_Full_cpm_filter_pearsoncor)
row.names(annot_list_hm) <- colnames(fC_Matrix_Full_cpm_filter_spearmancor)

####ANNOTATED HEATMAPS####
pheatmap(fC_Matrix_Full_cpm_filter_pearsoncor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 10, fontsize_number = 5, annotation_col = annot_list_hm, annotation_colors = annot_col_hm, main = "Pearson Correlation Heatmap")

pheatmap(fC_Matrix_Full_cpm_filter_spearmancor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 10, fontsize_number = 5, annotation_col = annot_list_hm, annotation_colors = annot_col_hm, main = "Spearman Correlation Heatmap")

#the correlations look good, the far left is made up of primarily DOX 24 treatment samples, divided by vehicle samples in the center, and the recovery timepoints are on the right side (one group of 24rec and another group of 144rec)

```

```{r Plot Genes of Interest, fig.width=8, fig.height=5}

##Add columns with more information to each gene I pull out##
ind_names <- c(rep("Ind1", 6), rep("Ind2", 6), rep("Ind3", 6), rep("Ind4", 6), rep("Ind5", 6), rep("Ind6", 6), rep("Ind6REP", 6))
time_names <- c(rep("24", 2), rep("24rec", 2), rep("144rec", 2))
time_names2 <- c("24", "24rec", "144rec")
time_names <- c(rep(time_names, 7))
time_names2 <- c(rep(time_names2, 7))
tx_names <- c("DOX", "DMSO")
tx_names <- c(rep(tx_names, 21))
tx_names2 <- c(rep("DOX", 3), rep("DMSO", 3))
tx_names2 <- c(rep(tx_names2, 21))
txtime_names <- c("DOX_24", "DMSO_24", "DOX_24rec", "DMSO_24rec", "DOX_144rec", "DMSO_144rec")
txtime_names <- c(rep(txtime_names, 7))
txtime_names2 <- c("DOX_24", "DOX_24rec", "DOX_144rec", "DMSO_24", "DMSO_24rec", "DMSO_144rec")
txtime_names2 <- c(rep(txtime_names2, 7))


genedf_dxr <-  counts_cpm_filt %>% 
  as.data.frame() %>% 
  rownames_to_column(var = "entrezgene_id")

#saveRDS(genedf_dxr, "data/genedf_dxr.RDS")

###CDKN1A - 1026###
CDKN1A_d <- genedf_dxr %>% filter(entrezgene_id=="1026")
CDKN1A_new_d <- as.data.frame(CDKN1A_d) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1","DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3",  "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")
CDKN1A_melt_d <- melt(CDKN1A_d, variable.name = "sample")
CDKN1A_melt_new_d <- melt(CDKN1A_new_d, variable.name = "sample")

CDKN1A_melt_df_d <- data.frame(tx = factor(tx_names, levels = unique(tx_names)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names, levels = unique(txtime_names)),
                               time = factor(time_names, levels = unique(time_names)))
CDKN1A_melt_df2_d <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
CDKN1A_melt_df_all_d <- cbind(CDKN1A_melt_d, CDKN1A_melt_df_d)
CDKN1A_melt_df_all2_d <- cbind(CDKN1A_melt_new_d, CDKN1A_melt_df2_d)


####CDKN1A####
CDKN1A_melt_df_all2_d %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "CDKN1A")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(0,15)+
  theme(plot.title = element_text(face = "italic"))

###MDM2 - 4193###
MDM2_d <- genedf_dxr %>% filter(entrezgene_id=="4193")
MDM2_new_d <- as.data.frame(MDM2_d) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1","DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3",  "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")
MDM2_melt_d <- melt(MDM2_d, variable.name = "sample")
MDM2_melt_new_d <- melt(MDM2_new_d, variable.name = "sample")

MDM2_melt_df_d <- data.frame(tx = factor(tx_names, levels = unique(tx_names)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names, levels = unique(txtime_names)),
                               time = factor(time_names, levels = unique(time_names)))
MDM2_melt_df2_d <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
MDM2_melt_df_all_d <- cbind(MDM2_melt_d, MDM2_melt_df_d)
MDM2_melt_df_all2_d <- cbind(MDM2_melt_new_d, MDM2_melt_df2_d)

####MDM2####
MDM2_melt_df_all2_d %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "MDM2")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(0,15)+
  theme(plot.title = element_text(face = "italic"))

#saveRDS(MDM2_geneplot_Dox, "data/MDM2_geneplot_Dox.RDS")


###SIRT1 - 23411###
SIRT1_d <- genedf_dxr %>% filter(entrezgene_id=="23411")
SIRT1_new_d <- as.data.frame(SIRT1_d) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1","DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3",  "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")
SIRT1_melt_d <- melt(SIRT1_d, variable.name = "sample")
SIRT1_melt_new_d <- melt(SIRT1_new_d, variable.name = "sample")

SIRT1_melt_df_d <- data.frame(tx = factor(tx_names, levels = unique(tx_names)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names, levels = unique(txtime_names)),
                               time = factor(time_names, levels = unique(time_names)))
SIRT1_melt_df2_d <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
SIRT1_melt_df_all_d <- cbind(SIRT1_melt_d, SIRT1_melt_df_d)
SIRT1_melt_df_all2_d <- cbind(SIRT1_melt_new_d, SIRT1_melt_df2_d)

####SIRT1####
SIRT1_vertplot <- 
  SIRT1_melt_df_all2_d %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "SIRT1")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(0,15)+
  theme(plot.title = element_text(face = "italic"))

SIRT1_melt_df_all2_d %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "SIRT1")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(0,15)+
  theme(plot.title = element_text(face = "italic"))

plot_leg_d_ver <- cowplot::get_legend(SIRT1_vertplot)

ggdraw(plot_leg_d_ver)

#saveRDS(plot_leg_d_ver, "data/plot_leg_d_vertical.RDS")

```


```{r Differential Expression Analysis - limma}

#I need to remove the Ind6REP individual before limma and Cormotif

filt_counts_norep <- filt_counts_dxr %>%
  as.data.frame() %>% 
  dplyr::select(-(contains("Ind6REP")))


group_fac <- rep(c("DOX_24",
                   "DMSO_24",
                   "DOX_24rec",
                   "DMSO_24rec",
                   "DOX_144rec",
                   "DMSO_144rec"), 6)

dge_dxr <- DGEList.data.frame(counts = filt_counts_norep, group = group_fac, genes = row.names(filt_counts_norep), )

#check the inital file before norm factors are calculated
View(dge_dxr$samples)

#calculate the normalization factors with method TMM
dge_calc_dxr <- calcNormFactors(dge_dxr, method = "TMM")

#check final file after norm calculation
View(dge_calc_dxr$samples)

#after calculating norm factors, that column was changed.

#saveRDS(dge_calc_dxr, "data/dge_calc_dxr.RDS")

#do I need to use cpm to finally calculate all of these counts?
#cpm internally recognizes the library norm factors and norms the counts

dge_calc_dxr_norm <- cpm(dge_calc_dxr)

#Pull out factors
snames <- data.frame("samples" = colnames(dge_calc_dxr)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)

#snames_list <- as.list(snames)

snames_time <- snames$Time
snames_tx <- snames$Treatment
snames_ind <- snames$Individual

#Create my model matrix
mm_dxr <- model.matrix(~0 + group_fac)
View(mm_dxr)
#note that the columns are in alphabetical order now instead of typical order

p_dxr <- voom(dge_calc_dxr$counts, mm_dxr, plot = TRUE)

corfit_dxr <- duplicateCorrelation(p_dxr, mm_dxr, block = snames_ind)

v_dxr <- voom(dge_calc_dxr$counts, mm_dxr, block = snames_ind, correlation = corfit_dxr$consensus)

fit_dxr <- lmFit(v_dxr, mm_dxr, block = snames_ind, correlation = corfit_dxr$consensus)

#make sure to check which order the columns are in - otherwise they won't match right (it was moved into alphabetical and number order)
colnames(mm_dxr) <- c("DMSO_144rec","DMSO_24","DMSO_24rec","DOX_144rec","DOX_24","DOX_24rec")

cm_dxr <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        levels = mm_dxr
)

vfit_dxr <- lmFit(p_dxr, mm_dxr)
vfit_dxr <- contrasts.fit(vfit_dxr, contrasts = cm_dxr)

efit_dxr <- eBayes(vfit_dxr)

results = decideTests(efit_dxr)
summary(results)
#        V.D24 V.D24r V.D144r
# Down    4305   2024     261
# NotSig  5985   7564   14016
# Up      4029   4731      42

#without reps
#        V.D24 V.D24r V.D144r
# Down    4083   1846     221
# NotSig  6671   8240   14070
# Up      3565   4233      28

#now that the filtering step is correct:
#DOX24 is balanced between up and down
#DOX24R is mainly upregulated
#DOX144R is mainly downregulated

####plot your voom####
voom_plot_dxr <- voom(dge_calc_dxr, mm_dxr, plot = TRUE)

```

```{r Toptables DE}
top.table_V.D24_dxr <- topTable(fit = efit_dxr, coef = "V.D24", number = nrow(dge_calc_dxr), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.D24_dxr)

#saveRDS(top.table_V.D24_dxr, "data/top.table_V.D24_dxr.RDS")

top.table_V.D24r_dxr <- topTable(fit = efit_dxr, coef = "V.D24r", number = nrow(dge_calc_dxr), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.D24r_dxr)

#saveRDS(top.table_V.D24r_dxr, "data/top.table_V.D24r_dxr.RDS")

top.table_V.D144r_dxr <- topTable(fit = efit_dxr, coef = "V.D144r", number = nrow(dge_calc_dxr), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.D144r_dxr)

#saveRDS(top.table_V.D144r_dxr, "data/top.table_V.D144r_dxr.RDS")

```

```{r Top 5 Most DE genes D24}
#plot the top 5 most DE genes for each condition
####start with DOX 24
#i sorted the toptable by the top 5 most DE genes for each + found the corresponding gene name

#take your counts matrix and convert to log2cpm to look at expression changes

# dge_cpm_dxr <- cpm(dge_calc_dxr$counts, log = TRUE)
# dim(dge_cpm_dxr)


#####24hr Tx DOX pval####
#slice the top 5 rows with the highest adjusted p (q) value in my toptable of the given condition
top5DE_d24_dxr <- top.table_V.D24_dxr %>% 
   dplyr::slice_min(., n=5, order_by=adj.P.Val) %>% 
   rownames_to_column(var = "GeneID")
 
#make a list of the gene IDs of the top 5 DE genes
 
GeneID_top5_d24_dxr <- c("6421", "1901", "8880", "80775", "284309")
GeneNames_top5_d24_dxr <- c("SFPQ", "S1PR1", "FUBP1", "TMEM177", "ZNF776")
 
#make a df of the samples (columns) that you want to look at
#d24 set
d24_gene_dxr <- filt_cpm_dxr %>% 
  as.data.frame() %>% 
  dplyr::select(matches("_24")) %>% 
  dplyr::select(-(contains("rec"))) %>% 
  rownames_to_column(var = "entrezgene_id")

#now pull those rows out of my new dataframe 

#I'll pull each of these genes out one by one

####Gene 1 DOX24 - SFPQ 6421####
top5DE_d24_log2cpm_dxr_1 <- d24_gene_dxr %>% 
  dplyr::filter(entrezgene_id == "6421") %>% 
  as.data.frame()

top5DE_d24_log2cpm_dxr_1_melt <- top5DE_d24_log2cpm_dxr_1 %>% 
  melt(id.vars = "entrezgene_id") %>% 
  separate_wider_delim(., cols = variable, 
                       names = c("Treatment", "Time", "Individual"), 
                       delim = "_", 
                       cols_remove = FALSE)

#####6421####
SFPQ <- top5DE_d24_log2cpm_dxr_1 %>% filter(entrezgene_id=="6421")
SFPQ_new <- as.data.frame(SFPQ) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DMSO_24_Ind1", "DOX_24_Ind2","DMSO_24_Ind2", "DOX_24_Ind3", "DMSO_24_Ind3", "DOX_24_Ind4", "DMSO_24_Ind4", "DOX_24_Ind5", "DMSO_24_Ind5", "DOX_24_Ind6", "DMSO_24_Ind6", "DOX_24_Ind6REP", "DMSO_24_Ind6REP")

SFPQ_melt <- melt(SFPQ, variable.name = "sample")
SFPQ_melt_new <- melt(SFPQ_new, variable.name = "sample")

SFPQ_melt_df <- data.frame(tx = factor(tx_names_de, levels = unique(tx_names)),
                               ind = factor(ind_names_de, levels = unique(ind_names_de)),
                               txtime = factor(txtime_names, levels = unique(txtime_names_de)),
                               time = factor(time_names_de, levels = unique(time_names_de)))

SFPQ_melt_df2 <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
SFPQ_melt_df_all <- cbind(SFPQ_melt, SFPQ_melt_df)
SFPQ_melt_df_all2 <- cbind(SFPQ_melt_new, SFPQ_melt_df2)


# ggplot(top5DE_d24_log2cpm_dxr_1_melt, aes(x = entrezgene_id))+
#   geom_boxplot(aes(colour = Treatment))+
#   geom_point(aes(x = value, y = variable))+
#   ggtitle(expression("Top 5 DEG DOX 24hr - 1"))+
#   xlab("")+
#   ylim(c(-10, 10))+
#   ylab(expression("log2cpm"))
#   theme_bw()+
#   theme(plot.title = element_text(size = rel(1.5)),
#         axis.title = element_text(size = 15, color = "black"),
#         axis.ticks = element_line(linewidth = 1.5),
#         axis.line = element_line(linewidth = 1.5),
#         axis.text = element_text(size =10, color = "black", angle = 0),
#         strip.text.y = element_text(color = "white"))


SFPQ_melt_df_all2 %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "SFPQ")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(-5, 10)
  theme(plot.title = element_text(face = "italic"))

```


```{r Volcano Plots}
generate_volcano_plot <- function(toptable, title) {
  
  #make significance labels
  toptable$Significance <- "Not Significant"
  toptable$Significance[toptable$logFC > 0 & toptable$adj.P.Val < 0.05] <- "Upregulated"
  toptable$Significance[toptable$logFC < 0 & toptable$adj.P.Val < 0.05] <- "Downregulated"
  
  #add number of genes for each significance label
  upgenes <- toptable %>% filter(Significance == "Upregulated") %>% nrow()
  nsgenes <- toptable %>% filter(Significance == "Not Significant") %>% nrow()
  downgenes <- toptable %>% filter(Significance == "Downregulated") %>% nrow()

  #make legend labels for no of genes
  legend_lab <- c(
    str_c("Upregulated: ", upgenes),
    str_c("Not Significant: ", nsgenes),
    str_c("Downregulated: ", downgenes)
  )
  
  #specify the colors for the legend
  legend_col <- c(
    str_c("Upregulated: " = "blue"),
    str_c("Not Significant: " = "gray"),
    str_c("Downregulated: " = "red")
  )

  #generate volcano plot w/ legend
  ggplot(toptable, aes(x = logFC, 
                       y = -log10(P.Value), 
                       color = Significance)) +
    geom_point(alpha = 0.4, size = 2) + 
    scale_color_manual(values = c("Upregulated" = "blue",
                                  "Not Significant" = "gray",
                                  "Downregulated" = "red"), 
                       labels = legend_lab) +
    xlim(-10, 10) +
    labs(title = title, 
         x = expression(x = "log"[2]*"FC"), 
         y = expression(y = "-log"[10]*"P-value")) +
    theme_bw()+
    guides(color = guide_legend(override.aes = list(color = legend_col)))+
    theme(legend.position = "right", 
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25)))
}


#now that I've made a function, I can make volcano plots for each of my comparisons (6 total)

volcano_plots <- list(
  "V.D24" = generate_volcano_plot(top.table_V.D24_dxr, "Volcano Plot DOX 24hr (adj P-val<0.05)"),
  "V.D24r" = generate_volcano_plot(top.table_V.D24r_dxr, "Volcano Plot DOX 24hr Recovery (adj P-val<0.05)"),
  "V.D144r" = generate_volcano_plot(top.table_V.D144r_dxr, "Volcano Plot DOX 144hr Recovery (adj P-val<0.05)")
)

# Display each volcano plot
for (plot_name in names(volcano_plots)) {
  print(volcano_plots[[plot_name]])
}

```

```{r DEG Overlap Venn Diagram}
#put all of the genes per condition into each category
DOX_24_top_dxr <- top.table_V.D24_dxr %>% 
  rownames_to_column(., var = "entrezgene_ID")
DOX_24r_top_dxr <- top.table_V.D24r_dxr %>% 
  rownames_to_column(., var = "entrezgene_ID")
DOX_144r_top_dxr <- top.table_V.D144r_dxr %>% 
  rownames_to_column(., var = "entrezgene_ID")

DEG_1_dxr <- DOX_24_top_dxr$entrezgene_ID[DOX_24_top_dxr$adj.P.Val < 0.05]
DEG_2_dxr <- DOX_24r_top_dxr$entrezgene_ID[DOX_24r_top_dxr$adj.P.Val < 0.05]
DEG_3_dxr <- DOX_144r_top_dxr$entrezgene_ID[DOX_144r_top_dxr$adj.P.Val < 0.05]

#now that I've made the DEG files with an adj. p. value cutoff of 0.05, I can look for overlap

library(ggVennDiagram)

venn_DEGs_dxr <- list(DEG_1_dxr, DEG_2_dxr, DEG_3_dxr)

ggVennDiagram(
  venn_DEGs_dxr, 
  category.names = c("DOX24T", "DOX24R", "DOX144R"),
  fill = c("red", "blue", "green")
) +
  ggtitle("DOX vs VEH Over Time")+
  theme(
    plot.title = element_text(size = 16),
    text = element_text(size = 16)
  )



```

```{r GO analysis DEG Overlap}
#Now, I will pull out the up and down regulated genes in each DEG set

DOX24_DEGs <- top.table_V.D24_dxr %>% 
  dplyr::filter(., adj.P.Val < 0.05)
#should be 8334 genes total after filtering

DOX24_DEGs_up <- DOX24_DEGs %>% 
  dplyr::filter(., logFC > 0)  %>% 
  rownames_to_column(., var = "entrezgene_ID") %>% 
  dplyr::select("entrezgene_ID")
#has 4029 genes

DOX24_DEGs_down <- DOX24_DEGs %>% 
  dplyr::filter(., logFC < 0)  %>% 
  rownames_to_column(., var = "entrezgene_ID") %>% 
  dplyr::select("entrezgene_ID")
#has 4305 genes

#total number of genes = 8334 -> 4029 + 4305 = 8334
  
DOX24R_DEGs <- top.table_V.D24r_dxr %>% 
  dplyr::filter(., adj.P.Val < 0.05) 
#should be 6755 genes after filtering

DOX24R_DEGs_up <- DOX24R_DEGs %>% 
  dplyr::filter(., logFC > 0)  %>% 
  rownames_to_column(., var = "entrezgene_ID") %>% 
  dplyr::select("entrezgene_ID")
#has 4731 genes

DOX24R_DEGs_down <- DOX24R_DEGs %>% 
  dplyr::filter(., logFC < 0)  %>% 
  rownames_to_column(., var = "entrezgene_ID") %>% 
  dplyr::select("entrezgene_ID")
#has 2024 genes

#total number of genes = 6775 -> 4731 + 2024 = 6775

DOX144R_DEGs <- top.table_V.D144r_dxr %>% 
  dplyr::filter(., adj.P.Val < 0.05)
#should be 303 genes total after filtering

DOX144R_DEGs_up <- DOX144R_DEGs %>% 
  dplyr::filter(., logFC > 0) %>% 
  rownames_to_column(., var = "entrezgene_ID") %>% 
  dplyr::select("entrezgene_ID")
#has 42 genes

DOX144R_DEGs_down <- DOX144R_DEGs %>% 
  dplyr::filter(., logFC < 0) %>% 
  rownames_to_column(., var = "entrezgene_ID") %>% 
  dplyr::select("entrezgene_ID")
#has 261 genes

#total number of genes = 303 -> 42 + 261 = 303


#Now that I've made these filtered groups of DEs based on direction, perform GO analysis

#####DOX24 Upregulated Genes#####
D24_DEGs_up_mat <- as.matrix(DOX24_DEGs_up)

DOX_24_up_dxr_gene <- gost(query = D24_DEGs_up_mat,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

DOX_24_up_gost_genes <- gostplot(DOX_24_up_dxr_gene, capped = FALSE, interactive = TRUE)
DOX_24_up_gost_genes

table_DOX24_up_genes <- DOX_24_up_dxr_gene$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

table_DOX24_up_genes %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(table_DOX24_up_genes, "output/table_DOX24_upreg_genes.csv")

#GO:BP
table_DOX24_up_genes_GOBP <- table_DOX24_up_genes %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(table_motif1_GOBP_d, "data/table_motif1_GOBP_d.RDS")

table_DOX24_up_genes_GOBP %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24 Up DEGs Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
table_DOX24_up_genes_KEGG <- table_DOX24_up_genes %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

table_DOX24_up_genes_KEGG %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24 Up DEGs Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#####DOX24 Downregulated Genes#####
D24_DEGs_down_mat <- as.matrix(DOX24_DEGs_down)

DOX_24_down_dxr_gene <- gost(query = D24_DEGs_down_mat,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

DOX_24_down_gost_genes <- gostplot(DOX_24_down_dxr_gene, capped = FALSE, interactive = TRUE)
DOX_24_down_gost_genes

table_DOX24_down_genes <- DOX_24_down_dxr_gene$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

table_DOX24_down_genes %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(table_DOX24_down_genes, "output/table_DOX24_downreg_genes.csv")

#GO:BP
table_DOX24_down_genes_GOBP <- table_DOX24_down_genes %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(table_motif1_GOBP_d, "data/table_motif1_GOBP_d.RDS")

table_DOX24_down_genes_GOBP %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24 Down DEGs Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
table_DOX24_down_genes_KEGG <- table_DOX24_down_genes %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

table_DOX24_down_genes_KEGG %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24 Down DEGs Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))


#####DOX24R Upregulated DEGs GO KEGG#####
D24R_DEGs_up_mat <- as.matrix(DOX24R_DEGs_up)

DOX_24R_up_dxr_gene <- gost(query = D24R_DEGs_up_mat,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

DOX_24R_up_gost_genes <- gostplot(DOX_24R_up_dxr_gene, capped = FALSE, interactive = TRUE)
DOX_24R_up_gost_genes

table_DOX24R_up_genes <- DOX_24R_up_dxr_gene$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

table_DOX24R_up_genes %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(table_DOX24R_up_genes, "output/table_DOX24R_upreg_genes.csv")

#GO:BP
table_DOX24R_up_genes_GOBP <- table_DOX24R_up_genes %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(table_motif1_GOBP_d, "data/table_motif1_GOBP_d.RDS")

table_DOX24R_up_genes_GOBP %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24R Up DEGs Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
table_DOX24R_up_genes_KEGG <- table_DOX24R_up_genes %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

table_DOX24R_up_genes_KEGG %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24R Up DEGs Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#####DOX24R Downregulated DEGs GO KEGG#####
D24R_DEGs_down_mat <- as.matrix(DOX24R_DEGs_down)

DOX_24R_down_dxr_gene <- gost(query = D24R_DEGs_down_mat,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

DOX_24R_down_gost_genes <- gostplot(DOX_24R_down_dxr_gene, capped = FALSE, interactive = TRUE)
DOX_24R_down_gost_genes

table_DOX24R_down_genes <- DOX_24R_down_dxr_gene$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

table_DOX24R_down_genes %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(table_DOX24R_down_genes, "output/table_DOX24R_downreg_genes.csv")

#GO:BP
table_DOX24R_down_genes_GOBP <- table_DOX24R_down_genes %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(table_motif1_GOBP_d, "data/table_motif1_GOBP_d.RDS")

table_DOX24R_down_genes_GOBP %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24R Down DEGs Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
table_DOX24R_down_genes_KEGG <- table_DOX24R_down_genes %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

table_DOX24R_down_genes_KEGG %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24R Down DEGs Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))


#####DOX144R Upregulated DEGs GO KEGG#####
DOX144R_DEGs_up_mat <- as.matrix(DOX144R_DEGs_up)

DOX_144R_up_dxr_gene <- gost(query = DOX144R_DEGs_up_mat,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

DOX_144R_up_gost_genes <- gostplot(DOX_144R_up_dxr_gene, capped = FALSE, interactive = TRUE)
DOX_144R_up_gost_genes

table_DOX144R_up_genes <- DOX_144R_up_dxr_gene$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

table_DOX144R_up_genes %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(table_DOX144R_up_genes, "output/table_DOX144R_upreg_genes.csv")

#GO:BP
table_DOX144R_up_genes_GOBP <- table_DOX144R_up_genes %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(table_motif1_GOBP_d, "data/table_motif1_GOBP_d.RDS")

table_DOX144R_up_genes_GOBP %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX144R Up DEGs Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
table_DOX144R_up_genes_KEGG <- table_DOX144R_up_genes %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#Not getting any results for this plot - so I'm commenting it out

# table_DOX144R_up_genes_KEGG %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
#   geom_point(aes(size = intersection_size)) +
#   ggtitle("DOX144R Up DEGs Enriched KEGG Terms")+
#   xlab(expression("-log"[10]~"p-value"))+
#   guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
#   ylab("KEGG term")+
#   scale_y_discrete(labels = scales::label_wrap(30))+
#   theme_bw()+
#   theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
#         axis.title = element_text(size = 15, colour = "black"),
#         axis.ticks = element_line(linewidth = 1.5),
#         axis.line = element_line(linewidth = 1.5),
#         axis.text = element_text(size = 10, colour = "black", angle = 0),
#         strip.text = element_text(size = 15, colour = "black", face = "bold"))

#####DOX144R Downregulated DEGs GO KEGG#####
DOX144R_DEGs_down_mat <- as.matrix(DOX144R_DEGs_down)

DOX_144R_down_dxr_gene <- gost(query = DOX144R_DEGs_down_mat,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

DOX_144R_down_gost_genes <- gostplot(DOX_144R_down_dxr_gene, capped = FALSE, interactive = TRUE)
DOX_144R_down_gost_genes

table_DOX144R_down_genes <- DOX_144R_down_dxr_gene$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

table_DOX144R_down_genes %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(table_DOX144R_down_genes, "output/table_DOX144R_downreg_genes.csv")

#GO:BP
table_DOX144R_down_genes_GOBP <- table_DOX144R_down_genes %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(table_motif1_GOBP_d, "data/table_motif1_GOBP_d.RDS")

table_DOX144R_down_genes_GOBP %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX144R Down DEGs Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
table_DOX144R_down_genes_KEGG <- table_DOX144R_down_genes %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

table_DOX144R_down_genes_KEGG %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX144R Down DEGs Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

```



```{r GO KEGG DE Genes}

DOX_24_top_dxr_1 <- as.matrix(DEG_1_dxr)

DOX_24_top_dxr_gene <- gost(query = DEG_1_dxr,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

DOX_24_gost_genes <- gostplot(DOX_24_top_dxr_gene, capped = FALSE, interactive = TRUE)
DOX_24_gost_genes

table_DOX24_genes <- DOX_24_top_dxr_gene$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

table_DOX24_genes %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(table_DOX24_genes, "output/table_DOX24_genes.csv")

#GO:BP
table_DOX24_genes_GOBP <- table_DOX24_genes %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(table_motif1_GOBP_d, "data/table_motif1_GOBP_d.RDS")

table_DOX24_genes_GOBP %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24 DEGs Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
table_DOX24_genes_KEGG <- table_DOX24_genes %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

table_DOX24_genes_KEGG %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24 DEGs Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#####DOX24R GO KEGG#####
DOX_24R_top_dxr_1 <- as.matrix(DEG_2_dxr)

DOX_24R_top_dxr_gene <- gost(query = DOX_24R_top_dxr_1,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

DOX_24R_gost_genes <- gostplot(DOX_24R_top_dxr_gene, capped = FALSE, interactive = TRUE)
DOX_24R_gost_genes

table_DOX24R_genes <- DOX_24R_top_dxr_gene$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

table_DOX24R_genes %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(table_DOX24_genes, "output/table_DOX24_genes.csv")

#GO:BP
table_DOX24R_genes_GOBP <- table_DOX24R_genes %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(table_motif1_GOBP_d, "data/table_motif1_GOBP_d.RDS")

table_DOX24R_genes_GOBP %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24R DEGs Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
table_DOX24R_genes_KEGG <- table_DOX24R_genes %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

table_DOX24R_genes_KEGG %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX24R DEGs Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#####DOX144R GO KEGG#####
DOX_144R_top_dxr_1 <- as.matrix(DEG_3_dxr)

DOX_144R_top_dxr_gene <- gost(query = DOX_144R_top_dxr_1,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

DOX_144R_gost_genes <- gostplot(DOX_144R_top_dxr_gene, capped = FALSE, interactive = TRUE)
DOX_144R_gost_genes

table_DOX144R_genes <- DOX_144R_top_dxr_gene$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

table_DOX144R_genes %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(table_DOX24_genes, "output/table_DOX24_genes.csv")

#GO:BP
table_DOX144R_genes_GOBP <- table_DOX144R_genes %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(table_motif1_GOBP_d, "data/table_motif1_GOBP_d.RDS")

table_DOX144R_genes_GOBP %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX144R DEGs Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
table_DOX144R_genes_KEGG <- table_DOX144R_genes %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

table_DOX144R_genes_KEGG %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("DOX144R DEGs Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))


```

```{r DEG Comparisons LogFC}

####DOX24 DEG Matrix####
#Extract the gene IDs from each motif

##DOX24
DOX24_genes <- DEG_1_dxr

##DOX24R 
DOX24R_genes <- DEG_2_dxr

##DOX144R
DOX144R_genes <- DEG_3_dxr


#Combine the toptables I have from pairwise analysis into a single dataframe
DOX24_toptable <- top.table_V.D24_dxr %>% 
  rownames_to_column(var = "entrezgene_ID") %>% 
  mutate(Time = "24")

DOX24R_toptable <- top.table_V.D24r_dxr %>% 
  rownames_to_column(var = "entrezgene_ID") %>% 
  mutate(Time = "24R")

DOX144R_toptable <- top.table_V.D144r_dxr %>% 
  rownames_to_column(var = "entrezgene_ID") %>% 
  mutate(Time = "144R")

combined_toptables_dxr <- bind_rows(
  DOX24_toptable,
  DOX24R_toptable,
  DOX144R_toptable)

#Filter the data based on each motif
filt_toptable_dxr <- combined_toptables_dxr %>% 
  dplyr::filter(entrezgene_ID  %in% DOX24_genes) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = logFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("logFC")+
  theme_bw()+
  ggtitle("LogFC for all genes DOX24T")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))

#motif 2
filt_toptable_dxr_2 <- combined_toptables_dxr %>% 
  dplyr::filter(entrezgene_ID  %in% DOX24R_genes) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = logFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("logFC")+
  theme_bw()+
  ggtitle("LogFC for all genes DOX24R")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))

#DOX144R
filt_toptable_dxr_3 <- combined_toptables_dxr %>% 
  dplyr::filter(entrezgene_ID  %in% DOX144R_genes) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = logFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("logFC")+
  theme_bw()+
  ggtitle("LogFC for all genes DOX144R")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))

#plots
filt_toptable_dxr
filt_toptable_dxr_2
filt_toptable_dxr_3


#now plot the abs logFC for each of these too
filt_toptable_abs_dxr <- combined_toptables_dxr %>% 
  dplyr::filter(entrezgene_ID  %in% DOX24_genes) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = absFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("|logFC|")+
  theme_bw()+
  ggtitle("Abs LogFC for all genes DOX24T")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))

#motif 2
filt_toptable_2_abs_dxr <- combined_toptables_dxr %>% 
  dplyr::filter(entrezgene_ID  %in% DOX24R_genes) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = absFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("|logFC|")+
  theme_bw()+
  ggtitle("Abs LogFC for all genes DOX24R")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))

#DOX144R
filt_toptable_3_abs_dxr <- combined_toptables_dxr %>% 
  dplyr::filter(entrezgene_ID  %in% DOX144R_genes) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = absFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("|logFC|")+
  theme_bw()+
  ggtitle("Abs LogFC for all genes DOX144R")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))


#plots
filt_toptable_abs_dxr
filt_toptable_2_abs_dxr
filt_toptable_3_abs_dxr


```



```{r Cormotif Function, eval=TRUE, warning=FALSE, include=FALSE}
limmafit.default <- function(exprs, groupid, compid) {
  limmafits <- list()
  compnum <- nrow(compid)
  genenum <- nrow(exprs)
  limmat <- matrix(0, genenum,compnum)
  limmas2 <- rep(0, compnum)
  limmadf <- rep(0, compnum)
  limmav0    <- rep(0,compnum)
  limmag1num <- rep(0,compnum)
  limmag2num <- rep(0,compnum)

  rownames(limmat)  <- rownames(exprs)
  colnames(limmat)  <- rownames(compid)
  names(limmas2)    <- rownames(compid)
  names(limmadf)    <- rownames(compid)
  names(limmav0)    <- rownames(compid)
  names(limmag1num) <- rownames(compid)
  names(limmag2num) <- rownames(compid)
  
  for(i in 1:compnum) {
    selid1 <- which(groupid == compid[i,1])
    selid2 <- which(groupid == compid[i,2])
    eset   <- new("ExpressionSet", exprs=cbind(exprs[,selid1],exprs[,selid2]))
    g1num  <- length(selid1)
    g2num  <- length(selid2)
    designmat <- cbind(base=rep(1,(g1num+g2num)), delta=c(rep(0,g1num),rep(1,g2num)))
    fit <- lmFit(eset,designmat)
    fit <- eBayes(fit)
    limmat[,i] <- fit$t[,2]
    limmas2[i] <- fit$s2.prior
    limmadf[i] <- fit$df.prior
    limmav0[i] <- fit$var.prior[2]
    limmag1num[i] <- g1num
    limmag2num[i] <- g2num
    limmafits[[i]] <- fit
  }
  names(limmafits) <- rownames(compid)
  limmacompnum<-nrow(compid)
  result<-list(t       = limmat,
               v0      = limmav0,
               df0     = limmadf,
               s20     = limmas2,
               g1num   = limmag1num,
               g2num   = limmag2num,
               compnum = limmacompnum,
               fits    = limmafits)
}

limmafit.counts <-
  function (exprs, groupid, compid, norm.factor.method = "TMM", voom.normalize.method = "none")
  {
    limmafits  <- list()
    compnum    <- nrow(compid)
    genenum    <- nrow(exprs)
    limmat     <- matrix(NA,genenum,compnum)
    limmas2    <- rep(0,compnum)
    limmadf    <- rep(0,compnum)
    limmav0    <- rep(0,compnum)
    limmag1num <- rep(0,compnum)
    limmag2num <- rep(0,compnum)

    rownames(limmat)  <- rownames(exprs)
    colnames(limmat)  <- rownames(compid)
    names(limmas2)    <- rownames(compid)
    names(limmadf)    <- rownames(compid)
    names(limmav0)    <- rownames(compid)
    names(limmag1num) <- rownames(compid)
    names(limmag2num) <- rownames(compid)

    for (i in 1:compnum) {
      message(paste("Running limma for comparision",i,"/",compnum))
      selid1 <- which(groupid == compid[i, 1])
      selid2 <- which(groupid == compid[i, 2])
      # make a new count data frame
      counts <- cbind(exprs[, selid1], exprs[, selid2])

      # remove NAs
      not.nas <- which(apply(counts, 1, function(x) !any(is.na(x))) == TRUE)

      # runn voom/limma
      d <- DGEList(counts[not.nas,])
      d <- calcNormFactors(d, method = norm.factor.method)
      g1num <- length(selid1)
      g2num <- length(selid2)
      designmat <- cbind(base = rep(1, (g1num + g2num)), delta = c(rep(0,
                                                                       g1num), rep(1, g2num)))

      y <- voom(d, designmat, normalize.method = voom.normalize.method)
      fit <- lmFit(y, designmat)
      fit <- eBayes(fit)

      limmafits[[i]] <- fit
      limmat[not.nas, i] <- fit$t[, 2]
      limmas2[i] <- fit$s2.prior
      limmadf[i] <- fit$df.prior
      limmav0[i] <- fit$var.prior[2]
      limmag1num[i] <- g1num
      limmag2num[i] <- g2num
    }
    limmacompnum <- nrow(compid)
    names(limmafits) <- rownames(compid)
    result <- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)
  }


limmafit.list <-
  function (fitlist, cmp.idx=2)
  {
    compnum    <- length(fitlist)

    genes <- c()
    for (i in 1:compnum) genes <- unique(c(genes, rownames(fitlist[[i]])))

    genenum    <- length(genes)
    limmat     <- matrix(NA,genenum,compnum)
    limmas2    <- rep(0,compnum)
    limmadf    <- rep(0,compnum)
    limmav0    <- rep(0,compnum)
    limmag1num <- rep(0,compnum)
    limmag2num <- rep(0,compnum)

    rownames(limmat)  <- genes
    colnames(limmat)  <- names(fitlist)
    names(limmas2)    <- names(fitlist)
    names(limmadf)    <- names(fitlist)
    names(limmav0)    <- names(fitlist)
    names(limmag1num) <- names(fitlist)
    names(limmag2num) <- names(fitlist)

    for (i in 1:compnum) {
      this.t <- fitlist[[i]]$t[,cmp.idx]
      limmat[names(this.t),i] <- this.t

      limmas2[i]    <- fitlist[[i]]$s2.prior
      limmadf[i]    <- fitlist[[i]]$df.prior
      limmav0[i]    <- fitlist[[i]]$var.prior[cmp.idx]
      limmag1num[i] <- sum(fitlist[[i]]$design[,cmp.idx]==0)
      limmag2num[i] <- sum(fitlist[[i]]$design[,cmp.idx]==1)
    }

    limmacompnum <- compnum
    result <- list(t       = limmat,
                   v0      = limmav0,
                   df0     = limmadf,
                   s20     = limmas2,
                   g1num   = limmag1num,
                   g2num   = limmag2num,
                   compnum = limmacompnum,
                   fits    = limmafits)

  }


generank<-function(x) {
  xcol<-ncol(x)
  xrow<-nrow(x)
  result<-matrix(0,xrow,xcol)
  z<-(1:1:xrow)
  for(i in 1:xcol) {
    y<-sort(x[,i],decreasing=TRUE,na.last=TRUE)
    result[,i]<-match(x[,i],y)
    result[,i]<-order(result[,i])
  }
  result
}

## Log-likelihood for moderated t under H0
modt.f0.loglike<-function(x,df) {
  a<-dt(x, df, log=TRUE)
  result<-as.vector(a)
  flag<-which(is.na(result)==TRUE)
  result[flag]<-0
  result
}

## Log-likelihood for moderated t under H1
## param=c(df,g1num,g2num,v0)
modt.f1.loglike<-function(x,param) {
  df<-param[1]
  g1num<-param[2]
  g2num<-param[3]
  v0<-param[4]
  w<-sqrt(1+v0/(1/g1num+1/g2num))
  dt(x/w, df, log=TRUE)-log(w)
  a<-dt(x/w, df, log=TRUE)-log(w)
  result<-as.vector(a)
  flag<-which(is.na(result)==TRUE)
  result[flag]<-0
  result
}

## Correlation Motif Fit
cmfit.X<-function(x, type, K=1, tol=1e-3, max.iter=100) {
  ## initialize
  xrow <- nrow(x)
  xcol <- ncol(x)
  loglike0 <- list()
  loglike1 <- list()
  p <- rep(1, K)/K
  q <- matrix(runif(K * xcol), K, xcol)
  q[1, ] <- rep(0.01, xcol)
  for (i in 1:xcol) {
    f0 <- type[[i]][[1]]
    f0param <- type[[i]][[2]]
    f1 <- type[[i]][[3]]
    f1param <- type[[i]][[4]]
    loglike0[[i]] <- f0(x[, i], f0param)
    loglike1[[i]] <- f1(x[, i], f1param)
  }
  condlike <- list()
  for (i in 1:xcol) {
    condlike[[i]] <- matrix(0, xrow, K)
  }
  loglike.old <- -1e+10
  for (i.iter in 1:max.iter) {
    if ((i.iter%%50) == 0) {
      print(paste("We have run the first ", i.iter, " iterations for K=",
                  K, sep = ""))
    }
    err <- tol + 1
    clustlike <- matrix(0, xrow, K)
    #templike <- matrix(0, xrow, 2)
    templike1 <- rep(0, xrow)
    templike2 <- rep(0, xrow)
    for (j in 1:K) {
      for (i in 1:xcol) {
        templike1 <- log(q[j, i]) + loglike1[[i]]
        templike2 <- log(1 - q[j, i]) + loglike0[[i]]
        tempmax <- Rfast::Pmax(templike1, templike2)

        templike1 <- exp(templike1 - tempmax)
        templike2 <- exp(templike2 - tempmax)

        tempsum <- templike1 + templike2
        clustlike[, j] <- clustlike[, j] + tempmax +
          log(tempsum)
        condlike[[i]][, j] <- templike1/tempsum
      }
      clustlike[, j] <- clustlike[, j] + log(p[j])
    }
    #tempmax <- apply(clustlike, 1, max)
    tempmax <- Rfast::rowMaxs(clustlike, value=TRUE)
    for (j in 1:K) {
      clustlike[, j] <- exp(clustlike[, j] - tempmax)
    }
    #tempsum <- apply(clustlike, 1, sum)
    tempsum <- Rfast::rowsums(clustlike)
    for (j in 1:K) {
      clustlike[, j] <- clustlike[, j]/tempsum
    }
    #p.new <- (apply(clustlike, 2, sum) + 1)/(xrow + K)
    p.new <- (Rfast::colsums(clustlike) + 1)/(xrow + K)
    q.new <- matrix(0, K, xcol)
    for (j in 1:K) {
      clustpsum <- sum(clustlike[, j])
      for (i in 1:xcol) {
        q.new[j, i] <- (sum(clustlike[, j] * condlike[[i]][,
                                                           j]) + 1)/(clustpsum + 2)
      }
    }
    err.p <- max(abs(p.new - p)/p)
    err.q <- max(abs(q.new - q)/q)
    err <- max(err.p, err.q)
    loglike.new <- (sum(tempmax + log(tempsum)) + sum(log(p.new)) +
                      sum(log(q.new) + log(1 - q.new)))/xrow
    p <- p.new
    q <- q.new
    loglike.old <- loglike.new
    if (err < tol) {
      break
    }
  }
  clustlike <- matrix(0, xrow, K)
  for (j in 1:K) {
    for (i in 1:xcol) {
      templike1 <- log(q[j, i]) + loglike1[[i]]
      templike2 <- log(1 - q[j, i]) + loglike0[[i]]
      tempmax <- Rfast::Pmax(templike1, templike2)

      templike1 <- exp(templike1 - tempmax)
      templike2 <- exp(templike2 - tempmax)

      tempsum <- templike1 + templike2
      clustlike[, j] <- clustlike[, j] + tempmax + log(tempsum)
      condlike[[i]][, j] <- templike1/tempsum
    }
    clustlike[, j] <- clustlike[, j] + log(p[j])
  }
  #tempmax <- apply(clustlike, 1, max)
  tempmax <- Rfast::rowMaxs(clustlike, value=TRUE)
  for (j in 1:K) {
    clustlike[, j] <- exp(clustlike[, j] - tempmax)
  }
  #tempsum <- apply(clustlike, 1, sum)
  tempsum <- Rfast::rowsums(clustlike)
  for (j in 1:K) {
    clustlike[, j] <- clustlike[, j]/tempsum
  }
  p.post <- matrix(0, xrow, xcol)
  for (j in 1:K) {
    for (i in 1:xcol) {
      p.post[, i] <- p.post[, i] + clustlike[, j] * condlike[[i]][,
                                                                  j]
    }
  }
  loglike.old <- loglike.old - (sum(log(p)) + sum(log(q) +
                                                    log(1 - q)))/xrow
  loglike.old <- loglike.old * xrow
  result <- list(p.post = p.post, motif.prior = p, motif.q = q,
                 loglike = loglike.old, clustlike=clustlike, condlike=condlike)
}

## Fit using (0,0,...,0) and (1,1,...,1)
cmfitall<-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow<-nrow(x)
  xcol<-ncol(x)
  loglike0<-list()
  loglike1<-list()
  p<-0.01

  ## compute loglikelihood
  L0<-matrix(0,xrow,1)
  L1<-matrix(0,xrow,1)
  for(i in 1:xcol) {
    f0<-type[[i]][[1]]
    f0param<-type[[i]][[2]]
    f1<-type[[i]][[3]]
    f1param<-type[[i]][[4]]
    loglike0[[i]]<-f0(x[,i],f0param)
    loglike1[[i]]<-f1(x[,i],f1param)
    L0<-L0+loglike0[[i]]
    L1<-L1+loglike1[[i]]
  }


  ## EM algorithm to get MLE of p and q
  loglike.old <- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste("We have run the first ", i.iter, " iterations",sep=""))
    }
    err<-tol+1

    ## compute posterior cluster membership
    clustlike<-matrix(0,xrow,2)
    clustlike[,1]<-log(1-p)+L0
    clustlike[,2]<-log(p)+L1

    tempmax<-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]<-exp(clustlike[,j]-tempmax)
    }
    tempsum<-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:2) {
      clustlike[,j]<-clustlike[,j]/tempsum
    }

    p.new<-(sum(clustlike[,2])+1)/(xrow+2)

    ## evaluate convergence
    err<-abs(p.new-p)/p

    ## evaluate whether the log.likelihood increases
    loglike.new<-(sum(tempmax+log(tempsum))+log(p.new)+log(1-p.new))/xrow

    loglike.old<-loglike.new
    p<-p.new

    if(err<tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike<-matrix(0,xrow,2)
  clustlike[,1]<-log(1-p)+L0
  clustlike[,2]<-log(p)+L1

  tempmax<-apply(clustlike,1,max)
  for(j in 1:2) {
    clustlike[,j]<-exp(clustlike[,j]-tempmax)
  }
  tempsum<-apply(clustlike,1,sum)

  for(j in 1:2) {
    clustlike[,j]<-clustlike[,j]/tempsum
  }

  p.post<-matrix(0,xrow,xcol)
  for(i in 1:xcol) {
    p.post[,i]<-clustlike[,2]
  }

  ## return

  #calculate back loglikelihood
  loglike.old<-loglike.old-(log(p)+log(1-p))/xrow
  loglike.old<-loglike.old*xrow
  result<-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

## Fit each dataset separately
cmfitsep<-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow<-nrow(x)
  xcol<-ncol(x)
  loglike0<-list()
  loglike1<-list()
  p<-0.01*rep(1,xcol)
  loglike.final<-rep(0,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0<-type[[i]][[1]]
    f0param<-type[[i]][[2]]
    f1<-type[[i]][[3]]
    f1param<-type[[i]][[4]]
    loglike0[[i]]<-f0(x[,i],f0param)
    loglike1[[i]]<-f1(x[,i],f1param)
  }

  p.post<-matrix(0,xrow,xcol)

  ## EM algorithm to get MLE of p
  for(coli in 1:xcol) {
    loglike.old <- -1e10
    for(i.iter in 1:max.iter) {
      if((i.iter%%50) == 0) {
        print(paste("We have run the first ", i.iter, " iterations",sep=""))
      }
      err<-tol+1

      ## compute posterior cluster membership
      clustlike<-matrix(0,xrow,2)
      clustlike[,1]<-log(1-p[coli])+loglike0[[coli]]
      clustlike[,2]<-log(p[coli])+loglike1[[coli]]

      tempmax<-apply(clustlike,1,max)
      for(j in 1:2) {
        clustlike[,j]<-exp(clustlike[,j]-tempmax)
      }
      tempsum<-apply(clustlike,1,sum)

      ## evaluate whether the log.likelihood increases
      loglike.new<-sum(tempmax+log(tempsum))/xrow

      ## update motif occurrence rate
      for(j in 1:2) {
        clustlike[,j]<-clustlike[,j]/tempsum
      }

      p.new<-(sum(clustlike[,2]))/(xrow)

      ## evaluate convergence
      err<-abs(p.new-p[coli])/p[coli]
      loglike.old<-loglike.new
      p[coli]<-p.new

      if(err<tol) {
        break;
      }
    }

    ## compute posterior p
    clustlike<-matrix(0,xrow,2)
    clustlike[,1]<-log(1-p[coli])+loglike0[[coli]]
    clustlike[,2]<-log(p[coli])+loglike1[[coli]]

    tempmax<-apply(clustlike,1,max)
    for(j in 1:2) {
      clustlike[,j]<-exp(clustlike[,j]-tempmax)
    }
    tempsum<-apply(clustlike,1,sum)

    for(j in 1:2) {
      clustlike[,j]<-clustlike[,j]/tempsum
    }

    p.post[,coli]<-clustlike[,2]
    loglike.final[coli]<-loglike.old
  }


  ## return
  loglike.final<-loglike.final*xrow
  result<-list(p.post=p.post, motif.prior=p, loglike=loglike.final)
}

## Fit the full model
cmfitfull<-function(x, type, tol=1e-3, max.iter=100) {
  ## initialize
  xrow<-nrow(x)
  xcol<-ncol(x)
  loglike0<-list()
  loglike1<-list()
  K<-2^xcol
  p<-rep(1,K)/K
  pattern<-rep(0,xcol)
  patid<-matrix(0,K,xcol)

  ## compute loglikelihood
  for(i in 1:xcol) {
    f0<-type[[i]][[1]]
    f0param<-type[[i]][[2]]
    f1<-type[[i]][[3]]
    f1param<-type[[i]][[4]]
    loglike0[[i]]<-f0(x[,i],f0param)
    loglike1[[i]]<-f1(x[,i],f1param)
  }
  L<-matrix(0,xrow,K)
  for(i in 1:K)
  {
    patid[i,]<-pattern
    for(j in 1:xcol) {
      if(pattern[j] < 0.5) {
        L[,i]<-L[,i]+loglike0[[j]]
      } else {
        L[,i]<-L[,i]+loglike1[[j]]
      }
    }

    if(i < K) {
      pattern[xcol]<-pattern[xcol]+1
      j<-xcol
      while(pattern[j] > 1) {
        pattern[j]<-0
        j<-j-1
        pattern[j]<-pattern[j]+1
      }
    }
  }

  ## EM algorithm to get MLE of p and q
  loglike.old <- -1e10
  for(i.iter in 1:max.iter) {
    if((i.iter%%50) == 0) {
      print(paste("We have run the first ", i.iter, " iterations",sep=""))
    }
    err<-tol+1

    ## compute posterior cluster membership
    clustlike<-matrix(0,xrow,K)
    for(j in 1:K) {
      clustlike[,j]<-log(p[j])+L[,j]
    }

    tempmax<-apply(clustlike,1,max)
    for(j in 1:K) {
      clustlike[,j]<-exp(clustlike[,j]-tempmax)
    }
    tempsum<-apply(clustlike,1,sum)

    ## update motif occurrence rate
    for(j in 1:K) {
      clustlike[,j]<-clustlike[,j]/tempsum
    }

    p.new<-(apply(clustlike,2,sum)+1)/(xrow+K)

    ## evaluate convergence
    err<-max(abs(p.new-p)/p)

    ## evaluate whether the log.likelihood increases
    loglike.new<-(sum(tempmax+log(tempsum))+sum(log(p.new)))/xrow

    loglike.old<-loglike.new
    p<-p.new

    if(err<tol) {
      break;
    }
  }

  ## compute posterior p
  clustlike<-matrix(0,xrow,K)
  for(j in 1:K) {
    clustlike[,j]<-log(p[j])+L[,j]
  }

  tempmax<-apply(clustlike,1,max)
  for(j in 1:K) {
    clustlike[,j]<-exp(clustlike[,j]-tempmax)
  }
  tempsum<-apply(clustlike,1,sum)

  for(j in 1:K) {
    clustlike[,j]<-clustlike[,j]/tempsum
  }

  p.post<-matrix(0,xrow,xcol)
  for(j in 1:K) {
    for(i in 1:xcol) {
      if(patid[j,i] > 0.5) {
        p.post[,i]<-p.post[,i]+clustlike[,j]
      }
    }
  }

  ## return
  #calculate back loglikelihood
  loglike.old<-loglike.old-sum(log(p))/xrow
  loglike.old<-loglike.old*xrow
  result<-list(p.post=p.post, motif.prior=p, loglike=loglike.old)
}

generatetype<-function(limfitted)
{
  jtype<-list()
  df<-limfitted$g1num+limfitted$g2num-2+limfitted$df0
  for(j in 1:limfitted$compnum)
  {
    jtype[[j]]<-list(f0=modt.f0.loglike, f0.param=df[j], f1=modt.f1.loglike, f1.param=c(df[j],limfitted$g1num[j],limfitted$g2num[j],limfitted$v0[j]))
  }
  jtype
}

cormotiffit <- function(exprs, groupid=NULL, compid=NULL, K=1, tol=1e-3,
                        max.iter=100, BIC=TRUE, norm.factor.method="TMM",
                        voom.normalize.method = "none", runtype=c("logCPM","counts","limmafits"), each=3)
{
  #Input can be either a normalized matrix, a count matrix, or a list of limma fits. Dispatch the correct
  # limmafit accordingly.
  limfitted <- list()
  if (runtype=="counts") {
    limfitted <- limmafit.counts(exprs,groupid,compid, norm.factor.method, voom.normalize.method)
  } else if (runtype=="logCPM") {
    limfitted <- limmafit.default(exprs,groupid,compid)
  } else if (runtype=="limmafits") {
    limfitted <- limmafit.list(exprs)
  } else {
    stop("runtype must be one of 'logCPM', 'counts', or 'limmafits'")
  }


  jtype<-generatetype(limfitted)
  fitresult<-list()
  ks <- rep(K, each = each)
  fitresult <- bplapply(1:length(ks), function(i, x, type, ks, tol, max.iter) {
    cmfit.X(x, type, K = ks[i], tol = tol, max.iter = max.iter)
  }, x=limfitted$t, type=jtype, ks=ks, tol=tol, max.iter=max.iter)

  best.fitresults <- list()
  for (i in 1:length(K)) {
    w.k <- which(ks==K[i])
    this.bic <- c()
    for (j in w.k) this.bic[j] <- -2 * fitresult[[j]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
    w.min <- which(this.bic == min(this.bic, na.rm = TRUE))[1]
    best.fitresults[[i]] <- fitresult[[w.min]]
  }
  fitresult <- best.fitresults

  bic <- rep(0, length(K))
  aic <- rep(0, length(K))
  loglike <- rep(0, length(K))
  for (i in 1:length(K)) loglike[i] <- fitresult[[i]]$loglike
  for (i in 1:length(K)) bic[i] <- -2 * fitresult[[i]]$loglike + (K[i] - 1 + K[i] * limfitted$compnum) * log(dim(limfitted$t)[1])
  for (i in 1:length(K)) aic[i] <- -2 * fitresult[[i]]$loglike + 2 * (K[i] - 1 + K[i] * limfitted$compnum)
  if(BIC==TRUE) {
    bestflag=which(bic==min(bic))
  }
  else {
    bestflag=which(aic==min(aic))
  }
  result<-list(bestmotif=fitresult[[bestflag]],bic=cbind(K,bic),
               aic=cbind(K,aic),loglike=cbind(K,loglike), allmotifs=fitresult)

}

cormotiffitall<-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted<-limmafit(exprs,groupid,compid)
  jtype<-generatetype(limfitted)
  fitresult<-cmfitall(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitsep<-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted<-limmafit(exprs,groupid,compid)
  jtype<-generatetype(limfitted)
  fitresult<-cmfitsep(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

cormotiffitfull<-function(exprs,groupid,compid, tol=1e-3, max.iter=100)
{
  limfitted<-limmafit(exprs,groupid,compid)
  jtype<-generatetype(limfitted)
  fitresult<-cmfitfull(limfitted$t,type=jtype,tol=1e-3,max.iter=max.iter)
}

plotIC<-function(fitted_cormotif)
{
  oldpar<-par(mfrow=c(1,2))
  plot(fitted_cormotif$bic[,1], fitted_cormotif$bic[,2], type="b",xlab="Motif Number", ylab="BIC", main="BIC")
  plot(fitted_cormotif$aic[,1], fitted_cormotif$aic[,2], type="b",xlab="Motif Number", ylab="AIC", main="AIC")
}

plotMotif<-function(fitted_cormotif,title="")
{
  layout(matrix(1:2,ncol=2))
  u<-1:dim(fitted_cormotif$bestmotif$motif.q)[2]
  v<-1:dim(fitted_cormotif$bestmotif$motif.q)[1]
  image(u,v,t(fitted_cormotif$bestmotif$motif.q),
        col=gray(seq(from=1,to=0,by=-0.1)),xlab="Study",yaxt = "n",
        ylab="Corr. Motifs",main=paste(title,"pattern",sep=" "))
  axis(2,at=1:length(v))
  for(i in 1:(length(u)+1))
  {
    abline(v=(i-0.5))
  }
  for(i in 1:(length(v)+1))
  {
    abline(h=(i-0.5))
  }
  Ng=10000
  if(is.null(fitted_cormotif$bestmotif$p.post)!=TRUE)
    Ng=nrow(fitted_cormotif$bestmotif$p.post)
  genecount=floor(fitted_cormotif$bestmotif$motif.p*Ng)
  NK=nrow(fitted_cormotif$bestmotif$motif.q)
  plot(0,0.7,pch=".",xlim=c(0,1.2),ylim=c(0.75,NK+0.25),
       frame.plot=FALSE,axes=FALSE,xlab="No. of genes",ylab="", main=paste(title,"frequency",sep=" "))
  segments(0,0.7,fitted_cormotif$bestmotif$motif.p[1],0.7)
  rect(0,1:NK-0.3,fitted_cormotif$bestmotif$motif.p,1:NK+0.3,
       col="dark grey")
  mtext(1:NK,at=1:NK,side=2,cex=0.8)
  text(fitted_cormotif$bestmotif$motif.p+0.15,1:NK,
       labels=floor(fitted_cormotif$bestmotif$motif.p*Ng))
}
```



```{r Cormotif Data}
#function is on the previous

ind_num_norep <- c("1", "1", "1", "1", "1", "1", "2", "2", "2", "2", "2", "2", "3", "3", "3", "3", "3", "3", "4", "4", "4", "4", "4", "4", "5", "5", "5", "5", "5", "5", "6", "6", "6", "6", "6", "6")


annot_d <- data.frame("samples" = colnames(filt_counts_norep)) %>% separate_wider_delim(., cols = samples, names = c("Tx", "Time", "Ind"), delim = "_", cols_remove = FALSE) %>% 
  tidyr::unite(., col = "Tx_Time", Tx, Time, sep = "_", remove = FALSE) %>% cbind(., ind_num_norep)

#Cormotif_d <- cpm(dge_calc_dxr, log = TRUE)

#calcnormfactors here?
View(dge_calc_dxr$samples)

Cormotif_d <- cpm(dge_calc_dxr, log = TRUE)

Cormotif_df_d <- as.data.frame(Cormotif_d)
dim(Cormotif_df_d)

groupid <- rep(c(1, 2, 3, 4, 5, 6), 6)

compid <- data.frame(c1 = c(1, 3, 5), c2 = c(2, 4, 6))

```


```{r Cormotif Run, include=FALSE}

#set.seed(11111)
cormotif_dxr <- cormotiffit(exprs = Cormotif_d,
                            groupid = groupid,
                            compid = compid,
                            K=1:10,
                            max.iter = 1000,
                            runtype = "logCPM")

#only need to run this once! 
#save this to an object so I can retrieve it as needed

#saveRDS(cormotif_dxr, "data/cormotif_dxr.RDS")

#cormotif_dxr_fin <- readRDS("C:Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/cormotif_dxr.RDS")

```

```{r Cormotif Motifs}

plotIC(cormotif_dxr)

cormotif_dxr$bic

```

```{r Cormotif Motif Plots}

plotMotif(cormotif_dxr)

myColors <-  rev(c("#FFFFFF", "#E6E6E6" ,"#CCCCCC", "#B3B3B3", "#999999", "#808080", "#666666","#4C4C4C", "#333333", "#191919","#000000"))

plot.new()
legend('bottomleft',fill=myColors, legend =rev(c("0", "0.1", "0.2", "0.3", "0.4",  "0.5", "0.6", "0.7", "0.8","0.9", "1")), box.col="white",title = "Probability\nlegend", horiz=FALSE,title.cex=.8)

#now we're showing four motifs, with the same patterns I saw originally
#Motif 1 - NR - 11156 genes
#Motif 2 - AR - 326 genes
#Motif 3 - light early, dark in recovery timepoints (unnamed) - 739 genes
#Motif 4 - decreases over time - 1948 genes
```

```{r Cormotif Gene Probabilities}

topgenelist <- generank(cormotif_dxr_1$bestmotif$p.post)
rownames(topgenelist) <- rownames(Cormotif_d)

motif_prob <- cormotif_dxr_1$bestmotif$clustlike
rownames(motif_prob) <- rownames(topgenelist)
#saveRDS(motif_prob, "data/Cormotif_prob_gene_list.RDS")

#Define the gene probability groups - I have 4

####motif 1####
clust1_dxr <- motif_prob %>% 
  as.data.frame() %>% 
  filter(V1>0.5 & V2<0.5 & V3<0.5 & V4<0.5) %>% 
  rownames

length(clust1_dxr)
#11482 total genes
#clust1_dxr seems to be non response NR

####motif 2####
clust2_dxr <- motif_prob %>% 
  as.data.frame() %>% 
  filter(V2>0.1 & V1<0.5 & V3<0.5 & V4<0.5) %>% 
  rownames

#if I change the cutoff to 0.1 - I get 342 genes
#if I leave the cutoff as 0.5 - I get 0 genes

length(clust2_dxr)
#0
#342 with 0.1 cutoff
#is clust2 an acute response?

####motif 3####
clust3_dxr <- motif_prob %>% 
  as.data.frame() %>% 
  filter(V3>0.5 & V1<0.5 & V2<0.5 & V4<0.5) %>% 
  rownames

length(clust3_dxr)
#538 genes
#is clust3 a sustained response?

####motif 4####
clust4_dxr <- motif_prob %>% 
  as.data.frame() %>% 
  filter(V4>0.5 & V1<0.5 & V2<0.5 & V3<0.5) %>% 
  rownames

length(clust4_dxr)
#1576 genes
#is clust3 a sustained response?


clusterdata_dxr <- data.frame(
  Category = c("Motif 1", "Motif 2", "Motif 3", "Motif 4"), 
  Value = c(length(clust1_dxr), length(clust2_dxr), length(clust3_dxr), length(clust4_dxr))
)

piecolors_dxr <- c("Motif 1" = "#007896", 
                   "Motif 2" = "#58508D",
                   "Motif 3" = "#BC5090",
                   "Motif 4" = "#598643")

#make a piechart of these distributions
clusterdata_dxr %>% ggplot(aes(x = "", y = Value, fill = Category))+
  geom_bar(width = 1, stat = "identity")+
  coord_polar("y", start = 0)+
  geom_text(aes(label = Value),
            position = position_stack(vjust = 0.5),
            size = 4, color = "black")+
  labs(title = "Distribution of Gene Clusters Identified By Cormotif", x = NULL, y = NULL)+
  theme_void()+
  scale_fill_manual(values = piecolors_dxr)

```

```{r Cormotif Genes}

clust1_dxr_df <- as.data.frame(clust1_dxr)

#example gene - TXLNGY - 246126 - 
exgene_motif1 <- Cormotif_df_d %>% 
  rownames_to_column(var = "entrezgene_id") %>% 
  dplyr::filter(entrezgene_id == "246126")

exgene_motif1_long <- melt(exgene_motif1,
                         id.vars = c("entrezgene_id"),
                         variable.name = "Sample",
                         value.name = "log2cpm")

#now add in my factors like time, tx, tx_time, and ind by breaking up the Sample column
exgene_motif1_long_df <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                                  ind = factor(ind_names, levels = unique(ind_names)),
                                  txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                                  time = factor(time_names2, levels = unique(time_names2)))

exgene_motif1_long_factors <- cbind(exgene_motif1_long_df, exgene_motif1_long)

exgene_motif1_long_factors %>% ggplot(aes(x = txtime, y = log2cpm))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "TXLNGY")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(-10,10)+
  theme(plot.title = element_text(face = "italic"))

#saveRDS(CDKN1A_geneplot_Dox, "data/CDKN1A_geneplot_Dox.RDS")

####clust2####
clust2_d_df <- as.data.frame(clust2_d)

#example gene - BUB1B - 701 - 
exgene_motif2 <- Cormotif_df_d %>% 
  rownames_to_column(var = "entrezgene_id") %>% 
  dplyr::filter(entrezgene_id == "701")

exgene_motif2_long <- melt(exgene_motif2,
                         id.vars = c("entrezgene_id"),
                         variable.name = "Sample",
                         value.name = "log2cpm")

#now add in my factors like time, tx, tx_time, and ind by breaking up the Sample column
exgene_motif2_long_df <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                                  ind = factor(ind_names, levels = unique(ind_names)),
                                  txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                                  time = factor(time_names2, levels = unique(time_names2)))

exgene_motif2_long_factors <- cbind(exgene_motif2_long_df, exgene_motif2_long)

exgene_motif2_long_factors %>% ggplot(aes(x = txtime, y = log2cpm))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "BUB1B")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(-10,10)+
  theme(plot.title = element_text(face = "italic"))


```

```{r Cormotif logFC Plots}

#Extract the gene IDs from each motif

##motif 1
motif1_genes <- clust1_dxr_df

##motif 2 
motif2_genes <- clust2_dxr_df

##motif 3
motif3_genes <- clust3_dxr_df

##motif4
motif4_genes <- clust4_dxr_df


#Combine the toptables I have from pairwise analysis into a single dataframe
d24_toptable_dxr <- top.table_V.D24_dxr %>% 
  rownames_to_column(var = "entrezgene_ID") %>% 
  mutate(Time = "24")

d24r_toptable_dxr <- top.table_V.D24r_dxr %>% 
  rownames_to_column(var = "entrezgene_ID") %>% 
  mutate(Time = "24R")

d144r_toptable_dxr <- top.table_V.D144r_dxr %>% 
  rownames_to_column(var = "entrezgene_ID") %>% 
  mutate(Time = "144R")

combined_toptables_dxr <- bind_rows(
  d24_toptable_dxr,
  d24r_toptable_dxr,
  d144r_toptable_dxr)

#Filter the data based on each motif
filt_toptable_dxr <- combined_toptables_dxr %>% 
  dplyr::filter(entrezgene_ID  %in% clust1_dxr) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = logFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("logFC")+
  theme_bw()+
  ggtitle("LogFC for all genes in Motif 1")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))

#motif 2
filt_toptable_2_dxr <- combined_toptables_dxr %>% 
  dplyr::filter(entrezgene_ID  %in% clust2_dxr) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = logFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("logFC")+
  theme_bw()+
  ggtitle("LogFC for all genes in Motif 2")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))

#plots
filt_toptable_dxr
filt_toptable_2_dxr


#now plot the abs logFC for each of these too
filt_toptable_abs <- combined_toptables_dox %>% 
  dplyr::filter(entrezgene_ID  %in% clust1_d) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = absFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("|logFC|")+
  theme_bw()+
  ggtitle("Abs LogFC for all genes in Motif 1")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))

#motif 2
filt_toptable_2_abs <- combined_toptables_dox %>% 
  dplyr::filter(entrezgene_ID  %in% clust2_d) %>% 
  mutate(absFC = abs(logFC)) %>% 
  mutate(time = factor(Time, levels = c("24", "24R", "144R"), labels = c("24hr", "24hr Recovery", "144hr Recovery"))) %>% 
  ggplot(., aes(x = time, y = absFC))+
  geom_boxplot(aes(fill = time))+
  scale_fill_manual(values = time_col)+
  guides(fill = guide_legend(title = "Time"))+
  theme_bw()+
  xlab(" ")+
  ylab("|logFC|")+
  theme_bw()+
  ggtitle("Abs LogFC for all genes in Motif 2")+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        strip.background = element_rect(fill = "#CAD899"),
        axis.text.x = element_text(size = 8, colour = "white", angle = 15),
        strip.text.x = element_text(size = 12, colour = "black", face = "bold"))


#plots
filt_toptable_abs
filt_toptable_2_abs

```
