---
title: "Differential Expression Analysis - Recovery 5FU"
author: "Emma M Pfortmiller"
date: "2025-02-14"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r Load Libraries, warning=FALSE, include=FALSE}
library(edgeR)
library(tidyverse)
library(readr)
library(BiocGenerics)
library(gridExtra)
library(VennDiagram)
library(kableExtra)
library(scales)
library(ggVennDiagram)
library(Cormotif)
library(RColorBrewer)
library(ggpubr)

```

Create the Data Set:

```{r Data Set, include=FALSE}

counts_DE_raw <- read_csv("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_AllSamples_EMP_250210.csv")
#View(counts_DE)

#this dataset comes from the csv I saved of all raw counts for each sample - but for this initial try I would like to remove my biological replicate for now

counts_DE<- as.data.frame(Counts_Full_df) %>% 
        dplyr::select(-"...1") %>% 
        dplyr::select(-(contains("Ind6REP")))

#now I've removed that individual, so I'll filter all counts by rowmeans
#first check the number of variables in the counts file
##I should have 54 samples if not including Ind6REP
dim(counts_DE)
#[1] 78932    54
#[1] 28395    54 - new inbuilt

#now use rowMeans to filter out values with mean < 0 (this means rowMeans > 0 is what I'm left with) for each row/gene in the original file
rowMeans_DE <- rowMeans(counts_DE)
counts_DE_filter <- counts_DE[rowMeans_DE > 0,]
dim(counts_DE_filter)
#[1] 65925    54
#[1] 26445    54 - new inbuilt

#with this I've filtered out lowly expressed genes from my raw counts dataset

#I'll double check the appearance of the data after filtering and before filtering
boxplot(counts_DE, 
        main = "Recovery DE Unfiltered Counts", 
        xlab = "Conditions", 
        ylab = "Counts",
        ylim = c(-5,40))


boxplot(counts_DE_filter, 
        main = "Recovery DE Filtered Counts", 
        xlab = "Conditions", 
        ylab = "Counts",
        ylim = c(-5,40))

counts_DE_hist_mut <- (t(counts_DE))
counts_DE_fil_hist_mut <- (t(counts_DE_filter))

hist(counts_DE_hist_mut, main = "Histogram of Raw Counts", ylim = c(-100, 6000000), xlab = "counts", ylab = "Frequency", xlim = c(-1000,150000))

hist(counts_DE_fil_hist_mut, main = "Histogram of Filtered Counts", ylim = c(-100, 6000000), xlab = "counts", ylab = "Frequency", xlim = c(-1000,150000))


#now I'll save this dataset as a csv
#write.csv(counts_DE_filter,  "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/Counts_Matrix_DE_filtered_EMP_250220.csv")

```


Now that I've confirmed that the filtering looks good, let's continue on with the pipeline for DE
```{r Create DGE List Object}

de_list <- DGEList(counts_DE)

#de_list <- DGEList(counts_DE_filter)
#de_list_genes <- DGEList(counts_DE_filter, genes = genes)

#I want to include the genes information, so I'll call that column alone to be its own list
genes <- counts_DE %>% rownames_to_column(var = "ensembl_gene_id") %>% dplyr::select("ensembl_gene_id")

#if this comes in handy later - we'll see
de_list_genes <- DGEList(counts_DE, genes = genes)

```


Now that I've made my DGElist object, let's calculate normalization factors
#calcNormFactors doesnâ€™t normalize the data, it just calculates normalization factors for use downstream#
```{r Normalization Factors Calculation}

#this is the inital file before norm factors are calculated
de_list$samples

#calculate the normalization factors with method TMM
de_list_calc <- calcNormFactors(de_list, method = "TMM")

#final file after norm calculation
de_list_calc$samples

#View(de_list_genes_calc)

```

Now I'll look at the factors I have and pull those out
- Time
- Treatment
- Individual
```{r Pull Out Factors}

snames <- data.frame("samples" = colnames(dge_calc)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)

#snames_list <- as.list(snames)

snames_time <- snames$Time
snames_tx <- snames$Treatment
snames_ind <- snames$Individual


#define colors for each of these factors
#time colors - have 3 timepoints
time_col <- list(Time = c("24hr" = "#046A38", "24rec" = "#0050B5", "144rec" = "#B3831B"))
#treatment colors - have 3 treatments
tx_col <- list(Treatment = c("DMSO" = "#63666D","5FU" = "#DCACED","DOX" = "#499FBD"))
#individual colors - have 6 individual not including rep at this time
ind_col <- list(Individual = c("Ind1" = "#003F5C", "Ind2" = "#45AE91", "Ind3" =  "#58508D", "Ind4" = "#BC4099", "Ind5" =  "#FF6361", "Ind6" = "#FF2362"))
#all colors together 
all_col <- list(time_col, tx_col, ind_col)

#create a new variable that groups these factors all together
group2 <- interaction(snames_tx, snames_time)
group3 <- interaction(snames_tx, snames_time, snames_ind)

```

Now let's filter out those lowly-expressed genes according to the limma pipeline
```{r Filter Lowly Expressed Genes}

####rowMeans > 0####
rowMeans_DE <- rowMeans(de_list_calc$counts)
de_list_calc_fil <- de_list_calc[rowMeans_DE > 0,]
dim(de_list_calc_fil)
#[1] 65925    54

####rowMeans > 1####
rowMeans_DE <- rowMeans(de_list_calc$counts)
de_list_calc_fil_1 <- de_list_calc[rowMeans_DE > 1,]
dim(de_list_calc_fil_1)
#[1] 33360    54

####rowMeans > 2####
rowMeans_DE <- rowMeans(de_list_calc$counts)
de_list_calc_fil_2 <- de_list_calc[rowMeans_DE > 2,]
dim(de_list_calc_fil_2)
#[1] 28917    54

####rowMeans > 3####
rowMeans_DE <- rowMeans(de_list_calc$counts)
de_list_calc_fil_3 <- de_list_calc[rowMeans_DE > 3,]
dim(de_list_calc_fil_3)
#[1] 26460   54

####rowMeans > 4####
rowMeans_DE <- rowMeans(de_list_calc$counts)
de_list_calc_fil_4 <- de_list_calc[rowMeans_DE > 4,]
dim(de_list_calc_fil_4)
#[1] 24807   54

```


```{r Multi-dimensional Scaling Plot (MDS)}

plotMDS(dge_calc, col = as.numeric(group_1))

plotMDS(de_list_calc_fil, col = as.numeric(group3))

plotMDS(de_list_calc_fil, col = as.numeric(group2))

plotMDS(de_list_calc_fil, col = as.numeric(group3))

```

Now that I've put my matrix together, we can start with Voom transformation and linear modeling
#this specifies a model where each coefficient corresponds to a group mean
```{r Voom Transformation and Linear Modeling}
#put together your model
model_all <- model.matrix(~0 + group2)
model_all

####Voom####

y <- voom(de_list_calc_fil_3, model_all, plot = T)
```

```{r Counts Set for DEG}
group1 <- interaction(Treatment, Time)
group1
mm <- model.matrix(~0 + group1)

colnames(mm) <- c("D24", "F24", "V24", "D24r", "F24r", "V24r", "D144r", "F144r", "V144r")


#various filtering cutoffs here
a <- voom(de_list_calc_fil, mm, plot = TRUE)

b <- voom(de_list_calc_fil_1, mm, plot = TRUE)

c <- voom(de_list_calc_fil_2, mm, plot = TRUE)

d <- voom(de_list_calc_fil_3, mm, plot = TRUE)

e <- voom(de_list_calc_fil_4, mm, plot = TRUE)


```
The graph doesn't look quite right until I filtered by rowMeans > 3


```{r Fitting Linear Models in Limma}

fit <- lmFit(d, mm)
head(coef(fit))

cm <- makeContrasts(
        V.D24 = D24 - V24,
        V.F24 = F24 - V24,
        V.D24r = D24r - V24r,
        V.F24r = F24r - V24r,
        V.D144r = D144r - V144r,
        V.F144r = F144r - V144r,
        levels = mm
)

vfit <- lmFit(d, mm)
vfit <- contrasts.fit(vfit, contrasts = cm)
efit2 <- eBayes(vfit)

#saveRDS(efit2, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/efit2_final.RDS")

sum <- summary(decideTests(efit2))
sum


```

Now I can make a top table of all of the genes I have, sorted by F

logFC: log2 fold change of tx/veh
AveExpr: Average expression across all samples, in log2 CPM
t: logFC divided by its standard error
P.Value: Raw p-value (based on t) from test that logFC differs from 0
adj.P.Val: Benjamini-Hochberg false discovery rate adjusted p-value
B: log-odds that gene is DE (arguably less useful than the other columns)
```{r Top Table}

top.table <- limma::topTable(efit2, n = Inf, sort.by = "F", adjust.method = "BH")
#in this code you can also add adjust.method = "BH" for benjamini-hotchberg testing corrections)
head(top.table)

#write.table(top.table, file = "top.table_5FU_EMP_250217.txt", row.names = F, sep = "\t", quote = F)

#check which genes are DE in this set according to pvalue significance
# [1] 15294

#now that I've made a large general top table, make one for each comparison

####top.table_V.D24####
top.table_V.D24 <- topTable(fit = efit4, coef = "V.D24", number = nrow(de_list_calc_fil_3), adjust.method = "BH", p.value = 1, sort.by = "none")
#head(top.table_V.D24)

####top.table_V.F24####
top.table_V.F24 <- topTable(fit = efit4, coef = "F.D24", number = nrow(de_list_calc_fil_3), adjust.method = "BH", p.value = 1, sort.by = "none")

####top.table_V.D24r####
top.table_V.D24 <- topTable(fit = efit4, coef = "V.D24r", number = nrow(de_list_calc_fil_3), adjust.method = "BH", p.value = 1, sort.by = "none")
#head(top.table_V.D24)

####top.table_V.F24r####
top.table_V.F24 <- topTable(fit = efit4, coef = "F.D24r", number = nrow(de_list_calc_fil_3), adjust.method = "BH", p.value = 1, sort.by = "none")

####top.table_V.D144r####
top.table_V.D24 <- topTable(fit = efit4, coef = "V.D144r", number = nrow(de_list_calc_fil_3), adjust.method = "BH", p.value = 1, sort.by = "none")
#head(top.table_V.D24)

####top.table_V.F144r####
top.table_V.F24 <- topTable(fit = efit4, coef = "F.D144r", number = nrow(de_list_calc_fil_3), adjust.method = "BH", p.value = 1, sort.by = "none")
```

Now that I've done pairwise comparisons of my treatments vs vehicles at each timepoint - let's make some volcano plots of my data
```{r Volcano Plots from Pairwise Gene Analysis}

#define a function to generate volcano plots
make_vol_plot <- function(top.table, title) {
        #add significance labels
        top.table$Significance <- "Not Significant"
        top.table$Significance[top.table$F]
}


```

I want to go ahead and try refitting my model to include effects for time, tx, and time-tx interactions
```{r Refit Model}

colnames(mm)



```


Now I'm going to try
```{r}

```

