<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Emma M Pfortmiller" />

<meta name="date" content="2025-02-28" />

<title>Recovery_DRC</title>

<script src="site_libs/header-attrs-2.29/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.5.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Recovery_RNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
<li>
  <a href="Recovery_DE.html">Differential Expression Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/emmapfort/Recovery_5FU_Project">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Recovery_DRC</h1>
<h4 class="author">Emma M Pfortmiller</h4>
<h4 class="date">2025-02-28</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2025-03-03
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>Recovery_5FU/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20250217code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20250217)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20250217code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20250217)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomemmapfortRecovery5FUProjecttreef49b130d552e1e2832ed81d9ee5ba386985fcb4ftargetblankf49b130a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/emmapfort/Recovery_5FU_Project/tree/f49b130d552e1e2832ed81d9ee5ba386985fcb4f" target="_blank">f49b130</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomemmapfortRecovery5FUProjecttreef49b130d552e1e2832ed81d9ee5ba386985fcb4ftargetblankf49b130a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/emmapfort/Recovery_5FU_Project/tree/f49b130d552e1e2832ed81d9ee5ba386985fcb4f" target="_blank">f49b130</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Unstaged changes:
    Modified:   analysis/Recovery_DE.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Recovery_DRC.Rmd</code>) and HTML
(<code>docs/Recovery_DRC.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emmapfort/Recovery_5FU_Project/blob/f49b130d552e1e2832ed81d9ee5ba386985fcb4f/analysis/Recovery_DRC.Rmd" target="_blank">f49b130</a>
</td>
<td>
emmapfort
</td>
<td>
2025-03-03
</td>
<td>
Add in Dose Response Experiment
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/emmapfort/Recovery_5FU_Project/blob/94abbf38339d302923b2e3310d6c082686d5ee91/analysis/Recovery_DRC.Rmd" target="_blank">94abbf3</a>
</td>
<td>
emmapfort
</td>
<td>
2025-02-28
</td>
<td>
RUVs updates
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<p>This contains data from all three individuals used for this
timecourse experiment 75-1, 78-1, 17-3 iPSCs 50,000 cells in 60 wells of
6x 96 well plates Treated with varying concentrations of doxrubicin,
5-fluorouracil, and DMSO vehicle matched controls</p>
<pre><code>── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.4     ✔ readr     2.1.5
✔ forcats   1.0.0     ✔ stringr   1.5.1
✔ ggplot2   3.5.1     ✔ tibble    3.2.1
✔ lubridate 1.9.3     ✔ tidyr     1.3.1
✔ purrr     1.0.2     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (&lt;http://conflicted.r-lib.org/&gt;) to force all conflicts to become errors

Attaching package: &#39;scales&#39;


The following object is masked from &#39;package:purrr&#39;:

    discard


The following object is masked from &#39;package:readr&#39;:

    col_factor


Loading required package: MASS


Attaching package: &#39;MASS&#39;


The following object is masked from &#39;package:dplyr&#39;:

    select



&#39;drc&#39; has been loaded.


Please cite R and &#39;drc&#39; if used for a publication,

for references type &#39;citation()&#39; and &#39;citation(&#39;drc&#39;)&#39;.



Attaching package: &#39;drc&#39;


The following objects are masked from &#39;package:stats&#39;:

    gaussian, getInitial</code></pre>
<p>Import Datasets</p>
<pre class="r"><code>AllData &lt;- read_excel(&quot;C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/DRCs/5FUProj_DRCTime_78-1_75-1_17-3_EMP_240226.xlsx&quot;,
     sheet = &quot;Fin_Sheet_Analysis&quot;)
View(AllData)

Ind1_75_data &lt;- read_excel(&quot;C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/DRCs/5FUProj_DRCTime_78-1_75-1_17-3_EMP_240226.xlsx&quot;,
     sheet = &quot;Fin_75_Sheet&quot;)
View(Ind1_75_data)

Ind2_78_data &lt;- read_excel(&quot;C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/DRCs/5FUProj_DRCTime_78-1_75-1_17-3_EMP_240226.xlsx&quot;,
     sheet = &quot;Fin_78_Sheet&quot;)
View(Ind2_78_data)

Ind3_17_data &lt;- read_excel(&quot;C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/DRCs/5FUProj_DRCTime_78-1_75-1_17-3_EMP_240226.xlsx&quot;,
     sheet = &quot;Fin_17_Sheet&quot;)
View(Ind3_17_data)</code></pre>
<p>Now that I’ve imported my data, let’s go ahead and filter the data to
reflect time and treatment groups.</p>
<p>I have three timepoints that I’m investigating through this
experiment, they will be treated separately Let’s filter the data
accordingly by timepoint (24hr, 48hr, 72hr)</p>
<p>Now that these are filtered by individual, time, and drug - I will
start plotting the data</p>
<p>First, I’ll plot the 24hr timepoint raw data</p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DOX - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr78$Ind2_78_In ~ dox24hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 Raw Data&quot;
)

boxplot(dox24hr78$Ind2_78_Fin ~ dox24hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 78-1 Raw Data&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2024hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DOX - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr75$Ind1_75_In ~ dox24hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 Raw Data&quot;
)

boxplot(dox24hr75$Ind1_75_Fin ~ dox24hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 75-1 Raw Data&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2024hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DOX - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr17$Ind3_17_In ~ dox24hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 Raw Data&quot;
)

boxplot(dox24hr17$Ind3_17_Fin ~ dox24hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 17-3 Raw Data&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2024hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration FLUO - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr78$Ind2_78_In ~ fluo24hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1&quot;
)

boxplot(fluo24hr78$Ind2_78_Fin ~ fluo24hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2024hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration FLUO - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr75$Ind1_75_In ~ fluo24hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1&quot;
)

boxplot(fluo24hr75$Ind1_75_Fin ~ fluo24hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 75-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2024hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration FLUO - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr17$Ind3_17_In ~ fluo24hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3&quot;
)

boxplot(fluo24hr17$Ind3_17_Fin ~ fluo24hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 17-3&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2024hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DMSO - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr78$Ind2_78_In ~ dmso24hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1&quot;
)

boxplot(dmso24hr78$Ind2_78_Fin ~ dmso24hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2024hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DMSO - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr75$Ind1_75_In ~ dmso24hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1&quot;
)

boxplot(dmso24hr75$Ind1_75_Fin ~ dmso24hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 75-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2024hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DMSO - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr17$Ind3_17_In ~ dmso24hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3&quot;
)

boxplot(dmso24hr17$Ind3_17_Fin ~ dmso24hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 17-3&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2024hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now let’s look at the raw data for the 48hr timepoint</p>
<pre class="r"><code>#####DOX Samples#####
#DOX 78-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DOX - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr78$Ind2_78_In ~ dox48hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1&quot;
)

boxplot(dox48hr78$Ind2_78_Fin ~ dox48hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2048hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DOXO 75-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DOX - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr75$Ind1_75_In ~ dox48hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1&quot;
)

boxplot(dox48hr75$Ind1_75_Fin ~ dox48hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 75-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2048hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DOX 17-3 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DOX - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr17$Ind3_17_In ~ dox48hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3&quot;
)

boxplot(dox48hr17$Ind3_17_Fin ~ dox48hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 17-3&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2048hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration FLUO - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr78$Ind2_78_In ~ fluo48hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1&quot;
)

boxplot(fluo48hr78$Ind2_78_Fin ~ fluo48hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2048hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration FLUO - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr75$Ind1_75_In ~ fluo48hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1&quot;
)

boxplot(fluo48hr75$Ind1_75_Fin ~ fluo48hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 75-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2048hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration FLUO - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr17$Ind3_17_In ~ fluo48hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3&quot;
)

boxplot(fluo48hr17$Ind3_17_Fin ~ fluo48hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 17-3&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2048hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DMSO - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr78$Ind2_78_In ~ dmso48hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1&quot;
)

boxplot(dmso48hr78$Ind2_78_Fin ~ dmso48hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2048hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DMSO - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr75$Ind1_75_In ~ dmso48hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1&quot;
)

boxplot(dmso48hr75$Ind1_75_Fin ~ dmso48hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 75-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2048hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DMSO - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr17$Ind3_17_In ~ dmso48hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3&quot;
)

boxplot(dmso48hr17$Ind3_17_Fin ~ dmso48hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 17-3&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2048hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DOX Samples#####
#DOX 78-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DOX - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr78$Ind2_78_In ~ dox72hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1&quot;
)

boxplot(dox72hr78$Ind2_78_Fin ~ dox72hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2072hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DOX 75-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DOX - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr75$Ind1_75_In ~ dox72hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1&quot;
)

boxplot(dox72hr75$Ind1_75_Fin ~ dox72hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 75-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2072hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DOX 17-3 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DOX - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr17$Ind3_17_In ~ dox72hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3&quot;
)

boxplot(dox72hr17$Ind3_17_Fin ~ dox72hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 17-3&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2072hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration FLUO - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr78$Ind2_78_In ~ fluo72hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1&quot;
)

boxplot(fluo72hr78$Ind2_78_Fin ~ fluo72hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2072hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration FLUO - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr75$Ind1_75_In ~ fluo72hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1&quot;
)

boxplot(fluo72hr75$Ind1_75_Fin ~ fluo72hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 75-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2072hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration FLUO - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr17$Ind3_17_In ~ fluo72hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3&quot;
)

boxplot(fluo72hr17$Ind3_17_Fin ~ fluo72hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 17-3&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2072hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DMSO - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr78$Ind2_78_In ~ dmso72hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1&quot;
)

boxplot(dmso72hr78$Ind2_78_Fin ~ dmso72hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2072hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DMSO - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr75$Ind1_75_In ~ dmso72hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1&quot;
)

boxplot(dmso72hr75$Ind1_75_Fin ~ dmso72hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 75-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2072hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RFLU by Concentration DMSO - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr17$Ind3_17_In ~ dmso72hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3&quot;
)

boxplot(dmso72hr17$Ind3_17_Fin ~ dmso72hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 17-3&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Raw%20Data%2072hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now that I’ve taken a look at the raw data, let’s look at the
analyzed data step by step The following steps were taken in analysis:
1. Import raw fluorescence values (RFLU) from platereader, baseline 0hr
and final timepoints 2. Subtract background wells (prestoblue only
wells) by plate 3. Subtract average of empty wells (galactose only, with
50k cells/well) by plate 4. Divide all values by matched vehicle average
(ie DOX 0.01 / DMSO 0.01) 5. Divide initial 0hr value from final
timepoint (24, 48, 72) value</p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 no BG 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr78$Ind2_78_In_noBG ~ dox24hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 no BG&quot;
)

boxplot(dox24hr78$Ind2_78_Fin_noBG ~ dox24hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20removed%2024hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr75$Ind1_75_In_noBG ~ dox24hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 no BG&quot;
)

boxplot(dox24hr75$Ind1_75_Fin_noBG ~ dox24hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 75-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20removed%2024hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr17$Ind3_17_In_noBG ~ dox24hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 no BG&quot;
)

boxplot(dox24hr17$Ind3_17_Fin_noBG ~ dox24hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 17-3 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20removed%2024hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 no BG 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr78$Ind2_78_In_noBG ~ fluo24hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1 no BG&quot;
)

boxplot(fluo24hr78$Ind2_78_Fin_noBG ~ fluo24hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20removed%2024hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 no BG 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr75$Ind1_75_In_noBG ~ fluo24hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1 no BG&quot;
)

boxplot(fluo24hr75$Ind1_75_Fin_noBG ~ fluo24hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 75-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20removed%2024hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 no BG 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr17$Ind3_17_In_noBG ~ fluo24hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3 no BG&quot;
)

boxplot(fluo24hr17$Ind3_17_Fin_noBG ~ fluo24hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 17-3 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20removed%2024hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr78$Ind2_78_In_noBG ~ dmso24hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1 no BG&quot;
)

boxplot(dmso24hr78$Ind2_78_Fin_noBG ~ dmso24hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20removed%2024hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 no BG 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr75$Ind1_75_In_noBG ~ dmso24hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1 no BG&quot;
)

boxplot(dmso24hr75$Ind1_75_Fin_noBG ~ dmso24hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 75-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20removed%2024hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 no BG 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr17$Ind3_17_In_noBG ~ dmso24hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3 no BG&quot;
)

boxplot(dmso24hr17$Ind3_17_Fin_noBG ~ dmso24hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 17-3 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20removed%2024hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 no BG 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr78$Ind2_78_In_noBG ~ dox48hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 no BG&quot;
)

boxplot(dox48hr78$Ind2_78_Fin_noBG ~ dox48hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2048hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr75$Ind1_75_In_noBG ~ dox48hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 no BG&quot;
)

boxplot(dox48hr75$Ind1_75_Fin_noBG ~ dox48hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 75-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2048hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr17$Ind3_17_In_noBG ~ dox48hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 no BG&quot;
)

boxplot(dox48hr17$Ind3_17_Fin_noBG ~ dox48hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 17-3 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2048hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 no BG 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr78$Ind2_78_In_noBG ~ fluo48hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1 no BG&quot;
)

boxplot(fluo48hr78$Ind2_78_Fin_noBG ~ fluo48hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2048hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 no BG 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr75$Ind1_75_In_noBG ~ fluo48hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1 no BG&quot;
)

boxplot(fluo48hr75$Ind1_75_Fin_noBG ~ fluo48hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 75-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2048hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 no BG 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr17$Ind3_17_In_noBG ~ fluo48hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3 no BG&quot;
)

boxplot(fluo48hr17$Ind3_17_Fin_noBG ~ fluo48hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 17-3 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2048hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr78$Ind2_78_In_noBG ~ dmso48hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1 no BG&quot;
)

boxplot(dmso48hr78$Ind2_78_Fin_noBG ~ dmso48hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2048hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 no BG 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr75$Ind1_75_In_noBG ~ dmso48hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1 no BG&quot;
)

boxplot(dmso48hr75$Ind1_75_Fin_noBG ~ dmso48hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 75-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2048hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 no BG 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr17$Ind3_17_In_noBG ~ dmso48hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3 no BG&quot;
)

boxplot(dmso48hr17$Ind3_17_Fin_noBG ~ dmso48hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 17-3 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2048hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 no BG 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr78$Ind2_78_In_noBG ~ dox72hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 no BG&quot;
)

boxplot(dox72hr78$Ind2_78_Fin_noBG ~ dox72hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2072hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr75$Ind1_75_In_noBG ~ dox72hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 no BG&quot;
)

boxplot(dox72hr75$Ind1_75_Fin_noBG ~ dox72hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 75-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2072hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr17$Ind3_17_In_noBG ~ dox72hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 no BG&quot;
)

boxplot(dox72hr17$Ind3_17_Fin_noBG ~ dox72hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 17-3 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2072hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 no BG 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr78$Ind2_78_In_noBG ~ fluo72hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1 no BG&quot;
)

boxplot(fluo72hr78$Ind2_78_Fin_noBG ~ fluo72hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2072hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 no BG 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr75$Ind1_75_In_noBG ~ fluo72hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1 no BG&quot;
)

boxplot(fluo72hr75$Ind1_75_Fin_noBG ~ fluo72hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 75-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2072hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 no BG 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr17$Ind3_17_In_noBG ~ fluo72hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3 no BG&quot;
)

boxplot(fluo72hr17$Ind3_17_Fin_noBG ~ fluo72hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 17-3 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2072hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr78$Ind2_78_In_noBG ~ dmso72hr78$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1 no BG&quot;
)

boxplot(dmso72hr78$Ind2_78_Fin_noBG ~ dmso72hr78$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 78-1&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2072hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 no BG 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr75$Ind1_75_In_noBG ~ dmso72hr75$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1 no BG&quot;
)

boxplot(dmso72hr75$Ind1_75_Fin_noBG ~ dmso72hr75$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 75-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2072hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 no BG 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr17$Ind3_17_In_noBG ~ dmso72hr17$Conc, 
        ylim = c(0, 80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3 no BG&quot;
)

boxplot(dmso72hr17$Ind3_17_Fin_noBG ~ dmso72hr17$Conc, 
        ylim = c(0,80000), 
        ylab = &quot;Cell Viability (RFLU)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 17-3 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20BG%20Removed%2072hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now let’s plot the next step: subtracting the empty wells</p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 Empty Wells 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr78$Ind2_78_In_noBG_Low ~ dox24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 no BG Empty&quot;
)

boxplot(dox24hr78$Ind2_78_Fin_noBG_Low ~ dox24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2024hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr75$Ind1_75_In_noBG_Low ~ dox24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 no BG Empty&quot;
)

boxplot(dox24hr75$Ind1_75_Fin_noBG_Low ~ dox24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 75-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2024hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr17$Ind3_17_In_noBG_Low ~ dox24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 no BG Empty&quot;
)

boxplot(dox24hr17$Ind3_17_Fin_noBG_Low ~ dox24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 17-3 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2024hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 no BG 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr78$Ind2_78_In_noBG_Low ~ fluo24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1 no BG Empty&quot;
)

boxplot(fluo24hr78$Ind2_78_Fin_noBG_Low ~ fluo24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 78-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2024hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 no BG Empty 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr75$Ind1_75_In_noBG_Low ~ fluo24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1 no BG Empty&quot;
)

boxplot(fluo24hr75$Ind1_75_Fin_noBG_Low ~ fluo24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 75-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2024hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 no BG Empty 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr17$Ind3_17_In_noBG_Low ~ fluo24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3 no BG Empty&quot;
)

boxplot(fluo24hr17$Ind3_17_Fin_noBG_Low ~ fluo24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 17-3 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2024hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr78$Ind2_78_In_noBG_Low ~ dmso24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1 no BG Empty&quot;
)

boxplot(dmso24hr78$Ind2_78_Fin_noBG_Low ~ dmso24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 78-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2024hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 no BG Empty 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr75$Ind1_75_In_noBG_Low ~ dmso24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1 no BG Empty&quot;
)

boxplot(dmso24hr75$Ind1_75_Fin_noBG_Low ~ dmso24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 75-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2024hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 no BG Empty 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr17$Ind3_17_In_noBG_Low ~ dmso24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3 no BG Empty&quot;
)

boxplot(dmso24hr17$Ind3_17_Fin_noBG_Low ~ dmso24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 17-3 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2024hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 Empty Wells 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr78$Ind2_78_In_noBG_Low ~ dox48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 no BG Empty&quot;
)

boxplot(dox48hr78$Ind2_78_Fin_noBG_Low ~ dox48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2048hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr75$Ind1_75_In_noBG_Low ~ dox48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 no BG Empty&quot;
)

boxplot(dox48hr75$Ind1_75_Fin_noBG_Low ~ dox48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 75-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2048hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr17$Ind3_17_In_noBG_Low ~ dox48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 no BG Empty&quot;
)

boxplot(dox48hr17$Ind3_17_Fin_noBG_Low ~ dox48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 17-3 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2048hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 no BG 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr78$Ind2_78_In_noBG_Low ~ fluo48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1 no BG Empty&quot;
)

boxplot(fluo48hr78$Ind2_78_Fin_noBG_Low ~ fluo48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 78-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2048hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 no BG Empty 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr75$Ind1_75_In_noBG_Low ~ fluo48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1 no BG Empty&quot;
)

boxplot(fluo48hr75$Ind1_75_Fin_noBG_Low ~ fluo48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 75-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2048hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 no BG Empty 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr17$Ind3_17_In_noBG_Low ~ fluo48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3 no BG Empty&quot;
)

boxplot(fluo48hr17$Ind3_17_Fin_noBG_Low ~ fluo48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 17-3 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2048hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr78$Ind2_78_In_noBG_Low ~ dmso48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1 no BG Empty&quot;
)

boxplot(dmso48hr78$Ind2_78_Fin_noBG_Low ~ dmso48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 78-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2048hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 no BG Empty 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr75$Ind1_75_In_noBG_Low ~ dmso48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1 no BG Empty&quot;
)

boxplot(dmso48hr75$Ind1_75_Fin_noBG_Low ~ dmso48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 75-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2048hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 no BG Empty 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr17$Ind3_17_In_noBG_Low ~ dmso48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3 no BG Empty&quot;
)

boxplot(dmso48hr17$Ind3_17_Fin_noBG_Low ~ dmso48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 17-3 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2048hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 Empty Wells 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr78$Ind2_78_In_noBG_Low ~ dox72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 no BG Empty&quot;
)

boxplot(dox72hr78$Ind2_78_Fin_noBG_Low ~ dox72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 78-1 no BG&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2072hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr75$Ind1_75_In_noBG_Low ~ dox72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 no BG Empty&quot;
)

boxplot(dox72hr75$Ind1_75_Fin_noBG_Low ~ dox72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 75-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2072hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr17$Ind3_17_In_noBG_Low ~ dox72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 no BG Empty&quot;
)

boxplot(dox72hr17$Ind3_17_Fin_noBG_Low ~ dox72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 17-3 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2072hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 no BG 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr78$Ind2_78_In_noBG_Low ~ fluo72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1 no BG Empty&quot;
)

boxplot(fluo72hr78$Ind2_78_Fin_noBG_Low ~ fluo72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 78-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2072hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 no BG Empty 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr75$Ind1_75_In_noBG_Low ~ fluo72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1 no BG Empty&quot;
)

boxplot(fluo72hr75$Ind1_75_Fin_noBG_Low ~ fluo72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 75-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2072hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 no BG Empty 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr17$Ind3_17_In_noBG_Low ~ fluo72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3 no BG Empty&quot;
)

boxplot(fluo72hr17$Ind3_17_Fin_noBG_Low ~ fluo72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 17-3 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2072hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr78$Ind2_78_In_noBG_Low ~ dmso72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1 no BG Empty&quot;
)

boxplot(dmso72hr78$Ind2_78_Fin_noBG_Low ~ dmso72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 78-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2072hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 no BG Empty 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr75$Ind1_75_In_noBG_Low ~ dmso72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1 no BG Empty&quot;
)

boxplot(dmso72hr75$Ind1_75_Fin_noBG_Low ~ dmso72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 75-1 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2072hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 no BG Empty 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr17$Ind3_17_In_noBG_Low ~ dmso72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3 no BG Empty&quot;
)

boxplot(dmso72hr17$Ind3_17_Fin_noBG_Low ~ dmso72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 17-3 no BG Empty&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Empty%20Wells%20Removed%2072hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now let’s plot the third step: dividing the matched vehicle
values</p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 Veh 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty Veh - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr78$Ind2_78_In_noBG_Low_Veh ~ dox24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 no BG Empty Veh&quot;
)

boxplot(dox24hr78$Ind2_78_Fin_noBG_Low_Veh ~ dox24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 78-1 no BG Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2024hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty Veh - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr75$Ind1_75_In_noBG_Low_Veh ~ dox24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 no BG Empty Veh&quot;
)

boxplot(dox24hr75$Ind1_75_Fin_noBG_Low_Veh ~ dox24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 75-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2024hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty Veh - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr17$Ind3_17_In_noBG_Low_Veh ~ dox24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 no BG Empty Veh&quot;
)

boxplot(dox24hr17$Ind3_17_Fin_noBG_Low_Veh ~ dox24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 24hr 17-3 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2024hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 no BG 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty Veh - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr78$Ind2_78_In_noBG_Low_Veh ~ fluo24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1 no BG Empty Veh&quot;
)

boxplot(fluo24hr78$Ind2_78_Fin_noBG_Low_Veh ~ fluo24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 78-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2024hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 no BG Empty 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty Veh - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr75$Ind1_75_In_noBG_Low_Veh ~ fluo24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1 no BG Empty Veh&quot;
)

boxplot(fluo24hr75$Ind1_75_Fin_noBG_Low_Veh ~ fluo24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 75-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2024hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 no BG Empty 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty Veh - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr17$Ind3_17_In_noBG_Low_Veh ~ fluo24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3 no BG Empty Veh&quot;
)

boxplot(fluo24hr17$Ind3_17_Fin_noBG_Low_Veh ~ fluo24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 24hr 17-3 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2024hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty Veh - 78-1 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr78$Ind2_78_In_noBG_Low_Veh ~ dmso24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1 no BG Empty Veh&quot;
)

boxplot(dmso24hr78$Ind2_78_Fin_noBG_Low_Veh ~ dmso24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 78-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2024hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 no BG Empty 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty Veh - 75-1 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr75$Ind1_75_In_noBG_Low_Veh ~ dmso24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1 no BG Empty Veh&quot;
)

boxplot(dmso24hr75$Ind1_75_Fin_noBG_Low_Veh ~ dmso24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 75-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2024hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 no BG Empty 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty Veh - 17-3 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr17$Ind3_17_In_noBG_Low_Veh ~ dmso24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3 no BG Empty Veh&quot;
)

boxplot(dmso24hr17$Ind3_17_Fin_noBG_Low_Veh ~ dmso24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 24hr 17-3 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2024hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 Veh 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty Veh - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr78$Ind2_78_In_noBG_Low_Veh ~ dox48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 no BG Empty Veh&quot;
)

boxplot(dox48hr78$Ind2_78_Fin_noBG_Low_Veh ~ dox48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 78-1 no BG Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2048hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty Veh - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr75$Ind1_75_In_noBG_Low_Veh ~ dox48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 no BG Empty Veh&quot;
)

boxplot(dox48hr75$Ind1_75_Fin_noBG_Low_Veh ~ dox48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 75-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2048hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty Veh - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr17$Ind3_17_In_noBG_Low_Veh ~ dox48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 no BG Empty Veh&quot;
)

boxplot(dox48hr17$Ind3_17_Fin_noBG_Low_Veh ~ dox48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 48hr 17-3 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2048hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 no BG 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty Veh - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr78$Ind2_78_In_noBG_Low_Veh ~ fluo48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1 no BG Empty Veh&quot;
)

boxplot(fluo48hr78$Ind2_78_Fin_noBG_Low_Veh ~ fluo48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 78-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2048hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 no BG Empty 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty Veh - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr75$Ind1_75_In_noBG_Low_Veh ~ fluo48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1 no BG Empty Veh&quot;
)

boxplot(fluo48hr75$Ind1_75_Fin_noBG_Low_Veh ~ fluo48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 75-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2048hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 no BG Empty 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty Veh - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr17$Ind3_17_In_noBG_Low_Veh ~ fluo48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3 no BG Empty Veh&quot;
)

boxplot(fluo48hr17$Ind3_17_Fin_noBG_Low_Veh ~ fluo48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 48hr 17-3 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2048hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty Veh - 78-1 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr78$Ind2_78_In_noBG_Low_Veh ~ dmso48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1 no BG Empty Veh&quot;
)

boxplot(dmso48hr78$Ind2_78_Fin_noBG_Low_Veh ~ dmso48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 78-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2048hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 no BG Empty 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty Veh - 75-1 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr75$Ind1_75_In_noBG_Low_Veh ~ dmso48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1 no BG Empty Veh&quot;
)

boxplot(dmso48hr75$Ind1_75_Fin_noBG_Low_Veh ~ dmso48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 75-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2048hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 no BG Empty 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty Veh - 17-3 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr17$Ind3_17_In_noBG_Low_Veh ~ dmso48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3 no BG Empty Veh&quot;
)

boxplot(dmso48hr17$Ind3_17_Fin_noBG_Low_Veh ~ dmso48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 48hr 17-3 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2048hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DOX Samples#####
#dox 78-1 Veh 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty Veh - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr78$Ind2_78_In_noBG_Low_Veh ~ dox72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 78-1 no BG Empty Veh&quot;
)

boxplot(dox72hr78$Ind2_78_Fin_noBG_Low_Veh ~ dox72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 78-1 no BG Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2072hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 75-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty Veh - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr75$Ind1_75_In_noBG_Low_Veh ~ dox72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 75-1 no BG Empty Veh&quot;
)

boxplot(dox72hr75$Ind1_75_Fin_noBG_Low_Veh ~ dox72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 75-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2072hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#dox 17-3 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOX no BG Empty Veh - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr17$Ind3_17_In_noBG_Low_Veh ~ dox72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 0hr 17-3 no BG Empty Veh&quot;
)

boxplot(dox72hr17$Ind3_17_Fin_noBG_Low_Veh ~ dox72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DOXORUBICIN 72hr 17-3 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2072hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
#FLUO 78-1 no BG 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty Veh - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr78$Ind2_78_In_noBG_Low_Veh ~ fluo72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 78-1 no BG Empty Veh&quot;
)

boxplot(fluo72hr78$Ind2_78_Fin_noBG_Low_Veh ~ fluo72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 78-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2072hr-4.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 75-1 no BG Empty 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty Veh - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr75$Ind1_75_In_noBG_Low_Veh ~ fluo72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 75-1 no BG Empty Veh&quot;
)

boxplot(fluo72hr75$Ind1_75_Fin_noBG_Low_Veh ~ fluo72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 75-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2072hr-5.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#FLUO 17-3 no BG Empty 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;FLUO no BG Empty Veh - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr17$Ind3_17_In_noBG_Low_Veh ~ fluo72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 0hr 17-3 no BG Empty Veh&quot;
)

boxplot(fluo72hr17$Ind3_17_Fin_noBG_Low_Veh ~ fluo72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;5-FLUOROURACIL 72hr 17-3 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2072hr-6.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO 78-1 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty Veh - 78-1 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr78$Ind2_78_In_noBG_Low_Veh ~ dmso72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 78-1 no BG Empty Veh&quot;
)

boxplot(dmso72hr78$Ind2_78_Fin_noBG_Low_Veh ~ dmso72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 78-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2072hr-7.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 75-1 no BG Empty 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty Veh - 75-1 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr75$Ind1_75_In_noBG_Low_Veh ~ dmso72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 75-1 no BG Empty Veh&quot;
)

boxplot(dmso72hr75$Ind1_75_Fin_noBG_Low_Veh ~ dmso72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 75-1 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2072hr-8.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#DMSO 17-3 no BG Empty 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO no BG Empty Veh - 17-3 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr17$Ind3_17_In_noBG_Low_Veh ~ dmso72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 0hr 17-3 no BG Empty Veh&quot;
)

boxplot(dmso72hr17$Ind3_17_Fin_noBG_Low_Veh ~ dmso72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;DMSO VEHICLE 72hr 17-3 no BG Empty Veh&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Matched%20Vehicle%20Divided%2072hr-9.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now for the final step, dividing the final timepoint by the initial
baseline reading at 0hr</p>
<pre class="r"><code>#####DOX Samples#####
#dox All 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOXORUBICIN - 24hr&quot;,cex = 2,font = 2)

boxplot(dox24hr78$Ind2_78_Div ~ dox24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;78-1 Div&quot;
)

boxplot(dox24hr75$Ind1_75_Div ~ dox24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;75-1 Div&quot;
)

boxplot(dox24hr17$Ind3_17_Div ~ dox24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;17-3 Div&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Final%20Data%2024hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;5-FLUOROURACIL - 24hr&quot;,cex = 2,font = 2)

boxplot(fluo24hr78$Ind2_78_Div ~ fluo24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;78-1 Div&quot;
)

boxplot(fluo24hr75$Ind1_75_Div ~ fluo24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;75-1 Div&quot;
)

boxplot(fluo24hr17$Ind3_17_Div ~ fluo24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;17-3 Div&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Final%20Data%2024hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO All 24HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO VEHICLE - 24hr&quot;,cex = 2,font = 2)

boxplot(dmso24hr78$Ind2_78_Div ~ dmso24hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;78-1 Div&quot;
)

boxplot(dmso24hr75$Ind1_75_Div ~ dmso24hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;75-1 Div&quot;
)

boxplot(dmso24hr17$Ind3_17_Div ~ dmso24hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;17-3 Div&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Final%20Data%2024hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DOX Samples#####
#dox All 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOXORUBICIN - 48hr&quot;,cex = 2,font = 2)

boxplot(dox48hr78$Ind2_78_Div ~ dox48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;78-1 Div&quot;
)

boxplot(dox48hr75$Ind1_75_Div ~ dox48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;75-1 Div&quot;
)

boxplot(dox48hr17$Ind3_17_Div ~ dox48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;17-3 Div&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Final%20Data%2048hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;5-FLUOROURACIL - 48hr&quot;,cex = 2,font = 2)

boxplot(fluo48hr78$Ind2_78_Div ~ fluo48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;78-1 Div&quot;
)

boxplot(fluo48hr75$Ind1_75_Div ~ fluo48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;75-1 Div&quot;
)

boxplot(fluo48hr17$Ind3_17_Div ~ fluo48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;17-3 Div&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Final%20Data%2048hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO All 48HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO VEHICLE - 48hr&quot;,cex = 2,font = 2)

boxplot(dmso48hr78$Ind2_78_Div ~ dmso48hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;78-1 Div&quot;
)

boxplot(dmso48hr75$Ind1_75_Div ~ dmso48hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;75-1 Div&quot;
)

boxplot(dmso48hr17$Ind3_17_Div ~ dmso48hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;17-3 Div&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Final%20Data%2048hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DOX Samples#####
#dox All 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOXORUBICIN - 72hr&quot;,cex = 2,font = 2)

boxplot(dox72hr78$Ind2_78_Div ~ dox72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;78-1 Div&quot;
)

boxplot(dox72hr75$Ind1_75_Div ~ dox72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;75-1 Div&quot;
)

boxplot(dox72hr17$Ind3_17_Div ~ dox72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;17-3 Div&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Final%20Data%2072hr-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####FLUO samples#####
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;5-FLUOROURACIL - 72hr&quot;,cex = 2,font = 2)

boxplot(fluo72hr78$Ind2_78_Div ~ fluo72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;78-1 Div&quot;
)

boxplot(fluo72hr75$Ind1_75_Div ~ fluo72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;75-1 Div&quot;
)

boxplot(fluo72hr17$Ind3_17_Div ~ fluo72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;17-3 Div&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Final%20Data%2072hr-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#####DMSO samples#####
#DMSO All 72HR
par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.5, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DMSO VEHICLE - 72hr&quot;,cex = 2,font = 2)

boxplot(dmso72hr78$Ind2_78_Div ~ dmso72hr78$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;78-1 Div&quot;
)

boxplot(dmso72hr75$Ind1_75_Div ~ dmso72hr75$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;75-1 Div&quot;
)

boxplot(dmso72hr17$Ind3_17_Div ~ dmso72hr17$Conc, 
        ylim = c(0, 1.5), 
        ylab = &quot;Cell Viability (%)&quot;, 
        xlab = &quot;Concentration (uM)&quot;, 
        main = &quot;17-3 Div&quot;
)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/Boxplots%20Final%20Data%2072hr-3.png" width="672" style="display: block; margin: auto;" /></p>
<p>Now that I’ve plotted out all of the analysis steps and nothing looks
amiss, let’s start fitting my models for DRC I used the DRC package in R
At this point, I am only showing the relevant data for both treatments,
as the vehicle has been divided out</p>
<pre class="r"><code>#####DOX Samples#####
#24hr DOX
dox24tox_75 &lt;- dox24hr75 %&gt;% mutate(response = Ind1_75_Div*100)
summary(dox24tox_75$response)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
  0.2582   9.4402  70.8116  60.2731  88.1141 143.8750 </code></pre>
<pre class="r"><code>dox24tox_78 &lt;- dox24hr78 %&gt;% mutate(response = Ind2_78_Div*100)
summary(dox24tox_78$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  2.304  14.760  68.363  61.596  94.656 136.295 </code></pre>
<pre class="r"><code>dox24tox_17 &lt;- dox24hr17 %&gt;% mutate(response = Ind3_17_Div*100)
summary(dox24tox_17$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
-12.545   3.773  68.230  51.372  79.390 127.911 </code></pre>
<pre class="r"><code>#48hr DOX
dox48tox_75 &lt;- dox48hr75 %&gt;% mutate(response = Ind1_75_Div*100)
summary(dox48tox_75$response)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
 -0.5438   7.0099  32.8295  34.9376  46.9964 108.9591 </code></pre>
<pre class="r"><code>dox48tox_78 &lt;- dox48hr78 %&gt;% mutate(response = Ind2_78_Div*100)
summary(dox48tox_78$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.356   4.381  56.012  56.614  94.550 144.755 </code></pre>
<pre class="r"><code>dox48tox_17 &lt;- dox48hr17 %&gt;% mutate(response = Ind3_17_Div*100)
summary(dox48tox_17$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
-11.355   3.258  39.529  38.204  54.584 108.172 </code></pre>
<pre class="r"><code>#72hr DOX
dox72tox_75 &lt;- dox72hr75 %&gt;% mutate(response = Ind1_75_Div*100)
summary(dox72tox_75$response)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
 -0.4736   2.2726  15.0670  28.2831  26.5193 122.8151 </code></pre>
<pre class="r"><code>dox72tox_78 &lt;- dox72hr78 %&gt;% mutate(response = Ind2_78_Div*100)
summary(dox72tox_78$response)</code></pre>
<pre><code>    Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
 -0.5377   2.6397  28.8202  43.8088  87.1758 130.4752 </code></pre>
<pre class="r"><code>dox72tox_17 &lt;- dox72hr17 %&gt;% mutate(response = Ind3_17_Div*100)
summary(dox72tox_17$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
-11.842   6.646  18.265  30.352  28.511 139.865 </code></pre>
<pre class="r"><code>#####FLUO Samples#####
#24hr FLUO
fluo24tox_75 &lt;- fluo24hr75 %&gt;% mutate(response = Ind1_75_Div*100)
summary(fluo24tox_75$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  62.91   89.33   97.51   96.54  104.49  126.50 </code></pre>
<pre class="r"><code>fluo24tox_78 &lt;- fluo24hr78 %&gt;% mutate(response = Ind2_78_Div*100)
summary(fluo24tox_78$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  56.17   88.26   98.50  102.94  118.24  174.12 </code></pre>
<pre class="r"><code>fluo24tox_17 &lt;- fluo24hr17 %&gt;% mutate(response = Ind3_17_Div*100)
summary(fluo24tox_17$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  51.60   81.22   98.63   97.69  109.90  143.90 </code></pre>
<pre class="r"><code>#48hr fluo
fluo48tox_75 &lt;- fluo48hr75 %&gt;% mutate(response = Ind1_75_Div*100)
summary(fluo48tox_75$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  66.17   86.65   96.38   96.94  106.33  121.16 </code></pre>
<pre class="r"><code>fluo48tox_78 &lt;- fluo48hr78 %&gt;% mutate(response = Ind2_78_Div*100)
summary(fluo48tox_78$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  61.42   84.96   95.34  100.91  117.15  170.63 </code></pre>
<pre class="r"><code>fluo48tox_17 &lt;- fluo48hr17 %&gt;% mutate(response = Ind3_17_Div*100)
summary(fluo48tox_17$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  54.16   77.44   99.37   93.46  108.00  125.49 </code></pre>
<pre class="r"><code>#72hr fluo
fluo72tox_75 &lt;- fluo72hr75 %&gt;% mutate(response = Ind1_75_Div*100)
summary(fluo72tox_75$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  53.78   77.13   85.58   88.18   99.53  118.57 </code></pre>
<pre class="r"><code>fluo72tox_78 &lt;- fluo72hr78 %&gt;% mutate(response = Ind2_78_Div*100)
summary(fluo72tox_78$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  67.82   83.26   99.48   96.85  107.07  130.66 </code></pre>
<pre class="r"><code>fluo72tox_17 &lt;- fluo72hr17 %&gt;% mutate(response = Ind3_17_Div*100)
summary(fluo72tox_17$response)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  19.08   60.08   79.20   78.08   92.94  148.16 </code></pre>
<p>Now that I’ve fit the models and taken a look at their summary, let’s
plot some dose-response curves through the DRC package</p>
<p>I would like to have all individuals on the same plot, for each
timepoint</p>
<pre class="r"><code>par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.3, 1, 1, 1))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;DOXORUBICIN RECOVERY CONCENTRATION-RESPONSE&quot;,cex = 2,font = 2)
#24hr plots
doxtox24_75_model &lt;- drm(response~Conc, data = dox24tox_75, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox24_75_model, type = &quot;none&quot;, broken = TRUE, 
     xt = c(0.01, 0.1, 0.5, 1, 5, 10, 50, 100, 1000), xttrim = FALSE,  
     ylim = c(0,110), cex = 1, col = &quot;#9ACCDE&quot;, lwd = 4, ylab = &quot;Percent Viability (%)&quot;,
     xlab = &quot;Concentration (uM)&quot;, main = &quot;24HR FOLLOWING TREATMENT&quot;, lty = c(1),
     xaxs = &quot;i&quot;, yaxs = &quot;i&quot;)

par(new=TRUE)
doxtox24_78_model &lt;- drm(response~Conc, data = dox24tox_78, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox24_78_model, add = TRUE, col = &quot;#499FBD&quot;, lwd = 4, lty = c(1), type = &quot;none&quot;)

par(new=TRUE)
doxtox24_17_model &lt;- drm(response~Conc, data = dox24tox_17, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox24_17_model, add = TRUE, col = &quot;#254F5E&quot;, lwd = 4, lty = c(1), type = &quot;none&quot;)

par(new=TRUE)
legend(x = &quot;bottomleft&quot;, legend = c(&quot;75-1&quot;, &quot;78-1&quot;, &quot;17-3&quot;), lty = c(1, 1, 1, 1), col = c(&quot;#9ACCDE&quot;, &quot;#499FBD&quot;, &quot;#254F5E&quot;), lwd = 4, cex = 1, title = &quot;iPSC-CM Lines&quot;)

#48hr plots
doxtox48_75_model &lt;- drm(response~Conc, data = dox48tox_75, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(doxtox48_75_model, type = &quot;none&quot;, broken = TRUE, 
     xt = c(0.01, 0.1, 0.5, 1, 5, 10, 50, 100, 1000), xttrim = FALSE,  
     ylim = c(0,110), cex = 1, col = &quot;#9ACCDE&quot;, lwd = 4, ylab = &quot;Percent Viability (%)&quot;, xlab = &quot;Concentration (uM)&quot;, main = &quot;48HR FOLLOWING TREATMENT&quot;, lty = c(2),
     xaxs = &quot;i&quot;, yaxs = &quot;i&quot;)

par(new=TRUE)
doxtox48_78_model &lt;- drm(response~Conc, data = dox48tox_78, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(doxtox48_78_model, add = TRUE, col = &quot;#499FBD&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

par(new=TRUE)
doxtox48_17_model &lt;- drm(response~Conc, data = dox48tox_17, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox48_17_model, add = TRUE, col = &quot;#254F5E&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

#72hr plots
doxtox72_75_model &lt;- drm(response~Conc, data = dox72tox_75, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(doxtox72_75_model, type = &quot;none&quot;, broken = TRUE, 
     xt = c(0.01, 0.1, 0.5, 1, 5, 10, 50, 100, 1000), xttrim = FALSE,  
     ylim = c(0, 110), cex = 1, col = &quot;#9ACCDE&quot;, lwd = 4, ylab = &quot;Percent Viability (%)&quot;,
     xlab = &quot;Concentration (uM)&quot;, main = &quot;72HR FOLLOWING TREATMENT&quot;, lty = c(3), 
     xaxs = &quot;i&quot;, yaxs = &quot;i&quot;)

par(new=TRUE)
doxtox72_78_model &lt;- drm(response~Conc, data = dox72tox_78, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox72_78_model, add = TRUE, col = &quot;#499FBD&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)

par(new=TRUE)
doxtox72_17_model &lt;- drm(response~Conc, data = dox72tox_17, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox72_17_model, add = TRUE, col = &quot;#254F5E&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/DRCs%20Doxorubicin%20All%20Timepoints-1.png" width="960" style="display: block; margin: auto;" /></p>
<pre class="r"><code>par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3, 1, 4), nrow = 2, ncol = 3), heights = c(0.3, 1, 1, 1))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;5-FLUOROURACIL RECOVERY CONCENTRATION-RESPONSE&quot;,cex = 2,font = 2)
#24hr plots
fluotox24_75_model &lt;- drm(response~Conc, data = fluo24tox_75, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox24_75_model, type = &quot;none&quot;, broken = TRUE, 
     xt = c(0.01, 0.1, 1, 5, 10, 50, 100, 1000, 10000), xttrim = FALSE,  
     ylim = c(0,110), cex = 1, col = &quot;#DCACED&quot;, lwd = 4, ylab = &quot;Percent Viability (%)&quot;,
     xlab = &quot;Concentration (uM)&quot;, main = &quot;24HR FOLLOWING TREATMENT&quot;, lty = c(1),
     xaxs = &quot;i&quot;, yaxs = &quot;i&quot;)

par(new=TRUE)
fluotox24_78_model &lt;- drm(response~Conc, data = fluo24tox_78, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(fluotox24_78_model, add = TRUE, col = &quot;#674C71&quot;, lwd = 4, lty = c(1), type = &quot;none&quot;)

par(new=TRUE)
fluotox24_17_model &lt;- drm(response~Conc, data = fluo24tox_17, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox24_17_model, add = TRUE, col = &quot;#EACDF4&quot;, lwd = 4, lty = c(1), type = &quot;none&quot;)

par(new=TRUE)
legend(x = &quot;bottomleft&quot;, legend = c(&quot;75-1&quot;, &quot;78-1&quot;, &quot;17-3&quot;), lty = c(1, 1, 1, 1), col = c(&quot;#DCACED&quot;, &quot;#674C71&quot;, &quot;#EACDF4&quot;), lwd = 4, cex = 1, title = &quot;iPSC-CM Lines&quot;)

#48hr plots
fluotox48_75_model &lt;- drm(response~Conc, data = fluo48tox_75, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox48_75_model, type = &quot;none&quot;, broken = TRUE, 
     xt = c(0.01, 0.1, 1, 5, 10, 50, 100, 1000, 10000), xttrim = FALSE,  
     ylim = c(0,110), cex = 1, col = &quot;#DCACED&quot;, lwd = 4, ylab = &quot;Percent Viability (%)&quot;, xlab = &quot;Concentration (uM)&quot;, main = &quot;48HR FOLLOWING TREATMENT&quot;, lty = c(2),
     xaxs = &quot;i&quot;, yaxs = &quot;i&quot;)

par(new=TRUE)
fluotox48_78_model &lt;- drm(response~Conc, data = fluo48tox_78, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox48_78_model, add = TRUE, col = &quot;#674C71&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

par(new=TRUE)
fluotox48_17_model &lt;- drm(response~Conc, data = fluo48tox_17, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(fluotox48_17_model, add = TRUE, col = &quot;#EACDF4&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

#72hr plots
fluotox72_75_model &lt;- drm(response~Conc, data = fluo72tox_75, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(fluotox72_75_model, type = &quot;none&quot;, broken = TRUE, 
     xt = c(0.01, 0.1, 1, 5, 10, 50, 100, 1000, 10000), xttrim = FALSE,  
     ylim = c(0, 110), cex = 1, col = &quot;#DCACED&quot;, lwd = 4, ylab = &quot;Percent Viability (%)&quot;,
     xlab = &quot;Concentration (uM)&quot;, main = &quot;72HR FOLLOWING TREATMENT&quot;, lty = c(3), 
     xaxs = &quot;i&quot;, yaxs = &quot;i&quot;)

par(new=TRUE)
fluotox72_78_model &lt;- drm(response~Conc, data = fluo72tox_78, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(fluotox72_78_model, add = TRUE, col = &quot;#674C71&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)

par(new=TRUE)
fluotox72_17_model &lt;- drm(response~Conc, data = fluo72tox_17, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox72_17_model, add = TRUE, col = &quot;#EACDF4&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/DRCs%205-Fluorouracil%20All%20Timepoints-1.png" width="960" style="display: block; margin: auto;" /></p>
<p>Test to see if I can plot all of these on the same plot using colors
and lines</p>
<pre class="r"><code>#for this to work I need to find unique lines/colors for all of these 
#this will include 5-Fluorouracil and Doxorubicin

par(mar = c(3.0, 3.0, 1.2, 1), mgp = c(2, .7, 0))
layout(matrix(c(1, 2, 1, 3), nrow = 2, ncol = 2), heights = c(0.4, 2, 2, 2))
#layout.show(3)
plot.new()
text(0.5,0.5,&quot;RECOVERY CONCENTRATION-RESPONSE&quot;,cex = 2,font = 2)
#24hr plots
doxtox24_75_model &lt;- drm(response~Conc, data = dox24tox_75, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox24_75_model, type = &quot;none&quot;, broken = TRUE, 
     xt = c(0.01, 0.1, 0.5, 1, 5, 10, 50, 100, 1000), xttrim = FALSE,  
     ylim = c(0,110), cex = 1, col = &quot;#9ACCDE&quot;, lwd = 4, ylab = &quot;Percent Viability (%)&quot;,
     xlab = &quot;Concentration (uM)&quot;, main = &quot;24HR FOLLOWING TREATMENT&quot;, lty = c(1),
     xaxs = &quot;i&quot;, yaxs = &quot;i&quot;)

par(new=TRUE)
doxtox24_78_model &lt;- drm(response~Conc, data = dox24tox_78, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox24_78_model, add = TRUE, col = &quot;#499FBD&quot;, lwd = 4, lty = c(1), type = &quot;none&quot;)

par(new=TRUE)
doxtox24_17_model &lt;- drm(response~Conc, data = dox24tox_17, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox24_17_model, add = TRUE, col = &quot;#254F5E&quot;, lwd = 4, lty = c(1), type = &quot;none&quot;)

#48hr plots
par(new=TRUE)
doxtox48_75_model &lt;- drm(response~Conc, data = dox48tox_75, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(doxtox48_75_model, add = TRUE, col = &quot;#9ACCDE&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

par(new=TRUE)
doxtox48_78_model &lt;- drm(response~Conc, data = dox48tox_78, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(doxtox48_78_model, add = TRUE, col = &quot;#499FBD&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

par(new=TRUE)
doxtox48_17_model &lt;- drm(response~Conc, data = dox48tox_17, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox48_17_model, add = TRUE, col = &quot;#254F5E&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

#72hr plots
par(new=TRUE)
doxtox72_75_model &lt;- drm(response~Conc, data = dox72tox_75, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(doxtox72_75_model, add = TRUE, col = &quot;#9ACCDE&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)

par(new=TRUE)
doxtox72_78_model &lt;- drm(response~Conc, data = dox72tox_78, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox72_78_model, add = TRUE, col = &quot;#499FBD&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)

par(new=TRUE)
doxtox72_17_model &lt;- drm(response~Conc, data = dox72tox_17, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(doxtox72_17_model, add = TRUE, col = &quot;#254F5E&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)

par(new=TRUE)
legend(x = &quot;bottomleft&quot;, legend = c(&quot;Ind1 24hr&quot;, &quot;Ind2 24hr&quot;, &quot;Ind3 24hr&quot;, &quot;Ind1 48hr&quot;, &quot;Ind2 48hr&quot;, &quot;Ind3 48hr&quot;, &quot;Ind1 72hr&quot;, &quot;Ind2 72hr&quot;, &quot;Ind3 72hr&quot;), lty = c(1, 1, 1, 2, 2, 2, 3, 3, 3, 1), col = rep(c(&quot;#9ACCDE&quot;, &quot;#499FBD&quot;, &quot;#254F5E&quot;),3), lwd = 4, cex = 1)

#now add the fluo plots
#24hr plots
fluotox24_75_model &lt;- drm(response~Conc, data = fluo24tox_75, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox24_75_model, type = &quot;none&quot;, broken = TRUE, 
     xt = c(0.01, 0.1, 1, 5, 10, 50, 100, 1000, 10000), xttrim = FALSE,  
     ylim = c(0,110), cex = 1, col = &quot;#DCACED&quot;, lwd = 4, ylab = &quot;Percent Viability (%)&quot;,
     xlab = &quot;Concentration (uM)&quot;, main = &quot;5-FLUOROURACIL TREATMENT&quot;, lty = c(1),
     xaxs = &quot;i&quot;, yaxs = &quot;i&quot;)

par(new=TRUE)
fluotox24_78_model &lt;- drm(response~Conc, data = fluo24tox_78, fct = LL.4(fixed = c(NA, 0, 100, NA)))
plot(fluotox24_78_model, add = TRUE, col = &quot;#9B72AA&quot;, lwd = 4, lty = c(1), type = &quot;none&quot;)

par(new=TRUE)
fluotox24_17_model &lt;- drm(response~Conc, data = fluo24tox_17, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox24_17_model, add = TRUE, col = &quot;#674C71&quot;, lwd = 4, lty = c(1), type = &quot;none&quot;)

#48hr plots
par(new=TRUE)
fluotox48_75_model &lt;- drm(response~Conc, data = fluo48tox_75, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox48_75_model, add = TRUE, col = &quot;#DCACED&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

par(new=TRUE)
fluotox48_78_model &lt;- drm(response~Conc, data = fluo48tox_78, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox48_78_model, add = TRUE, col = &quot;#9B72AA&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

par(new=TRUE)
fluotox48_17_model &lt;- drm(response~Conc, data = fluo48tox_17, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(fluotox48_17_model, add = TRUE, col = &quot;#674C71&quot;, lwd = 4, lty = c(2), type = &quot;none&quot;)

#72hr plots
par(new=TRUE)
fluotox72_78_model &lt;- drm(response~Conc, data = fluo72tox_78, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(fluotox72_75_model, add = TRUE, col = &quot;#DCACED&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)

par(new=TRUE)
fluotox72_78_model &lt;- drm(response~Conc, data = fluo72tox_78, fct = LL.4(fixed = c(NA, NA, 100, NA)))
plot(fluotox72_78_model, add = TRUE, col = &quot;#9B72AA&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)

par(new=TRUE)
fluotox72_17_model &lt;- drm(response~Conc, data = fluo72tox_17, fct = LL.4(fixed = c(NA, NA, NA, NA)))
plot(fluotox72_17_model, add = TRUE, col = &quot;#674C71&quot;, lwd = 4, lty = c(3), type = &quot;none&quot;)

legend(x = &quot;bottomleft&quot;, legend = c(&quot;Ind1 24hr&quot;, &quot;Ind2 24hr&quot;, &quot;Ind3 24hr&quot;, &quot;Ind1 48hr&quot;, &quot;Ind2 48hr&quot;, &quot;Ind3 48hr&quot;, &quot;Ind1 72hr&quot;, &quot;Ind2 72hr&quot;, &quot;Ind3 72hr&quot;), lty = c(1, 1, 1, 2, 2, 2, 3, 3, 3, 1), col = rep(c(&quot;#DCACED&quot;, &quot;#9B72AA&quot;, &quot;#674C71&quot;),3), lwd = 4)</code></pre>
<p><img src="figure/Recovery_DRC.Rmd/DRCs%20All%20Timepoints-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#had to change the legends to show all of the line types and colors
#24hr = line 1, 48hr = line 2, 72hr = line 3</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.2 (2024-10-31 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 11 x64 (build 22000)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] readxl_1.4.3    drc_3.0-1       MASS_7.3-61     scales_1.3.0   
 [5] lubridate_1.9.3 forcats_1.0.0   stringr_1.5.1   dplyr_1.1.4    
 [9] purrr_1.0.2     readr_2.1.5     tidyr_1.3.1     tibble_3.2.1   
[13] ggplot2_3.5.1   tidyverse_2.0.0 workflowr_1.7.1

loaded via a namespace (and not attached):
 [1] gtable_0.3.6      xfun_0.49         bslib_0.9.0       processx_3.8.5   
 [5] lattice_0.22-6    callr_3.7.6       tzdb_0.4.0        vctrs_0.6.5      
 [9] tools_4.4.2       ps_1.8.1          generics_0.1.3    sandwich_3.1-1   
[13] pkgconfig_2.0.3   Matrix_1.7-1      lifecycle_1.0.4   compiler_4.4.2   
[17] git2r_0.35.0      munsell_0.5.1     codetools_0.2-20  getPass_0.2-4    
[21] carData_3.0-5     httpuv_1.6.15     htmltools_0.5.8.1 sass_0.4.9       
[25] yaml_2.3.10       Formula_1.2-5     later_1.4.1       pillar_1.10.1    
[29] car_3.1-3         jquerylib_0.1.4   whisker_0.4.1     cachem_1.1.0     
[33] abind_1.4-8       multcomp_1.4-28   gtools_3.9.5      tidyselect_1.2.1 
[37] digest_0.6.37     mvtnorm_1.3-3     stringi_1.8.4     splines_4.4.2    
[41] rprojroot_2.0.4   fastmap_1.2.0     grid_4.4.2        colorspace_2.1-1 
[45] cli_3.6.3         magrittr_2.0.3    survival_3.7-0    TH.data_1.1-3    
[49] withr_3.0.2       promises_1.3.2    plotrix_3.8-4     timechange_0.3.0 
[53] rmarkdown_2.29    httr_1.4.7        cellranger_1.1.0  zoo_1.8-12       
[57] hms_1.1.3         evaluate_1.0.3    knitr_1.49        rlang_1.1.4      
[61] Rcpp_1.0.13-1     glue_1.8.0        rstudioapi_0.17.1 jsonlite_1.8.9   
[65] R6_2.6.1          fs_1.6.5         </code></pre>
</div>
</div>
</div>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
