---
title: "Recovery RNAseq QC Analysis"
author: "Emma M Pfortmiller"
date: "2025-02-10"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
    toc_depth: 2
    toc_float: true
    theme: journal
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
```

```{r Libraries, echo=FALSE, results='hide', collapse=TRUE}
library(ggplot2)
library(ggrepel)
library(reshape2)
library(hrbrthemes)
library(tibble)
library(tidyverse)
library(edgebundleR)
library(readxl)
library(readr)
library(edgeR)
library(pheatmap)
library(ggfortify)
library(PCAtools)
library(Cormotif)
library(RColorBrewer)
library(biomaRt)
library(RUVSeq)
library(Rfast)
```


```{r Sheet with Mapped and Successfully Aligned Reads, include = FALSE}
fC_AllCounts <- read_csv("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/Recovery_MappingStats_Subread_EMP_241104_250211.csv")
View(fC_AllCounts)
dim(fC_AllCounts)
#[1] 63  6

#saveRDS(fC_AllCounts, "data/fC_AllCounts")
```

These samples (63 in total) have been aligned to the Ensembl 113 release of GRCh38 via subread, and counts have been generated using featureCounts with inbuilt hg38 genome

Data is located under 5FU Project on my computer
All analysis saved in RDirectory Recovery_RNAseq

Individual 1 - 84-1 (9 samples)
```{r Import Counts Data Individual 1 - 84-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line
#problem sample 24!!

#84-1 DOX_24-> R27
  Counts_84_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R27_R1.counts.txt")  
  #View(Counts_84_DOX_24)



#84-1 FLUO_24 -> R28
  Counts_84_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R28_R1.counts.txt")
  #View(Counts_84_FLUO_24)
  
#84-1 DMSO_24 -> R29
  Counts_84_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R29_R1.counts.txt")
  #View(Counts_84_DMSO_24)
  
#84-1 DOX_24+24-> R30
  Counts_84_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R30_R1.counts.txt")
  #View(Counts_84_DOX_24rec)
  
#84-1 FLUO_24+24-> R31
  Counts_84_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R31_R1.counts.txt")
  #View(Counts_84_FLUO_24rec)
  
#84-1 DMSO_24+24-> R32
  Counts_84_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R32_R1.counts.txt")
  #View(Counts_84_DMSO_24rec)
  
#84-1 DOX_24+144 -> R33
  Counts_84_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R33_R1.counts.txt")
  #View(Counts_84_DOX_144rec)

#84-1 FLUO_24+144 -> R34
   Counts_84_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R34_R1.counts.txt")
  #View(Counts_84_FLUO_144rec)
  
#84-1 DMSO_24+144 -> R35
   Counts_84_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R35_R1.counts.txt")
  #View(Counts_84_DMSO_144rec)
  
```

Individual 2 - 87-1 (9 samples)
```{r Import Counts Individual 2 87-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#87-1 DOX_24-> R36
  Counts_87_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R36_R1.counts.txt")  
  #View(Counts_87_DOX_24)

#87-1 FLUO_24 -> R37
  Counts_87_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R37_R1.counts.txt")
  #View(Counts_87_FLUO_24)
  
#87-1 DMSO_24 -> R38
  Counts_87_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R38_R1.counts.txt")
  #View(Counts_87_DMSO_24)
  
#87-1 DOX_24+24-> R39
  Counts_87_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R39_R1.counts.txt")
  #View(Counts_87_DOX_24rec)
  
#87-1 FLUO_24+24-> R40
  Counts_87_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R40_R1.counts.txt")
  #View(Counts_87_FLUO_24rec)
  
#87-1 DMSO_24+24-> R41
  Counts_87_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R41_R1.counts.txt")
  #View(Counts_87_DMSO_24rec)
  
#87-1 DOX_24+144 -> R42
  Counts_87_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R42_R1.counts.txt")
  #View(Counts_87_DOX_144rec)

#87-1 FLUO_24+144 -> R43
   Counts_87_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R43_R1.counts.txt")
  #View(Counts_87_FLUO_144rec)
  
#87-1 DMSO_24+144 -> R44
   Counts_87_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R44_R1.counts.txt")
  #View(Counts_87_DMSO_144rec)
  
```

Individual 3 - 75-1 (9 samples)
```{r Import Counts Individual 3 78-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#78-1 DOX_24-> R45
  Counts_78_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R45_R1.counts.txt")  
  #View(Counts_78_DOX_24)

#78-1 FLUO_24 -> R46
  Counts_78_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R46_R1.counts.txt")
  #View(Counts_78_FLUO_24)
  
#78-1 DMSO_24 -> R47
  Counts_78_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R47_R1.counts.txt")
  #View(Counts_78_DMSO_24)
  
#78-1 DOX_24+24-> R48
  Counts_78_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R48_R1.counts.txt")
  #View(Counts_78_DOX_24rec)
  
#78-1 FLUO_24+24-> R49
  Counts_78_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R49_R1.counts.txt")
  #View(Counts_78_FLUO_24rec)
  
#78-1 DMSO_24+24-> R50
  Counts_78_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R50_R1.counts.txt")
  #View(Counts_78_DMSO_24rec)
  
#78-1 DOX_24+144 -> R51
  Counts_78_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R51_R1.counts.txt")
  #View(Counts_78_DOX_144rec)

#78-1 FLUO_24+144 -> R52
   Counts_78_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R52_R1.counts.txt")
  #View(Counts_78_FLUO_144rec)
  
#78-1 DMSO_24+144 -> R53
   Counts_78_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R53_R1.counts.txt")
  #View(Counts_78_DMSO_144rec)
   
```

Individual 4 - 75-1 (9 samples)
```{r Import Counts Individual 4 75-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#75-1 DOX_24-> R54
  Counts_75_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R54_R1.counts.txt")  
  #View(Counts_75_DOX_24)

#75-1 FLUO_24 -> R55
  Counts_75_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R55_R1.counts.txt")
  #View(Counts_75_FLUO_24)
  
#75-1 DMSO_24 -> R56
  Counts_75_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R56_R1.counts.txt")
  #View(Counts_75_DMSO_24)
  
#75-1 DOX_24+24-> R57
  Counts_75_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R57_R1.counts.txt")
  #View(Counts_75_DOX_24rec)
  
#75-1 FLUO_24+24-> R58
  Counts_75_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R58_R1.counts.txt")
  #View(Counts_75_FLUO_24rec)
  
#75-1 DMSO_24+24-> R59
  Counts_75_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R59_R1.counts.txt")
  #View(Counts_75_DMSO_24rec)
  
#75-1 DOX_24+144 -> R60
  Counts_75_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R60_R1.counts.txt")
  #View(Counts_75_DOX_144rec)

#75-1 FLUO_24+144 -> R61
   Counts_75_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R61_R1.counts.txt")
  #View(Counts_75_FLUO_144rec)
  
#75-1 DMSO_24+144 -> R62
   Counts_75_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R62_R1.counts.txt")
  #View(Counts_75_DMSO_144rec)
   
```


Individual 5 - 17-3 (9 samples)
```{r Import Counts Individual 5 17-3, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#17-3 DOX_24-> R63
  Counts_17_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R63_R1.counts.txt")  
  #View(Counts_17_DOX_24)

#17-3 FLUO_24 -> R64
  Counts_17_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R64_R1.counts.txt")
  #View(Counts_17_FLUO_24)
  
#17-3 DMSO_24 -> R65
  Counts_17_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R65_R1.counts.txt")
  #View(Counts_17_DMSO_24)
  
#17-3 DOX_24+24-> R66
  Counts_17_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R66_R1.counts.txt")
  #View(Counts_17_DOX_24rec)
  
#17-3 FLUO_24+24-> R67
  Counts_17_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R67_R1.counts.txt")
  #View(Counts_17_FLUO_24rec)
  
#17-3 DMSO_24+24-> R68
  Counts_17_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R68_R1.counts.txt")
  #View(Counts_17_DMSO_24rec)
  
#17-3 DOX_24+144 -> R69
  Counts_17_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R69_R1.counts.txt")
  #View(Counts_17_DOX_144rec)

#17-3 FLUO_24+144 -> R70
   Counts_17_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R70_R1.counts.txt")
  #View(Counts_17_FLUO_144rec)
  
#17-3 DMSO_24+144 -> R71
   Counts_17_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R71_R1.counts.txt")
  #View(Counts_17_DMSO_144rec)
   
```


Individual 6 - 90-1 (9 samples)
```{r Import Counts Individual 6 90-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#90-1 DOX_24-> R72
  Counts_90_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R72_R1.counts.txt")  
  #View(Counts_90_DOX_24)

#90-1 FLUO_24 -> R73
  Counts_90_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R73_R1.counts.txt")
  #View(Counts_90_FLUO_24)
  
#90-1 DMSO_24 -> R74
  Counts_90_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R74_R1.counts.txt")
  #View(Counts_90_DMSO_24)
  
#90-1 DOX_24+24-> R75
  Counts_90_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R75_R1.counts.txt")
  #View(Counts_90_DOX_24rec)
  
#90-1 FLUO_24+24-> R76
  Counts_90_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R76_R1.counts.txt")
  #View(Counts_90_FLUO_24rec)
  
#90-1 DMSO_24+24-> R77
  Counts_90_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R77_R1.counts.txt")
  #View(Counts_90_DMSO_24rec)
  
#90-1 DOX_24+144 -> R78
  Counts_90_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R78_R1.counts.txt")
  #View(Counts_90_DOX_144rec)

#90-1 FLUO_24+144 -> R79
   Counts_90_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R79_R1.counts.txt")
  #View(Counts_90_FLUO_144rec)
  
#90-1 DMSO_24+144 -> R80
   Counts_90_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R80_R1.counts.txt")
  #View(Counts_90_DMSO_144rec)
   
```


Individual 7 - 90-1 REP (9 samples)
```{r Import Counts Individual 7 90-1 REP, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#90-1 REP DOX_24-> R81
  Counts_90REP_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R81_R1.counts.txt")  
  #View(Counts_90_DOX_24)

#90-1 REP FLUO_24 -> R82
  Counts_90REP_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R82_R1.counts.txt")
  #View(Counts_90_FLUO_24)
  
#90-1 REP DMSO_24 -> R83
  Counts_90REP_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R83_R1.counts.txt")
  #View(Counts_90_DMSO_24)
  
#90-1 REP DOX_24+24-> R84
  Counts_90REP_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R84_R1.counts.txt")
  #View(Counts_90_DOX_24rec)
  
#90-1 REP FLUO_24+24-> R85
  Counts_90REP_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R85_R1.counts.txt")
  #View(Counts_90_FLUO_24rec)
  
#90-1 REP DMSO_24+24-> R86
  Counts_90REP_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R86_R1.counts.txt")
  #View(Counts_90_DMSO_24rec)
  
#90-1 REP DOX_24+144 -> R87
  Counts_90REP_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R87_R1.counts.txt")
  #View(Counts_90_DOX_144rec)

#90-1 REP FLUO_24+144 -> R88
   Counts_90REP_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R88_R1.counts.txt")
  #View(Counts_90_FLUO_144rec)
  
#90-1 REP DMSO_24+144 -> R89
   Counts_90REP_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R89_R1.counts.txt")
  #View(Counts_90_DMSO_144rec)
   
```
Now that I have imported all of these counts files, I will make a matrix with all of the samples

First I'll break them up by individual, and then combine them all together
```{r Counts Matrix 84-1, include=FALSE}

fC_Matrix_84_All <-
  data.frame(
    Counts_84_DOX_24,
    Counts_84_FLUO_24$MCW_EMP_JT_R28_R1.bam,
    Counts_84_DMSO_24$MCW_EMP_JT_R29_R1.bam,
    Counts_84_DOX_24rec$MCW_EMP_JT_R30_R1.bam,
    Counts_84_FLUO_24rec$MCW_EMP_JT_R31_R1.bam,
    Counts_84_DMSO_24rec$MCW_EMP_JT_R32_R1.bam,
    Counts_84_DOX_144rec$MCW_EMP_JT_R33_R1.bam,
    Counts_84_FLUO_144rec$MCW_EMP_JT_R34_R1.bam,
    Counts_84_DMSO_144rec$MCW_EMP_JT_R35_R1.bam
  )
colnames(fC_Matrix_84_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind1",
                                "FLUO_24_Ind1",
                                "DMSO_24_Ind1",
                                "DOX_24rec_Ind1",
                                "FLUO_24rec_Ind1",
                                "DMSO_24rec_Ind1",
                                "DOX_144rec_Ind1",
                                "FLUO_144rec_Ind1",
                                "DMSO_144rec_Ind1")

fC_Matrix_84 <- fC_Matrix_84_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
#write.csv(fC_Matrix_84_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_84-1_EMP_250210.csv")

boxplot(fC_Matrix_84,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 84-1")

hist(fC_Matrix_84,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 84-1")


```

```{r Counts Matrix Individual 2 87-1, include=FALSE}

fC_Matrix_87_All <-
  data.frame(
    Counts_87_DOX_24,
    Counts_87_FLUO_24$MCW_EMP_JT_R37_R1.bam,
    Counts_87_DMSO_24$MCW_EMP_JT_R38_R1.bam,
    Counts_87_DOX_24rec$MCW_EMP_JT_R39_R1.bam,
    Counts_87_FLUO_24rec$MCW_EMP_JT_R40_R1.bam,
    Counts_87_DMSO_24rec$MCW_EMP_JT_R41_R1.bam,
    Counts_87_DOX_144rec$MCW_EMP_JT_R42_R1.bam,
    Counts_87_FLUO_144rec$MCW_EMP_JT_R43_R1.bam,
    Counts_87_DMSO_144rec$MCW_EMP_JT_R44_R1.bam
  )
colnames(fC_Matrix_87_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind2",
                                "FLUO_24_Ind2",
                                "DMSO_24_Ind2",
                                "DOX_24rec_Ind2",
                                "FLUO_24rec_Ind2",
                                "DMSO_24rec_Ind2",
                                "DOX_144rec_Ind2",
                                "FLUO_144rec_Ind2",
                                "DMSO_144rec_Ind2")

fC_Matrix_87 <- fC_Matrix_87_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_87_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_87-1_EMP_250210.csv")

boxplot(fC_Matrix_87,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 87-1")

hist(fC_Matrix_87,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 87-1")


```


```{r Counts Matrix Individual 3 78-1, include=FALSE}

fC_Matrix_78_All <-
  data.frame(
    Counts_78_DOX_24,
    Counts_78_FLUO_24$MCW_EMP_JT_R46_R1.bam,
    Counts_78_DMSO_24$MCW_EMP_JT_R47_R1.bam,
    Counts_78_DOX_24rec$MCW_EMP_JT_R48_R1.bam,
    Counts_78_FLUO_24rec$MCW_EMP_JT_R49_R1.bam,
    Counts_78_DMSO_24rec$MCW_EMP_JT_R50_R1.bam,
    Counts_78_DOX_144rec$MCW_EMP_JT_R51_R1.bam,
    Counts_78_FLUO_144rec$MCW_EMP_JT_R52_R1.bam,
    Counts_78_DMSO_144rec$MCW_EMP_JT_R53_R1.bam
  )
colnames(fC_Matrix_78_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind3",
                                "FLUO_24_Ind3",
                                "DMSO_24_Ind3",
                                "DOX_24rec_Ind3",
                                "FLUO_24rec_Ind3",
                                "DMSO_24rec_Ind3",
                                "DOX_144rec_Ind3",
                                "FLUO_144rec_Ind3",
                                "DMSO_144rec_Ind3")

fC_Matrix_78 <- fC_Matrix_78_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_78_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_78-1_EMP_250210.csv")

boxplot(fC_Matrix_78,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 78-1")

hist(fC_Matrix_78,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 78-1")


```


```{r Counts Matrix Individual 4 75-1, include=FALSE}

fC_Matrix_75_All <-
  data.frame(
    Counts_75_DOX_24,
    Counts_75_FLUO_24$MCW_EMP_JT_R55_R1.bam,
    Counts_75_DMSO_24$MCW_EMP_JT_R56_R1.bam,
    Counts_75_DOX_24rec$MCW_EMP_JT_R57_R1.bam,
    Counts_75_FLUO_24rec$MCW_EMP_JT_R58_R1.bam,
    Counts_75_DMSO_24rec$MCW_EMP_JT_R59_R1.bam,
    Counts_75_DOX_144rec$MCW_EMP_JT_R60_R1.bam,
    Counts_75_FLUO_144rec$MCW_EMP_JT_R61_R1.bam,
    Counts_75_DMSO_144rec$MCW_EMP_JT_R62_R1.bam
  )
colnames(fC_Matrix_75_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind4",
                                "FLUO_24_Ind4",
                                "DMSO_24_Ind4",
                                "DOX_24rec_Ind4",
                                "FLUO_24rec_Ind4",
                                "DMSO_24rec_Ind4",
                                "DOX_144rec_Ind4",
                                "FLUO_144rec_Ind4",
                                "DMSO_144rec_Ind4")

fC_Matrix_75 <- fC_Matrix_75_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_75_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_75-1_EMP_250210.csv")

boxplot(fC_Matrix_75,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 75-1")

hist(fC_Matrix_75,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 75-1")


```


```{r Counts Matrix Individual 5 17-3, include=FALSE}

fC_Matrix_17_All <-
  data.frame(
    Counts_17_DOX_24,
    Counts_17_FLUO_24$MCW_EMP_JT_R64_R1.bam,
    Counts_17_DMSO_24$MCW_EMP_JT_R65_R1.bam,
    Counts_17_DOX_24rec$MCW_EMP_JT_R66_R1.bam,
    Counts_17_FLUO_24rec$MCW_EMP_JT_R67_R1.bam,
    Counts_17_DMSO_24rec$MCW_EMP_JT_R68_R1.bam,
    Counts_17_DOX_144rec$MCW_EMP_JT_R69_R1.bam,
    Counts_17_FLUO_144rec$MCW_EMP_JT_R70_R1.bam,
    Counts_17_DMSO_144rec$MCW_EMP_JT_R71_R1.bam
  )
colnames(fC_Matrix_17_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind5",
                                "FLUO_24_Ind5",
                                "DMSO_24_Ind5",
                                "DOX_24rec_Ind5",
                                "FLUO_24rec_Ind5",
                                "DMSO_24rec_Ind5",
                                "DOX_144rec_Ind5",
                                "FLUO_144rec_Ind5",
                                "DMSO_144rec_Ind5")

fC_Matrix_17 <- fC_Matrix_17_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_17_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_17-1_EMP_250210.csv")

boxplot(fC_Matrix_17,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 17-3")

hist(fC_Matrix_17,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 17-3")


```


```{r Counts Matrix Individual 6 90-1, include=FALSE}

fC_Matrix_90_All <-
  data.frame(
    Counts_90_DOX_24,
    Counts_90_FLUO_24$MCW_EMP_JT_R73_R1.bam,
    Counts_90_DMSO_24$MCW_EMP_JT_R74_R1.bam,
    Counts_90_DOX_24rec$MCW_EMP_JT_R75_R1.bam,
    Counts_90_FLUO_24rec$MCW_EMP_JT_R76_R1.bam,
    Counts_90_DMSO_24rec$MCW_EMP_JT_R77_R1.bam,
    Counts_90_DOX_144rec$MCW_EMP_JT_R78_R1.bam,
    Counts_90_FLUO_144rec$MCW_EMP_JT_R79_R1.bam,
    Counts_90_DMSO_144rec$MCW_EMP_JT_R80_R1.bam
  )
colnames(fC_Matrix_90_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind6",
                                "FLUO_24_Ind6",
                                "DMSO_24_Ind6",
                                "DOX_24rec_Ind6",
                                "FLUO_24rec_Ind6",
                                "DMSO_24rec_Ind6",
                                "DOX_144rec_Ind6",
                                "FLUO_144rec_Ind6",
                                "DMSO_144rec_Ind6")

fC_Matrix_90 <- fC_Matrix_90_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_90_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_90-1_EMP_250210.csv")

boxplot(fC_Matrix_90,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 90-1")

hist(fC_Matrix_90,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 90-1")


```


```{r Counts Matrix Individual 6 90-1REP, include=FALSE}

fC_Matrix_90REP_All <-
  data.frame(
    Counts_90REP_DOX_24,
    Counts_90REP_FLUO_24$MCW_EMP_JT_R82_R1.bam,
    Counts_90REP_DMSO_24$MCW_EMP_JT_R83_R1.bam,
    Counts_90REP_DOX_24rec$MCW_EMP_JT_R84_R1.bam,
    Counts_90REP_FLUO_24rec$MCW_EMP_JT_R85_R1.bam,
    Counts_90REP_DMSO_24rec$MCW_EMP_JT_R86_R1.bam,
    Counts_90REP_DOX_144rec$MCW_EMP_JT_R87_R1.bam,
    Counts_90REP_FLUO_144rec$MCW_EMP_JT_R88_R1.bam,
    Counts_90REP_DMSO_144rec$MCW_EMP_JT_R89_R1.bam
  )
colnames(fC_Matrix_90REP_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind6REP",
                                "FLUO_24_Ind6REP",
                                "DMSO_24_Ind6REP",
                                "DOX_24rec_Ind6REP",
                                "FLUO_24rec_Ind6REP",
                                "DMSO_24rec_Ind6REP",
                                "DOX_144rec_Ind6REP",
                                "FLUO_144rec_Ind6REP",
                                "DMSO_144rec_Ind6REP")

fC_Matrix_90REP <- fC_Matrix_90REP_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_90REP_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_90-1REP_EMP_250210.csv")

boxplot(fC_Matrix_90REP,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 90-1 REP")

hist(fC_Matrix_90REP,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 90-1 REP")


```


Now that I've made individual matrices for each individual - I will combine them together into one large matrix

```{r Counts Matrix All Individuals, results='hide', echo=FALSE}

fC_Matrix_AllInd <-
  data.frame(
    Counts_84_DOX_24,
    Counts_84_FLUO_24$MCW_EMP_JT_R28_R1.bam,
    Counts_84_DMSO_24$MCW_EMP_JT_R29_R1.bam,
    Counts_84_DOX_24rec$MCW_EMP_JT_R30_R1.bam,
    Counts_84_FLUO_24rec$MCW_EMP_JT_R31_R1.bam,
    Counts_84_DMSO_24rec$MCW_EMP_JT_R32_R1.bam,
    Counts_84_DOX_144rec$MCW_EMP_JT_R33_R1.bam,
    Counts_84_FLUO_144rec$MCW_EMP_JT_R34_R1.bam,
    Counts_84_DMSO_144rec$MCW_EMP_JT_R35_R1.bam,
    Counts_87_DOX_24$MCW_EMP_JT_R36_R1.bam,
    Counts_87_FLUO_24$MCW_EMP_JT_R37_R1.bam,
    Counts_87_DMSO_24$MCW_EMP_JT_R38_R1.bam,
    Counts_87_DOX_24rec$MCW_EMP_JT_R39_R1.bam,
    Counts_87_FLUO_24rec$MCW_EMP_JT_R40_R1.bam,
    Counts_87_DMSO_24rec$MCW_EMP_JT_R41_R1.bam,
    Counts_87_DOX_144rec$MCW_EMP_JT_R42_R1.bam,
    Counts_87_FLUO_144rec$MCW_EMP_JT_R43_R1.bam,
    Counts_87_DMSO_144rec$MCW_EMP_JT_R44_R1.bam,
    Counts_78_DOX_24$MCW_EMP_JT_R45_R1.bam,
    Counts_78_FLUO_24$MCW_EMP_JT_R46_R1.bam,
    Counts_78_DMSO_24$MCW_EMP_JT_R47_R1.bam,
    Counts_78_DOX_24rec$MCW_EMP_JT_R48_R1.bam,
    Counts_78_FLUO_24rec$MCW_EMP_JT_R49_R1.bam,
    Counts_78_DMSO_24rec$MCW_EMP_JT_R50_R1.bam,
    Counts_78_DOX_144rec$MCW_EMP_JT_R51_R1.bam,
    Counts_78_FLUO_144rec$MCW_EMP_JT_R52_R1.bam,
    Counts_78_DMSO_144rec$MCW_EMP_JT_R53_R1.bam,
    Counts_75_DOX_24$MCW_EMP_JT_R54_R1.bam,
    Counts_75_FLUO_24$MCW_EMP_JT_R55_R1.bam,
    Counts_75_DMSO_24$MCW_EMP_JT_R56_R1.bam,
    Counts_75_DOX_24rec$MCW_EMP_JT_R57_R1.bam,
    Counts_75_FLUO_24rec$MCW_EMP_JT_R58_R1.bam,
    Counts_75_DMSO_24rec$MCW_EMP_JT_R59_R1.bam,
    Counts_75_DOX_144rec$MCW_EMP_JT_R60_R1.bam,
    Counts_75_FLUO_144rec$MCW_EMP_JT_R61_R1.bam,
    Counts_75_DMSO_144rec$MCW_EMP_JT_R62_R1.bam,
    Counts_17_DOX_24$MCW_EMP_JT_R63_R1.bam,
    Counts_17_FLUO_24$MCW_EMP_JT_R64_R1.bam,
    Counts_17_DMSO_24$MCW_EMP_JT_R65_R1.bam,
    Counts_17_DOX_24rec$MCW_EMP_JT_R66_R1.bam,
    Counts_17_FLUO_24rec$MCW_EMP_JT_R67_R1.bam,
    Counts_17_DMSO_24rec$MCW_EMP_JT_R68_R1.bam,
    Counts_17_DOX_144rec$MCW_EMP_JT_R69_R1.bam,
    Counts_17_FLUO_144rec$MCW_EMP_JT_R70_R1.bam,
    Counts_17_DMSO_144rec$MCW_EMP_JT_R71_R1.bam,
    Counts_90_DOX_24$MCW_EMP_JT_R72_R1.bam,
    Counts_90_FLUO_24$MCW_EMP_JT_R73_R1.bam,
    Counts_90_DMSO_24$MCW_EMP_JT_R74_R1.bam,
    Counts_90_DOX_24rec$MCW_EMP_JT_R75_R1.bam,
    Counts_90_FLUO_24rec$MCW_EMP_JT_R76_R1.bam,
    Counts_90_DMSO_24rec$MCW_EMP_JT_R77_R1.bam,
    Counts_90_DOX_144rec$MCW_EMP_JT_R78_R1.bam,
    Counts_90_FLUO_144rec$MCW_EMP_JT_R79_R1.bam,
    Counts_90_DMSO_144rec$MCW_EMP_JT_R80_R1.bam,
    Counts_90REP_DOX_24$MCW_EMP_JT_R81_R1.bam,
    Counts_90REP_FLUO_24$MCW_EMP_JT_R82_R1.bam,
    Counts_90REP_DMSO_24$MCW_EMP_JT_R83_R1.bam,
    Counts_90REP_DOX_24rec$MCW_EMP_JT_R84_R1.bam,
    Counts_90REP_FLUO_24rec$MCW_EMP_JT_R85_R1.bam,
    Counts_90REP_DMSO_24rec$MCW_EMP_JT_R86_R1.bam,
    Counts_90REP_DOX_144rec$MCW_EMP_JT_R87_R1.bam,
    Counts_90REP_FLUO_144rec$MCW_EMP_JT_R88_R1.bam,
    Counts_90REP_DMSO_144rec$MCW_EMP_JT_R89_R1.bam
  )

colnames(fC_Matrix_AllInd) <- c("ensembl_gene_id", 
                                "DOX_24_Ind1",
                                "FLUO_24_Ind1",
                                "DMSO_24_Ind1",
                                "DOX_24rec_Ind1",
                                "FLUO_24rec_Ind1",
                                "DMSO_24rec_Ind1",
                                "DOX_144rec_Ind1",
                                "FLUO_144rec_Ind1",
                                "DMSO_144rec_Ind1",
                                "DOX_24_Ind2",
                                "FLUO_24_Ind2",
                                "DMSO_24_Ind2",
                                "DOX_24rec_Ind2",
                                "FLUO_24rec_Ind2",
                                "DMSO_24rec_Ind2",
                                "DOX_144rec_Ind2",
                                "FLUO_144rec_Ind2",
                                "DMSO_144rec_Ind2",
                                "DOX_24_Ind3",
                                "FLUO_24_Ind3",
                                "DMSO_24_Ind3",
                                "DOX_24rec_Ind3",
                                "FLUO_24rec_Ind3",
                                "DMSO_24rec_Ind3",
                                "DOX_144rec_Ind3",
                                "FLUO_144rec_Ind3",
                                "DMSO_144rec_Ind3",
                                "DOX_24_Ind4",
                                "FLUO_24_Ind4",
                                "DMSO_24_Ind4",
                                "DOX_24rec_Ind4",
                                "FLUO_24rec_Ind4",
                                "DMSO_24rec_Ind4",
                                "DOX_144rec_Ind4",
                                "FLUO_144rec_Ind4",
                                "DMSO_144rec_Ind4",
                                "DOX_24_Ind5",
                                "FLUO_24_Ind5",
                                "DMSO_24_Ind5",
                                "DOX_24rec_Ind5",
                                "FLUO_24rec_Ind5",
                                "DMSO_24rec_Ind5",
                                "DOX_144rec_Ind5",
                                "FLUO_144rec_Ind5",
                                "DMSO_144rec_Ind5",
                                "DOX_24_Ind6",
                                "FLUO_24_Ind6",
                                "DMSO_24_Ind6",
                                "DOX_24rec_Ind6",
                                "FLUO_24rec_Ind6",
                                "DMSO_24rec_Ind6",
                                "DOX_144rec_Ind6",
                                "FLUO_144rec_Ind6",
                                "DMSO_144rec_Ind6",
                                "DOX_24_Ind6REP",
                                "FLUO_24_Ind6REP",
                                "DMSO_24_Ind6REP",
                                "DOX_24rec_Ind6REP",
                                "FLUO_24rec_Ind6REP",
                                "DMSO_24rec_Ind6REP",
                                "DOX_144rec_Ind6REP",
                                "FLUO_144rec_Ind6REP",
                                "DMSO_144rec_Ind6REP")

fC_Matrix_Full <- fC_Matrix_AllInd %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
#write.csv(fC_Matrix_AllInd, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_AllSamples_EMP_250220.csv")

#saveRDS(fC_Matrix_Full, "data/fC_Matrix_Full.RDS")


boxplot(fC_Matrix_Full,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 1000), main = "Boxplot of Raw Counts All Individuals")

hist(fC_Matrix_Full,  xlab = "Unfiltered Counts", xlim = c(-1000,125000), ylim = c(-100, 6000000), main = "Histogram of Raw Counts All Individuals")


```


```{r Data Handling, results='hide', echo=FALSE, collapse=TRUE}

#Raw Counts
fC_Matrix_Full

#saveRDS(fC_Matrix_Full, "data/fC_Matrix_Full.RDS")

Counts_Full_df <- data.frame(fC_Matrix_Full)

#This code chunk in tandem with the one below to make sure my data's tucked away!


#define colors
ind_col <- c("#003F5C", "#45AE91",  "#58508D", "#BC4099", "#8B3E9B", "#FF6361", "#FF2362")

tx_col <- c("#499FBD", "#DCACED","#63666D")

time_col <- c("#fbb4b9", "#f768a1", "#ae017e")


#factor my df to get all the factors
annotate_Counts <- data.frame("samples" = colnames(Counts_Full_df)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)
#%>% unite(., col = "Treatment_Time", "Treatment", "Time", sep = "_", remove = FALSE)

#now cbind these factors of my data to my raw counts file, transpose first

annotate_Counts_full <- (t(Counts_Full_df)) %>%  cbind(., annotate_Counts)
 
#now I'll factor by sample to keep order

#Counts_Full_df %>% dplyr::add_row()

annot_Counts_full_df <- as.data.frame(t(annotate_Counts_full))

#this is for the log2cpm transformation below
#####log2cpm####
fC_Matrix_Full_cpm <- cpm(fC_Matrix_Full, log = TRUE)



```


```{r QC checks Raw Counts, fig.height=10, fig.width=12, collapse=TRUE}
#Now that I've put together my matrix - I will do some QC checks

#now that I have all of my factors and my gene info, I will make some graphs to show the QC data

reads_by_sample <- c(tx_col)
fC_AllCounts %>% 
  ggplot(., aes (x = Conditions, y = Total_Align, fill = Treatment, group_by = Line))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Read Counts by Treatment####
fC_AllCounts %>% 
  ggplot(., aes (x =Treatment, y= Total_Align, fill = Treatment))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by treatment"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Reads Per Individual####
fC_AllCounts %>% 
  ggplot(., aes (x =as.factor(Line), y=Total_Align))+
  geom_boxplot(aes(fill=as.factor(Line)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total number of reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Mapped Reads Per Drug####

reads_by_sample <- c(tx_col)
fC_AllCounts %>% 
  ggplot(., aes (x = Conditions, y = Assigned_Align, fill = Treatment, group_by = Line))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Read Counts by Treatment####
fC_AllCounts %>% 
  ggplot(., aes (x =Treatment, y= Assigned_Align, fill = Treatment))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by treatment"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Reads Per Individual####
fC_AllCounts %>% 
  ggplot(., aes (x =as.factor(Line), y=Assigned_Align))+
  geom_boxplot(aes(fill=as.factor(Line)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total number of mapped reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


```

Now that I've made the final full matrix of all samples -
1. convert to log2cpm using cpm
2. filter for lowly expressed reads (rowMeans > 0)

```{r log2cpm Conversion Full Matrix}
fC_Matrix_Full

fC_Matrix_Full_cpm <- cpm(fC_Matrix_Full, log = TRUE)

boxplot(fC_Matrix_Full_cpm, xlab = "Conditions", ylab = expression("log"[2]* "counts per million"), ylim= c(-10, 20))

hist(fC_Matrix_Full_cpm, xlab = expression("log" [2]* "counts per million"))

```


```{r Filter Lowly Expressed Reads All Samples, collapse=TRUE}

#first check the number of variables in the cpm file
dim(fC_Matrix_Full_cpm)
#[1] 28395    63

#now use rowMeans to filter out values with mean < 0 (this means rowMeans > 0 is what I'm left with) for each row/gene in the original file
rowMeans_All <- rowMeans(fC_Matrix_Full_cpm)
fC_Matrix_Full_cpm_filter <- fC_Matrix_Full_cpm[rowMeans_All > 0,]
dim(fC_Matrix_Full_cpm_filter)
#[1] 14225    63


#with this I've filtered out a significant number of rows, want ~14000

hist(fC_Matrix_Full_cpm_filter, 
     main = "Recovery All Individuals log2cpm Filtered", 
        xlab = expression("log"[2]*"counts per million"))
boxplot(fC_Matrix_Full_cpm_filter, 
        main = "Recovery All Individuals log2cpm Filtered", 
        xlab = "Conditions", 
        ylab = expression("log"[2]*"counts per million"),
        ylim = c(-10,15))

#save this to an R object
#saveRDS(fC_Matrix_Full_cpm_filter, "data/fC_Matrix_Full_cpm_filter.RDS")

```

```{r Filter List Object}
#this will be used to filter my datasets against for filtered counts
#through this, I can save the rownames to have my final filtered dataset for DE

filt_gene_list <- row.names(fC_Matrix_Full_cpm_filter)
#saveRDS(filt_gene_list, "data/filt_gene_list.RDS")

#this is what I'll be using for differential expression analysis (see DE page)

```

```{r QC Checks Unfiltered vs Filtered Reads}

####histogram of total counts, unfiltered####
hist(fC_Matrix_Full_cpm, main = "Recovery All Individuals log2cpm Unfiltered",
     xlab = expression("log" [2]* "counts per million"))

####histogram of total counts, filtered####
hist(fC_Matrix_Full_cpm_filter, 
     main = "Recovery All Individuals log2cpm Filtered rowMeans > 0", 
        xlab = expression("log"[2]*"counts per million"))

####boxplot log2cpm all samples####

boxplot(fC_Matrix_Full_cpm, xlab = "Conditions", ylab = expression("log"[2]* "counts per million"), ylim= c(-10, 20), main = "Boxplots of log2cpm per sample")

####boxplot log2cpm filtered all samples####

boxplot(fC_Matrix_Full_cpm_filter, xlab = "Conditions", ylab = expression("log"[2]* "counts per million"), ylim= c(-10, 20), main = "Boxplots of log2cpm Filtered rowMeans > 0 per sample")

```

Now that I've filtered the samples after log2cpm conversion, I can start to make heatmaps and PCA plots to look at my data. I want these plots to have treatment and time as the primary variables.

```{r PCA Plots Retired, eval=FALSE, fig.height=8, fig.width=8, collapse=TRUE, include=FALSE}
# PCA_data <- fC_Matrix_Full_cpm_filter %>% 
#   prcomp(.) %>% 
#   t()
# 
# PCA_data_test <- (prcomp(t(fC_Matrix_Full_cpm_filter), scale. = TRUE))
# 
# ####now make an annotation for my PCA####
# annot <- data.frame("samples" = colnames(fC_Matrix_Full_cpm_filter)) %>% separate_wider_delim(., cols = samples, names = c("Tx", "Time", "Ind"), delim = "_", cols_remove = FALSE) %>% unite(., col = "Tx_Time", Tx, Time, sep = "_", remove = FALSE)
# 
# #combine the prcomp matrix and annotation
# annot_PCA_matrix <- PCA_data_test$x %>% cbind(., annot)
# 
# #now I can make a graph where I have filled values for individual! I have seven colors for seven individuals
# # I have three fill values for three timepoints
# 
# #using annotation matrix above as well as annotated PCA matrix (annot_PCA_matrix)
# 
# #extra info for colors in the graph (fill parameter)
# fill_col_ind <- c("#66C2A5", "#FC8D62", "#1F78B4", "#E78AC3", "#A6D854", "#FFD92A", "#8B3E9B")
# 
# fill_col_ind_dark <- c("#003F5C", "#45AE91",  "#58508D", "#BC4099", "#8B3E9B", "#FF6361", "#FF2362")
# 
# fill_col_tx <- c("#499FBD", "#DCACED", "#63666D")
# 
# #Now I have to switch the fill and color parameters as it's doing the opposite of what I wanted: need individuals to be the outline color (color), and tx to be the inside color (fill)
# 
# ####PC1/PC2####
# annot_PCA_matrix %>% ggplot(., aes(x=PC1, y=PC2, size = 10)) +
#   geom_point(aes(color = Ind, shape = Time, fill = Tx)) +
#                        scale_shape_manual(values = c(21, 22, 24)) +
#   scale_fill_manual(values =  fill_col_tx)+
#   scale_color_manual(values = c(fill_col_ind_dark))+
#   guides(fill=guide_legend(override.aes=list(shape=21)))+
#   guides(color=guide_legend(override.aes=list(shape=21)))+
#   guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))
# 
# ####PC2/PC3####
# annot_PCA_matrix %>% ggplot(., aes(x=PC2, y=PC3)) +
#   geom_point(aes(color = Ind, shape = Time, fill = Tx, size = 10)) +
#                        scale_shape_manual(values = c(21, 22, 24)) +
#   scale_fill_manual(values =  fill_col_tx)+
#   scale_color_manual(values = c(fill_col_ind_dark))+
#   guides(fill=guide_legend(override.aes=list(shape=21)))+
#   guides(color=guide_legend(override.aes=list(shape=21)))+
#   guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))
# 
# ####PC3/PC4####
# annot_PCA_matrix %>% ggplot(., aes(x=PC3, y=PC4)) +
#   geom_point(aes(color = Ind, shape = Time, fill = Tx, size = 10)) +
#                        scale_shape_manual(values = c(21, 22, 24)) +
#   scale_fill_manual(values =  fill_col_tx)+
#   scale_color_manual(values = c(fill_col_ind_dark))+
#   guides(fill=guide_legend(override.aes=list(shape=21)))+
#   guides(color=guide_legend(override.aes=list(shape=21)))+
#   guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))

```


```{r PCA Plots, fig.width=8, fig.height=8, collapse=TRUE}
PCA_data <- fC_Matrix_Full_cpm_filter %>% 
  prcomp(.) %>% 
  t()

PCA_data_test <- (prcomp(t(fC_Matrix_Full_cpm_filter), scale. = TRUE))


#ind_num <- c("1", "1", "1", "1", "1", "1", "1", "1", "1","2", "2", "2", "2", "2", "2", "2", "2", "2", "3", "3", "3", "3", "3", "3", "3", "3", "3","4", "4", "4", "4", "4", "4", "4", "4", "4","5", "5", "5", "5", "5", "5", "5", "5", "5","6", "6", "6", "6", "6", "6", "6", "6", "6","6R", "6R", "6R", "6R", "6R", "6R", "6R", "6R", "6R")

#saveRDS(ind_num, "data/ind_num.RDS")

ind_num <- readRDS("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/ind_num.RDS")

####now make an annotation for my PCA####
annot <- data.frame("samples" = colnames(fC_Matrix_Full_cpm_filter)) %>% separate_wider_delim(., cols = samples, names = c("Tx", "Time", "Ind"), delim = "_", cols_remove = FALSE) %>% unite(., col = "Tx_Time", Tx, Time, sep = "_", remove = FALSE) %>% cbind(., ind_num)

#saveRDS(annot, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/annot.RDS")


#combine the prcomp matrix and annotation
annot_PCA_matrix <- PCA_data_test$x %>% cbind(., annot)


#now I can make a graph where I have filled values for individual! I have seven colors for seven individuals
# I have three fill values for three timepoints

#using annotation matrix above as well as annotated PCA matrix (annot_PCA_matrix)

#extra info for colors in the graph (fill parameter)
fill_col_ind <- c("#66C2A5", "#FC8D62", "#1F78B4", "#E78AC3", "#A6D854", "#FFD92A", "#8B3E9B")

fill_col_ind_dark <- c("#003F5C", "#45AE91",  "#58508D", "#BC4099", "#8B3E9B", "#FF6361", "#FF2362")

fill_col_tx <- c("#63666D", "#499FBD", "#DCACED")

#I want the color parameter to be treatment, the shape as time, and individual as number

####PC1/PC2####
annot_PCA_matrix %>% ggplot(., aes(x=PC1, y=PC2, size=10)) +
  geom_point(aes(color = Tx, shape = Time)) +
  scale_color_manual(values = c(fill_col_tx))+
  scale_shape_manual(values = c(15, 19, 17))+
  geom_text_repel(aes(label = ind_num))+
  theme_bw(base_size = 10)+
  ggtitle(expression("PCA of filtered log"[2]*"cpm"))+
  guides(size="none")
  

####PC2/PC3####
annot_PCA_matrix %>% ggplot(., aes(x=PC2, y=PC3, size=10)) +
  geom_point(aes(color = Tx, shape = Time)) +
  scale_color_manual(values = c(fill_col_tx))+
  scale_shape_manual(values = c(15, 19, 17))+
  geom_text_repel(aes(label = ind_num))+
  theme_bw(base_size = 10)+
  ggtitle(expression("PCA of filtered log"[2]*"cpm"))+
  guides(size="none")

####PC3/PC4####
annot_PCA_matrix %>% ggplot(., aes(x=PC3, y=PC4, size=10)) +
  geom_point(aes(color = Tx, shape = Time)) +
  scale_color_manual(values = c(fill_col_tx))+
  scale_shape_manual(values = c(15, 19, 17))+
  geom_text_repel(aes(label = ind_num))+
  theme_bw(base_size = 10)+
  ggtitle(expression("PCA of filtered log"[2]*"cpm"))+
  guides(size="none")

```

Now, let's make some correlation heatmaps
These are pairwise across all genes by condition
```{r Correlation Heatmaps, fig.width=12, fig.height=16, collapse=TRUE}

#correlation heatmap for all samples

####PEARSON FILTERED####
fC_Matrix_Full_cpm_filter_pearsoncor <-
  cor(
    fC_Matrix_Full_cpm_filter,
    y = NULL,
    use = "everything",
    method = "pearson"
  )

####SPEARMAN FILTERED####
fC_Matrix_Full_cpm_filter_spearmancor <-
  cor(
    fC_Matrix_Full_cpm_filter,
    y = NULL,
    use = "everything",
    method = "spearman"
  )



#Now let's graph these correlations

pheatmap(fC_Matrix_Full_cpm_filter_pearsoncor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 12, fontsize_number = 9)

pheatmap(fC_Matrix_Full_cpm_filter_spearmancor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 12, fontsize_number = 9)



#now annotate these heatmaps
#Factor 1 - Individual
Individual <- as.factor(c(rep("Ind1", 9), rep("Ind2", 9), rep("Ind3", 9), rep("Ind4", 9), rep("Ind5", 9), rep("Ind6", 9), rep("Ind6REP", 9)))

#Factor 2 - Treatment
tx_factor <- c("DOX", "FLUO", "DMSO")
Tx <- as.factor(c(rep(tx_factor, 21)))
#view(Treatment)

#Factor 3 - Timepoint
time_factor <- c(rep("24", 3), rep("24rec", 3), rep("144rec", 3))
Time <- as.factor(c(rep(time_factor, 7)))

####annotation for colors####
annot_col_hm = list(Tx = c(DOX = "red", FLUO = "blue", DMSO = "black"),
                             Ind = c(Ind1 = "#66C2A5", Ind2 = "#FC8D62", Ind3 = "#1F78B4", Ind4 = "#E78AC3", Ind5 = "#A6D854", Ind6 = "#FFD92A", Ind6REP = "#8B3E9B"),
                             Time = c("24" = "#046A38", "24rec" = "#0050B5", "144rec" = "#B725AD"))

####annotation for values####
annot_list_hm <- data.frame(Individual = as.factor(c(rep("Ind1", 9), rep("Ind2", 9), rep("Ind3", 9), rep("Ind4", 9), rep("Ind5", 9), rep("Ind6", 9), rep("Ind6REP", 9))),
                                               Tx = as.factor(c(rep(tx_factor, 21))), 
                                               Time = as.factor(c(rep(time_factor, 7))))

  
##add in the annotations from above into the dataframe
row.names(annot_list_hm) <- colnames(fC_Matrix_Full_cpm_filter_pearsoncor)
row.names(annot_list_hm) <- colnames(fC_Matrix_Full_cpm_filter_spearmancor)

####ANNOTATED HEATMAPS####
pheatmap(fC_Matrix_Full_cpm_filter_pearsoncor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 10, fontsize_number = 5, annotation_col = annot_list_hm, annotation_colors = annot_col_hm)

pheatmap(fC_Matrix_Full_cpm_filter_spearmancor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 10, fontsize_number = 5, annotation_col = annot_list_hm, annotation_colors = annot_col_hm)


```
