---
title: "5FU Recovery RNAseq Analysis"
author: "Emma M Pfortmiller"
date: "2025-02-10"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
    toc_depth: 2
    toc_float: true
    theme: journal
editor_options: 
  chunk_output_type: console
---

Welcome to my research website.

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


```{r Libraries, echo=FALSE, results='hide', collapse=TRUE}
library(ggplot2)
library(reshape2)
library(hrbrthemes)
library(tibble)
library(tidyverse)
library(edgebundleR)
library(readxl)
library(readr)
library(edgeR)
library(pheatmap)
library(ggfortify)
library(PCAtools)
library(Cormotif)
library(RColorBrewer)
library(biomaRt)
library(RUVSeq)
```



```{r Sheet with Mapped and Successfully Aligned Reads, include = FALSE}
fC_AllCounts <- read_csv("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/Recovery_MappingStats_Subread_EMP_241104_250211.csv")
View(fC_AllCounts)
dim(fC_AllCounts)
#[1] 63  6
```

These samples (63 in total) have been aligned to the Ensembl 113 release of GRCh38 via subread, and counts have been generated using featureCounts with inbuilt hg38 genome

Data is located under 5FU Project on my computer
All analysis saved in RDirectory Recovery_RNAseq

Individual 1 - 84-1 (9 samples)
```{r Import Counts Data Individual 1 - 84-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line
#problem sample 24!!

#84-1 DOX_24-> R27
  Counts_84_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R27_R1.counts.txt")  
  #View(Counts_84_DOX_24)



#84-1 FLUO_24 -> R28
  Counts_84_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R28_R1.counts.txt")
  #View(Counts_84_FLUO_24)
  
#84-1 DMSO_24 -> R29
  Counts_84_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R29_R1.counts.txt")
  #View(Counts_84_DMSO_24)
  
#84-1 DOX_24+24-> R30
  Counts_84_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R30_R1.counts.txt")
  #View(Counts_84_DOX_24rec)
  
#84-1 FLUO_24+24-> R31
  Counts_84_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R31_R1.counts.txt")
  #View(Counts_84_FLUO_24rec)
  
#84-1 DMSO_24+24-> R32
  Counts_84_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R32_R1.counts.txt")
  #View(Counts_84_DMSO_24rec)
  
#84-1 DOX_24+144 -> R33
  Counts_84_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R33_R1.counts.txt")
  #View(Counts_84_DOX_144rec)

#84-1 FLUO_24+144 -> R34
   Counts_84_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R34_R1.counts.txt")
  #View(Counts_84_FLUO_144rec)
  
#84-1 DMSO_24+144 -> R35
   Counts_84_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R35_R1.counts.txt")
  #View(Counts_84_DMSO_144rec)
  
```

Individual 2 - 87-1 (9 samples)
```{r Import Counts Individual 2 87-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#87-1 DOX_24-> R36
  Counts_87_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R36_R1.counts.txt")  
  #View(Counts_87_DOX_24)

#87-1 FLUO_24 -> R37
  Counts_87_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R37_R1.counts.txt")
  #View(Counts_87_FLUO_24)
  
#87-1 DMSO_24 -> R38
  Counts_87_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R38_R1.counts.txt")
  #View(Counts_87_DMSO_24)
  
#87-1 DOX_24+24-> R39
  Counts_87_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R39_R1.counts.txt")
  #View(Counts_87_DOX_24rec)
  
#87-1 FLUO_24+24-> R40
  Counts_87_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R40_R1.counts.txt")
  #View(Counts_87_FLUO_24rec)
  
#87-1 DMSO_24+24-> R41
  Counts_87_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R41_R1.counts.txt")
  #View(Counts_87_DMSO_24rec)
  
#87-1 DOX_24+144 -> R42
  Counts_87_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R42_R1.counts.txt")
  #View(Counts_87_DOX_144rec)

#87-1 FLUO_24+144 -> R43
   Counts_87_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R43_R1.counts.txt")
  #View(Counts_87_FLUO_144rec)
  
#87-1 DMSO_24+144 -> R44
   Counts_87_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R44_R1.counts.txt")
  #View(Counts_87_DMSO_144rec)
  
```

Individual 3 - 75-1 (9 samples)
```{r Import Counts Individual 3 78-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#78-1 DOX_24-> R45
  Counts_78_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R45_R1.counts.txt")  
  #View(Counts_78_DOX_24)

#78-1 FLUO_24 -> R46
  Counts_78_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R46_R1.counts.txt")
  #View(Counts_78_FLUO_24)
  
#78-1 DMSO_24 -> R47
  Counts_78_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R47_R1.counts.txt")
  #View(Counts_78_DMSO_24)
  
#78-1 DOX_24+24-> R48
  Counts_78_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R48_R1.counts.txt")
  #View(Counts_78_DOX_24rec)
  
#78-1 FLUO_24+24-> R49
  Counts_78_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R49_R1.counts.txt")
  #View(Counts_78_FLUO_24rec)
  
#78-1 DMSO_24+24-> R50
  Counts_78_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R50_R1.counts.txt")
  #View(Counts_78_DMSO_24rec)
  
#78-1 DOX_24+144 -> R51
  Counts_78_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R51_R1.counts.txt")
  #View(Counts_78_DOX_144rec)

#78-1 FLUO_24+144 -> R52
   Counts_78_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R52_R1.counts.txt")
  #View(Counts_78_FLUO_144rec)
  
#78-1 DMSO_24+144 -> R53
   Counts_78_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R53_R1.counts.txt")
  #View(Counts_78_DMSO_144rec)
   
```

Individual 4 - 75-1 (9 samples)
```{r Import Counts Individual 4 75-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#75-1 DOX_24-> R54
  Counts_75_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R54_R1.counts.txt")  
  #View(Counts_75_DOX_24)

#75-1 FLUO_24 -> R55
  Counts_75_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R55_R1.counts.txt")
  #View(Counts_75_FLUO_24)
  
#75-1 DMSO_24 -> R56
  Counts_75_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R56_R1.counts.txt")
  #View(Counts_75_DMSO_24)
  
#75-1 DOX_24+24-> R57
  Counts_75_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R57_R1.counts.txt")
  #View(Counts_75_DOX_24rec)
  
#75-1 FLUO_24+24-> R58
  Counts_75_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R58_R1.counts.txt")
  #View(Counts_75_FLUO_24rec)
  
#75-1 DMSO_24+24-> R59
  Counts_75_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R59_R1.counts.txt")
  #View(Counts_75_DMSO_24rec)
  
#75-1 DOX_24+144 -> R60
  Counts_75_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R60_R1.counts.txt")
  #View(Counts_75_DOX_144rec)

#75-1 FLUO_24+144 -> R61
   Counts_75_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R61_R1.counts.txt")
  #View(Counts_75_FLUO_144rec)
  
#75-1 DMSO_24+144 -> R62
   Counts_75_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R62_R1.counts.txt")
  #View(Counts_75_DMSO_144rec)
   
```


Individual 5 - 17-3 (9 samples)
```{r Import Counts Individual 5 17-3, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#17-3 DOX_24-> R63
  Counts_17_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R63_R1.counts.txt")  
  #View(Counts_17_DOX_24)

#17-3 FLUO_24 -> R64
  Counts_17_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R64_R1.counts.txt")
  #View(Counts_17_FLUO_24)
  
#17-3 DMSO_24 -> R65
  Counts_17_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R65_R1.counts.txt")
  #View(Counts_17_DMSO_24)
  
#17-3 DOX_24+24-> R66
  Counts_17_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R66_R1.counts.txt")
  #View(Counts_17_DOX_24rec)
  
#17-3 FLUO_24+24-> R67
  Counts_17_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R67_R1.counts.txt")
  #View(Counts_17_FLUO_24rec)
  
#17-3 DMSO_24+24-> R68
  Counts_17_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R68_R1.counts.txt")
  #View(Counts_17_DMSO_24rec)
  
#17-3 DOX_24+144 -> R69
  Counts_17_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R69_R1.counts.txt")
  #View(Counts_17_DOX_144rec)

#17-3 FLUO_24+144 -> R70
   Counts_17_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R70_R1.counts.txt")
  #View(Counts_17_FLUO_144rec)
  
#17-3 DMSO_24+144 -> R71
   Counts_17_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R71_R1.counts.txt")
  #View(Counts_17_DMSO_144rec)
   
```


Individual 6 - 90-1 (9 samples)
```{r Import Counts Individual 6 90-1, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#90-1 DOX_24-> R72
  Counts_90_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R72_R1.counts.txt")  
  #View(Counts_90_DOX_24)

#90-1 FLUO_24 -> R73
  Counts_90_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R73_R1.counts.txt")
  #View(Counts_90_FLUO_24)
  
#90-1 DMSO_24 -> R74
  Counts_90_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R74_R1.counts.txt")
  #View(Counts_90_DMSO_24)
  
#90-1 DOX_24+24-> R75
  Counts_90_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R75_R1.counts.txt")
  #View(Counts_90_DOX_24rec)
  
#90-1 FLUO_24+24-> R76
  Counts_90_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R76_R1.counts.txt")
  #View(Counts_90_FLUO_24rec)
  
#90-1 DMSO_24+24-> R77
  Counts_90_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R77_R1.counts.txt")
  #View(Counts_90_DMSO_24rec)
  
#90-1 DOX_24+144 -> R78
  Counts_90_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R78_R1.counts.txt")
  #View(Counts_90_DOX_144rec)

#90-1 FLUO_24+144 -> R79
   Counts_90_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R79_R1.counts.txt")
  #View(Counts_90_FLUO_144rec)
  
#90-1 DMSO_24+144 -> R80
   Counts_90_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R80_R1.counts.txt")
  #View(Counts_90_DMSO_144rec)
   
```


Individual 7 - 90-1 REP (9 samples)
```{r Import Counts Individual 7 90-1 REP, include=FALSE}
#abbreviated to the first two numbers of the cell line for ease
#9 samples per cell line

#90-1 REP DOX_24-> R81
  Counts_90REP_DOX_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R81_R1.counts.txt")  
  #View(Counts_90_DOX_24)

#90-1 REP FLUO_24 -> R82
  Counts_90REP_FLUO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R82_R1.counts.txt")
  #View(Counts_90_FLUO_24)
  
#90-1 REP DMSO_24 -> R83
  Counts_90REP_DMSO_24 <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R83_R1.counts.txt")
  #View(Counts_90_DMSO_24)
  
#90-1 REP DOX_24+24-> R84
  Counts_90REP_DOX_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R84_R1.counts.txt")
  #View(Counts_90_DOX_24rec)
  
#90-1 REP FLUO_24+24-> R85
  Counts_90REP_FLUO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R85_R1.counts.txt")
  #View(Counts_90_FLUO_24rec)
  
#90-1 REP DMSO_24+24-> R86
  Counts_90REP_DMSO_24rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R86_R1.counts.txt")
  #View(Counts_90_DMSO_24rec)
  
#90-1 REP DOX_24+144 -> R87
  Counts_90REP_DOX_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R87_R1.counts.txt")
  #View(Counts_90_DOX_144rec)

#90-1 REP FLUO_24+144 -> R88
   Counts_90REP_FLUO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R88_R1.counts.txt")
  #View(Counts_90_FLUO_144rec)
  
#90-1 REP DMSO_24+144 -> R89
   Counts_90REP_DMSO_144rec <- read.delim("C:/Users/emmap/OneDrive/Desktop/Ward Lab/Experiments/5FU Project/RNAseq/new counts/MCW_EMP_JT_R89_R1.counts.txt")
  #View(Counts_90_DMSO_144rec)
   
```
Now that I have imported all of these counts files, I will make a matrix with all of the samples

First I'll break them up by individual, and then combine them all together
```{r Counts Matrix 84-1, include=FALSE}

fC_Matrix_84_All <-
  data.frame(
    Counts_84_DOX_24,
    Counts_84_FLUO_24$MCW_EMP_JT_R28_R1.bam,
    Counts_84_DMSO_24$MCW_EMP_JT_R29_R1.bam,
    Counts_84_DOX_24rec$MCW_EMP_JT_R30_R1.bam,
    Counts_84_FLUO_24rec$MCW_EMP_JT_R31_R1.bam,
    Counts_84_DMSO_24rec$MCW_EMP_JT_R32_R1.bam,
    Counts_84_DOX_144rec$MCW_EMP_JT_R33_R1.bam,
    Counts_84_FLUO_144rec$MCW_EMP_JT_R34_R1.bam,
    Counts_84_DMSO_144rec$MCW_EMP_JT_R35_R1.bam
  )
colnames(fC_Matrix_84_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind1",
                                "FLUO_24_Ind1",
                                "DMSO_24_Ind1",
                                "DOX_24rec_Ind1",
                                "FLUO_24rec_Ind1",
                                "DMSO_24rec_Ind1",
                                "DOX_144rec_Ind1",
                                "FLUO_144rec_Ind1",
                                "DMSO_144rec_Ind1")

fC_Matrix_84 <- fC_Matrix_84_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
#write.csv(fC_Matrix_84_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_84-1_EMP_250210.csv")

boxplot(fC_Matrix_84,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 84-1")

hist(fC_Matrix_84,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 84-1")


```

```{r Counts Matrix Individual 2 87-1, include=FALSE}

fC_Matrix_87_All <-
  data.frame(
    Counts_87_DOX_24,
    Counts_87_FLUO_24$MCW_EMP_JT_R37_R1.bam,
    Counts_87_DMSO_24$MCW_EMP_JT_R38_R1.bam,
    Counts_87_DOX_24rec$MCW_EMP_JT_R39_R1.bam,
    Counts_87_FLUO_24rec$MCW_EMP_JT_R40_R1.bam,
    Counts_87_DMSO_24rec$MCW_EMP_JT_R41_R1.bam,
    Counts_87_DOX_144rec$MCW_EMP_JT_R42_R1.bam,
    Counts_87_FLUO_144rec$MCW_EMP_JT_R43_R1.bam,
    Counts_87_DMSO_144rec$MCW_EMP_JT_R44_R1.bam
  )
colnames(fC_Matrix_87_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind2",
                                "FLUO_24_Ind2",
                                "DMSO_24_Ind2",
                                "DOX_24rec_Ind2",
                                "FLUO_24rec_Ind2",
                                "DMSO_24rec_Ind2",
                                "DOX_144rec_Ind2",
                                "FLUO_144rec_Ind2",
                                "DMSO_144rec_Ind2")

fC_Matrix_87 <- fC_Matrix_87_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_87_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_87-1_EMP_250210.csv")

boxplot(fC_Matrix_87,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 87-1")

hist(fC_Matrix_87,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 87-1")


```


```{r Counts Matrix Individual 3 78-1, include=FALSE}

fC_Matrix_78_All <-
  data.frame(
    Counts_78_DOX_24,
    Counts_78_FLUO_24$MCW_EMP_JT_R46_R1.bam,
    Counts_78_DMSO_24$MCW_EMP_JT_R47_R1.bam,
    Counts_78_DOX_24rec$MCW_EMP_JT_R48_R1.bam,
    Counts_78_FLUO_24rec$MCW_EMP_JT_R49_R1.bam,
    Counts_78_DMSO_24rec$MCW_EMP_JT_R50_R1.bam,
    Counts_78_DOX_144rec$MCW_EMP_JT_R51_R1.bam,
    Counts_78_FLUO_144rec$MCW_EMP_JT_R52_R1.bam,
    Counts_78_DMSO_144rec$MCW_EMP_JT_R53_R1.bam
  )
colnames(fC_Matrix_78_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind3",
                                "FLUO_24_Ind3",
                                "DMSO_24_Ind3",
                                "DOX_24rec_Ind3",
                                "FLUO_24rec_Ind3",
                                "DMSO_24rec_Ind3",
                                "DOX_144rec_Ind3",
                                "FLUO_144rec_Ind3",
                                "DMSO_144rec_Ind3")

fC_Matrix_78 <- fC_Matrix_78_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_78_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_78-1_EMP_250210.csv")

boxplot(fC_Matrix_78,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 78-1")

hist(fC_Matrix_78,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 78-1")


```


```{r Counts Matrix Individual 4 75-1, include=FALSE}

fC_Matrix_75_All <-
  data.frame(
    Counts_75_DOX_24,
    Counts_75_FLUO_24$MCW_EMP_JT_R55_R1.bam,
    Counts_75_DMSO_24$MCW_EMP_JT_R56_R1.bam,
    Counts_75_DOX_24rec$MCW_EMP_JT_R57_R1.bam,
    Counts_75_FLUO_24rec$MCW_EMP_JT_R58_R1.bam,
    Counts_75_DMSO_24rec$MCW_EMP_JT_R59_R1.bam,
    Counts_75_DOX_144rec$MCW_EMP_JT_R60_R1.bam,
    Counts_75_FLUO_144rec$MCW_EMP_JT_R61_R1.bam,
    Counts_75_DMSO_144rec$MCW_EMP_JT_R62_R1.bam
  )
colnames(fC_Matrix_75_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind4",
                                "FLUO_24_Ind4",
                                "DMSO_24_Ind4",
                                "DOX_24rec_Ind4",
                                "FLUO_24rec_Ind4",
                                "DMSO_24rec_Ind4",
                                "DOX_144rec_Ind4",
                                "FLUO_144rec_Ind4",
                                "DMSO_144rec_Ind4")

fC_Matrix_75 <- fC_Matrix_75_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_75_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_75-1_EMP_250210.csv")

boxplot(fC_Matrix_75,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 75-1")

hist(fC_Matrix_75,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 75-1")


```


```{r Counts Matrix Individual 5 17-3, include=FALSE}

fC_Matrix_17_All <-
  data.frame(
    Counts_17_DOX_24,
    Counts_17_FLUO_24$MCW_EMP_JT_R64_R1.bam,
    Counts_17_DMSO_24$MCW_EMP_JT_R65_R1.bam,
    Counts_17_DOX_24rec$MCW_EMP_JT_R66_R1.bam,
    Counts_17_FLUO_24rec$MCW_EMP_JT_R67_R1.bam,
    Counts_17_DMSO_24rec$MCW_EMP_JT_R68_R1.bam,
    Counts_17_DOX_144rec$MCW_EMP_JT_R69_R1.bam,
    Counts_17_FLUO_144rec$MCW_EMP_JT_R70_R1.bam,
    Counts_17_DMSO_144rec$MCW_EMP_JT_R71_R1.bam
  )
colnames(fC_Matrix_17_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind5",
                                "FLUO_24_Ind5",
                                "DMSO_24_Ind5",
                                "DOX_24rec_Ind5",
                                "FLUO_24rec_Ind5",
                                "DMSO_24rec_Ind5",
                                "DOX_144rec_Ind5",
                                "FLUO_144rec_Ind5",
                                "DMSO_144rec_Ind5")

fC_Matrix_17 <- fC_Matrix_17_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_17_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_17-1_EMP_250210.csv")

boxplot(fC_Matrix_17,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 17-3")

hist(fC_Matrix_17,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 17-3")


```


```{r Counts Matrix Individual 6 90-1, include=FALSE}

fC_Matrix_90_All <-
  data.frame(
    Counts_90_DOX_24,
    Counts_90_FLUO_24$MCW_EMP_JT_R73_R1.bam,
    Counts_90_DMSO_24$MCW_EMP_JT_R74_R1.bam,
    Counts_90_DOX_24rec$MCW_EMP_JT_R75_R1.bam,
    Counts_90_FLUO_24rec$MCW_EMP_JT_R76_R1.bam,
    Counts_90_DMSO_24rec$MCW_EMP_JT_R77_R1.bam,
    Counts_90_DOX_144rec$MCW_EMP_JT_R78_R1.bam,
    Counts_90_FLUO_144rec$MCW_EMP_JT_R79_R1.bam,
    Counts_90_DMSO_144rec$MCW_EMP_JT_R80_R1.bam
  )
colnames(fC_Matrix_90_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind6",
                                "FLUO_24_Ind6",
                                "DMSO_24_Ind6",
                                "DOX_24rec_Ind6",
                                "FLUO_24rec_Ind6",
                                "DMSO_24rec_Ind6",
                                "DOX_144rec_Ind6",
                                "FLUO_144rec_Ind6",
                                "DMSO_144rec_Ind6")

fC_Matrix_90 <- fC_Matrix_90_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_90_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_90-1_EMP_250210.csv")

boxplot(fC_Matrix_90,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 90-1")

hist(fC_Matrix_90,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 90-1")


```


```{r Counts Matrix Individual 6 90-1REP, include=FALSE}

fC_Matrix_90REP_All <-
  data.frame(
    Counts_90REP_DOX_24,
    Counts_90REP_FLUO_24$MCW_EMP_JT_R82_R1.bam,
    Counts_90REP_DMSO_24$MCW_EMP_JT_R83_R1.bam,
    Counts_90REP_DOX_24rec$MCW_EMP_JT_R84_R1.bam,
    Counts_90REP_FLUO_24rec$MCW_EMP_JT_R85_R1.bam,
    Counts_90REP_DMSO_24rec$MCW_EMP_JT_R86_R1.bam,
    Counts_90REP_DOX_144rec$MCW_EMP_JT_R87_R1.bam,
    Counts_90REP_FLUO_144rec$MCW_EMP_JT_R88_R1.bam,
    Counts_90REP_DMSO_144rec$MCW_EMP_JT_R89_R1.bam
  )
colnames(fC_Matrix_90REP_All) <- c("ensembl_gene_id", 
                                "DOX_24_Ind6REP",
                                "FLUO_24_Ind6REP",
                                "DMSO_24_Ind6REP",
                                "DOX_24rec_Ind6REP",
                                "FLUO_24rec_Ind6REP",
                                "DMSO_24rec_Ind6REP",
                                "DOX_144rec_Ind6REP",
                                "FLUO_144rec_Ind6REP",
                                "DMSO_144rec_Ind6REP")

fC_Matrix_90REP <- fC_Matrix_90REP_All %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
# write.csv(fC_Matrix_90REP_All, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_90-1REP_EMP_250210.csv")

boxplot(fC_Matrix_90REP,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 100), main = "Boxplot of Raw Counts 90-1 REP")

hist(fC_Matrix_90REP,  xlab = "Unfiltered Counts", xlim = c(-1000,25000), ylim = c(-100, 20000), main = "Histogram of Raw Counts 90-1 REP")


```


Now that I've made individual matrices for each individual - I will combine them together into one large matrix

```{r Counts Matrix All Individuals, results='hide', echo=FALSE}

fC_Matrix_AllInd <-
  data.frame(
    Counts_84_DOX_24,
    Counts_84_FLUO_24$MCW_EMP_JT_R28_R1.bam,
    Counts_84_DMSO_24$MCW_EMP_JT_R29_R1.bam,
    Counts_84_DOX_24rec$MCW_EMP_JT_R30_R1.bam,
    Counts_84_FLUO_24rec$MCW_EMP_JT_R31_R1.bam,
    Counts_84_DMSO_24rec$MCW_EMP_JT_R32_R1.bam,
    Counts_84_DOX_144rec$MCW_EMP_JT_R33_R1.bam,
    Counts_84_FLUO_144rec$MCW_EMP_JT_R34_R1.bam,
    Counts_84_DMSO_144rec$MCW_EMP_JT_R35_R1.bam,
    Counts_87_DOX_24$MCW_EMP_JT_R36_R1.bam,
    Counts_87_FLUO_24$MCW_EMP_JT_R37_R1.bam,
    Counts_87_DMSO_24$MCW_EMP_JT_R38_R1.bam,
    Counts_87_DOX_24rec$MCW_EMP_JT_R39_R1.bam,
    Counts_87_FLUO_24rec$MCW_EMP_JT_R40_R1.bam,
    Counts_87_DMSO_24rec$MCW_EMP_JT_R41_R1.bam,
    Counts_87_DOX_144rec$MCW_EMP_JT_R42_R1.bam,
    Counts_87_FLUO_144rec$MCW_EMP_JT_R43_R1.bam,
    Counts_87_DMSO_144rec$MCW_EMP_JT_R44_R1.bam,
    Counts_78_DOX_24$MCW_EMP_JT_R45_R1.bam,
    Counts_78_FLUO_24$MCW_EMP_JT_R46_R1.bam,
    Counts_78_DMSO_24$MCW_EMP_JT_R47_R1.bam,
    Counts_78_DOX_24rec$MCW_EMP_JT_R48_R1.bam,
    Counts_78_FLUO_24rec$MCW_EMP_JT_R49_R1.bam,
    Counts_78_DMSO_24rec$MCW_EMP_JT_R50_R1.bam,
    Counts_78_DOX_144rec$MCW_EMP_JT_R51_R1.bam,
    Counts_78_FLUO_144rec$MCW_EMP_JT_R52_R1.bam,
    Counts_78_DMSO_144rec$MCW_EMP_JT_R53_R1.bam,
    Counts_75_DOX_24$MCW_EMP_JT_R54_R1.bam,
    Counts_75_FLUO_24$MCW_EMP_JT_R55_R1.bam,
    Counts_75_DMSO_24$MCW_EMP_JT_R56_R1.bam,
    Counts_75_DOX_24rec$MCW_EMP_JT_R57_R1.bam,
    Counts_75_FLUO_24rec$MCW_EMP_JT_R58_R1.bam,
    Counts_75_DMSO_24rec$MCW_EMP_JT_R59_R1.bam,
    Counts_75_DOX_144rec$MCW_EMP_JT_R60_R1.bam,
    Counts_75_FLUO_144rec$MCW_EMP_JT_R61_R1.bam,
    Counts_75_DMSO_144rec$MCW_EMP_JT_R62_R1.bam,
    Counts_17_DOX_24$MCW_EMP_JT_R63_R1.bam,
    Counts_17_FLUO_24$MCW_EMP_JT_R64_R1.bam,
    Counts_17_DMSO_24$MCW_EMP_JT_R65_R1.bam,
    Counts_17_DOX_24rec$MCW_EMP_JT_R66_R1.bam,
    Counts_17_FLUO_24rec$MCW_EMP_JT_R67_R1.bam,
    Counts_17_DMSO_24rec$MCW_EMP_JT_R68_R1.bam,
    Counts_17_DOX_144rec$MCW_EMP_JT_R69_R1.bam,
    Counts_17_FLUO_144rec$MCW_EMP_JT_R70_R1.bam,
    Counts_17_DMSO_144rec$MCW_EMP_JT_R71_R1.bam,
    Counts_90_DOX_24$MCW_EMP_JT_R72_R1.bam,
    Counts_90_FLUO_24$MCW_EMP_JT_R73_R1.bam,
    Counts_90_DMSO_24$MCW_EMP_JT_R74_R1.bam,
    Counts_90_DOX_24rec$MCW_EMP_JT_R75_R1.bam,
    Counts_90_FLUO_24rec$MCW_EMP_JT_R76_R1.bam,
    Counts_90_DMSO_24rec$MCW_EMP_JT_R77_R1.bam,
    Counts_90_DOX_144rec$MCW_EMP_JT_R78_R1.bam,
    Counts_90_FLUO_144rec$MCW_EMP_JT_R79_R1.bam,
    Counts_90_DMSO_144rec$MCW_EMP_JT_R80_R1.bam,
    Counts_90REP_DOX_24$MCW_EMP_JT_R81_R1.bam,
    Counts_90REP_FLUO_24$MCW_EMP_JT_R82_R1.bam,
    Counts_90REP_DMSO_24$MCW_EMP_JT_R83_R1.bam,
    Counts_90REP_DOX_24rec$MCW_EMP_JT_R84_R1.bam,
    Counts_90REP_FLUO_24rec$MCW_EMP_JT_R85_R1.bam,
    Counts_90REP_DMSO_24rec$MCW_EMP_JT_R86_R1.bam,
    Counts_90REP_DOX_144rec$MCW_EMP_JT_R87_R1.bam,
    Counts_90REP_FLUO_144rec$MCW_EMP_JT_R88_R1.bam,
    Counts_90REP_DMSO_144rec$MCW_EMP_JT_R89_R1.bam
  )

colnames(fC_Matrix_AllInd) <- c("ensembl_gene_id", 
                                "DOX_24_Ind1",
                                "FLUO_24_Ind1",
                                "DMSO_24_Ind1",
                                "DOX_24rec_Ind1",
                                "FLUO_24rec_Ind1",
                                "DMSO_24rec_Ind1",
                                "DOX_144rec_Ind1",
                                "FLUO_144rec_Ind1",
                                "DMSO_144rec_Ind1",
                                "DOX_24_Ind2",
                                "FLUO_24_Ind2",
                                "DMSO_24_Ind2",
                                "DOX_24rec_Ind2",
                                "FLUO_24rec_Ind2",
                                "DMSO_24rec_Ind2",
                                "DOX_144rec_Ind2",
                                "FLUO_144rec_Ind2",
                                "DMSO_144rec_Ind2",
                                "DOX_24_Ind3",
                                "FLUO_24_Ind3",
                                "DMSO_24_Ind3",
                                "DOX_24rec_Ind3",
                                "FLUO_24rec_Ind3",
                                "DMSO_24rec_Ind3",
                                "DOX_144rec_Ind3",
                                "FLUO_144rec_Ind3",
                                "DMSO_144rec_Ind3",
                                "DOX_24_Ind4",
                                "FLUO_24_Ind4",
                                "DMSO_24_Ind4",
                                "DOX_24rec_Ind4",
                                "FLUO_24rec_Ind4",
                                "DMSO_24rec_Ind4",
                                "DOX_144rec_Ind4",
                                "FLUO_144rec_Ind4",
                                "DMSO_144rec_Ind4",
                                "DOX_24_Ind5",
                                "FLUO_24_Ind5",
                                "DMSO_24_Ind5",
                                "DOX_24rec_Ind5",
                                "FLUO_24rec_Ind5",
                                "DMSO_24rec_Ind5",
                                "DOX_144rec_Ind5",
                                "FLUO_144rec_Ind5",
                                "DMSO_144rec_Ind5",
                                "DOX_24_Ind6",
                                "FLUO_24_Ind6",
                                "DMSO_24_Ind6",
                                "DOX_24rec_Ind6",
                                "FLUO_24rec_Ind6",
                                "DMSO_24rec_Ind6",
                                "DOX_144rec_Ind6",
                                "FLUO_144rec_Ind6",
                                "DMSO_144rec_Ind6",
                                "DOX_24_Ind6REP",
                                "FLUO_24_Ind6REP",
                                "DMSO_24_Ind6REP",
                                "DOX_24rec_Ind6REP",
                                "FLUO_24rec_Ind6REP",
                                "DMSO_24rec_Ind6REP",
                                "DOX_144rec_Ind6REP",
                                "FLUO_144rec_Ind6REP",
                                "DMSO_144rec_Ind6REP")

fC_Matrix_Full <- fC_Matrix_AllInd %>% remove_rownames %>% column_to_rownames(var = "ensembl_gene_id") %>% as.matrix()

#write this one to a csv so I can hold onto it
#write.csv(fC_Matrix_AllInd, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_AllSamples_EMP_250220.csv")

boxplot(fC_Matrix_Full,  xlab = "Conditions", ylab = "Unfiltered Counts", ylim= c(-10, 1000), main = "Boxplot of Raw Counts All Individuals")

hist(fC_Matrix_Full,  xlab = "Unfiltered Counts", xlim = c(-1000,125000), ylim = c(-100, 6000000), main = "Histogram of Raw Counts All Individuals")


```


```{r Data Handling, results='hide', echo=FALSE, collapse=TRUE}

#Raw Counts
fC_Matrix_Full

Counts_Full_df <- data.frame(fC_Matrix_Full)

#This code chunk in tandem with the one below to make sure my data's tucked away!


#define colors
ind_col <- c("#003F5C", "#45AE91",  "#58508D", "#BC4099", "#8B3E9B", "#FF6361", "#FF2362")

tx_col <- c("#499FBD", "#DCACED","#63666D")

time_col <- c("#fbb4b9", "#f768a1", "#ae017e")


#factor my df to get all the factors
annotate_Counts <- data.frame("samples" = colnames(Counts_Full_df)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)
#%>% unite(., col = "Treatment_Time", "Treatment", "Time", sep = "_", remove = FALSE)

#now cbind these factors of my data to my raw counts file, transpose first

annotate_Counts_full <- (t(Counts_Full_df)) %>%  cbind(., annotate_Counts)
 
#now I'll factor by sample to keep order

#Counts_Full_df %>% dplyr::add_row()

annot_Counts_full_df <- as.data.frame(t(annotate_Counts_full))

#this is for the log2cpm transformation below
#####log2cpm####
fC_Matrix_Full_cpm <- cpm(fC_Matrix_Full, log = TRUE)



```



```{r QC checks Raw Counts, fig.height=10, fig.width=12, collapse=TRUE}
#Now that I've put together my matrix - I will do some QC checks

#now that I have all of my factors and my gene info, I will make some graphs to show the QC data

reads_by_sample <- c(tx_col)
fC_AllCounts %>% 
  ggplot(., aes (x = Conditions, y = Total_Align, fill = Treatment, group_by = Line))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Read Counts by Treatment####
fC_AllCounts %>% 
  ggplot(., aes (x =Treatment, y= Total_Align, fill = Treatment))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of reads by treatment"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Reads Per Individual####
fC_AllCounts %>% 
  ggplot(., aes (x =as.factor(Line), y=Total_Align))+
  geom_boxplot(aes(fill=as.factor(Line)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total number of reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Mapped Reads Per Drug####

reads_by_sample <- c(tx_col)
fC_AllCounts %>% 
  ggplot(., aes (x = Conditions, y = Assigned_Align, fill = Treatment, group_by = Line))+
  geom_col()+
 geom_hline(aes(yintercept=20000000))+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by sample"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


####Read Counts by Treatment####
fC_AllCounts %>% 
  ggplot(., aes (x =Treatment, y= Assigned_Align, fill = Treatment))+
  geom_boxplot()+
 scale_fill_manual(values=reads_by_sample)+
  ggtitle(expression("Total number of mapped reads by treatment"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+
  
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))

####Total Reads Per Individual####
fC_AllCounts %>% 
  ggplot(., aes (x =as.factor(Line), y=Assigned_Align))+
  geom_boxplot(aes(fill=as.factor(Line)))+
 scale_fill_brewer(palette = "Dark2", name = "Individual")+
  ggtitle(expression("Total number of mapped reads by individual"))+
  xlab("")+
  ylab(expression("RNA-sequencing reads"))+
  theme_bw()+

  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
        axis.text.x = element_text(size =10, color = "black", angle = 0, hjust = 1),
        #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
        strip.text.y = element_text(color = "white"))


```

Now that I've made the final full matrix of all samples -
1. convert to log2cpm using cpm
2. filter for lowly expressed reads (rowMeans > 0)

```{r log2cpm Conversion Full Matrix}

boxplot(fC_Matrix_Full_cpm, xlab = "Conditions", ylab = expression("log"[2]* "counts per million"), ylim= c(-10, 20))

hist(fC_Matrix_Full_cpm, xlab = expression("log" [2]* "counts per million"))

#write this to a csv just in case to save it
#write.csv(fC_Matrix_Full_cpm, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_AllSamples_cpm_EMP_250210.csv")

```

```{r Filter Lowly Expressed Reads All Samples, collapse=TRUE}

#first check the number of variables in the cpm file
dim(fC_Matrix_Full_cpm)
#[1] 28395    63

#now use rowMeans to filter out values with mean < 0 (this means rowMeans > 0 is what I'm left with) for each row/gene in the original file
rowMeans_All <- rowMeans(fC_Matrix_Full_cpm)
fC_Matrix_Full_cpm_filter <- fC_Matrix_Full_cpm[rowMeans_All > 0,]
dim(fC_Matrix_Full_cpm_filter)
#[1] 14225    63

#with this I've filtered out a significant number of rows when filtering after cpm conversion - does this make it more balanced?

hist(fC_Matrix_Full_cpm_filter, 
     main = "Recovery All Individuals log2cpm Filtered", 
        xlab = expression("log"[2]*"counts per million"))
boxplot(fC_Matrix_Full_cpm_filter, 
        main = "Recovery All Individuals log2cpm Filtered", 
        xlab = "Conditions", 
        ylab = expression("log"[2]*"counts per million"),
        ylim = c(-10,15))

#write this to a csv to save
#write.csv(fC_Matrix_Full_cpm_filter, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_AllSamples_cpm_filtered_EMP_250210.csv")

```


```{r QC Checks Unfiltered vs Filtered Reads}


####histogram of total counts, unfiltered####
hist(fC_Matrix_Full_cpm, main = "Recovery All Individuals log2cpm Unfiltered",
     xlab = expression("log" [2]* "counts per million"))

####histogram of total counts, filtered####
hist(fC_Matrix_Full_cpm_filter, 
     main = "Recovery All Individuals log2cpm Filtered rowMeans > 0", 
        xlab = expression("log"[2]*"counts per million"))

####boxplot log2cpm all samples####

boxplot(fC_Matrix_Full_cpm, xlab = "Conditions", ylab = expression("log"[2]* "counts per million"), ylim= c(-10, 20), main = "Boxplots of log2cpm per sample")

####boxplot log2cpm filtered all samples####

boxplot(fC_Matrix_Full_cpm_filter, xlab = "Conditions", ylab = expression("log"[2]* "counts per million"), ylim= c(-10, 20), main = "Boxplots of log2cpm Filtered rowMeans > 0 per sample")


```


Now that I've filtered the samples after log2cpm conversion, I can start to make heatmaps and PCA plots to look at my data

```{r PCA Plots All Data, include = FALSE}
# #mutate the file so that the samples are rows and columns are genes
# fC_Matrix_Full_cpm_filter_mutate <- data.frame(t(fC_Matrix_Full_cpm_filter))
# 
# #now let's separate out the factors we have in this experiment
# #Factor 1 - Individual
# Individual <- as.factor(c(rep("Ind1", 9), rep("Ind2", 9), rep("Ind3", 9), rep("Ind4", 9), rep("Ind5", 9), rep("Ind6", 9), rep("Ind6REP", 9)))
# 
# #Factor 2 - Treatment
# tx_factor <- c("DOX", "FLUO", "DMSO")
# Treatment <- as.factor(c(rep(tx_factor, 21)))
# #view(Treatment)
# 
# #Factor 3 - Timepoint
# time_factor <- c(rep("24hr", 3), rep("24hr Rec", 3), rep("6d Rec", 3))
# Time <- as.factor(c(rep(time_factor, 7)))
# #view(Time)
# 
# Full_filter_cpm_mutate_categories <- cbind(Individual, Treatment, Time, fC_Matrix_Full_cpm_filter_mutate)
# 
# pca_plot <- prcomp(fC_Matrix_Full_cpm_filter_mutate, scale. = TRUE)
# 
# # autoplot(pca_plot, data = Full_filter_cpm_mutate_categories, color = "Individual", shape = "Treatment", fill = "Time", size = 5, x=1, y=2)
# 
# 
# #now I want to try ggplot as the previous plot isn't showing my timepoints filled
# #can I pull PC1 and PC2 from my object (pca_plot[["rotation"]])?
# 
# #try and mutate it and see what happens with PC1 and PC2
# PCAs <- (pca_plot$rotation)
# #make this into a matrix
# Full_filter_cpm_mutate_categories_mat <- as.matrix(Full_filter_cpm_mutate_categories)
# 
# ggplot(pca_plot, Full_filter_cpm_mutate_categories_mat, aes(x=1,y=2, color=Time, shape=Treatment, fill=Individual))+
#   geom_point(size=3)+
#   scale_shape_manual(values = c(21, 22, 23) )+
#   scale_color_manual(values = c("white", "darkgray", "black") )+
#   scale_fill_manual(values = c("#003f5c", "#374c80", "#7A5195", "#BC5090", "#EF5675", "#FF764A", "#FFA600"))+
#   xlab("PC1")+
#   ylab("PC2")+
#   guides(fill=guide_legend(override.aes = list(shape=c(21,22, 23)), fill=c("#003f5c", "#374c80", "#7A5195", "#BC5090", "#EF5675", "#FF764A", "#FFA600"), color=c("white", "darkgray", "black")))
#   
#   
# 
# autoplot(pca_plot, data = Full_filter_cpm_mutate_categories, aes(x=1, y=2), color = Time, shape = Treatment, fill = Individual,
#          scale_shape_manual(values = c(21, 22, 23) ),
#          scale_color_manual(values = c("white", "darkgrey", "black") ),
#          scale_fill_manual(values = c("#003f5c", "#374c80", "#7A5195", "#BC5090", "#EF5675", "#FF764A", "#FFA600")),
#          guides(fill=guide_legend(override.aes = list(shape=c(21,22,23), fill=c("#003f5c", "#374c80", "#7A5195", "#BC5090", "#EF5675", "#FF764A", "#FFA600"), color=c("white", "darkgray", "black")))))
# 
# 
# autoplot(pca_plot, data = Full_filter_cpm_mutate_categories, color = Time, shape = Treatment, fill = Individual)+
#   scale_color_manual(values = c("white", "darkgray", "black"), aesthetics = "color")+
#   scale_fill_manual(values = c("#003f5c", "#374c80", "#7A5195", "#BC5090", "#EF5675", "#FF764A", "#FFA600"), aesthetics = "fill")+
#   scale_shape_manual(values = c(25,22,23))

```

```{r PCA Plots ggplot2 SIMPLIFIED, fig.width=6, fig.height=6, include = FALSE}

# PCA_data <- fC_Matrix_Full_cpm_filter %>% 
#   prcomp(.) %>% 
#   t()
# 
# PCA_data_test <- (prcomp(t(fC_Matrix_Full_cpm_filter), scale. = TRUE))
# 
# #make a function for pca - pulls stdv and makes a variance plot - look at the variance contributed by each PC
# pca_var_plot <- function(pca) {
#   pca.var <- pca$sdev ^ 2
#   pca.prop <- pca.var / sum(pca.var)
#   var.plot <- 
#     qplot(PC, prop, data = data.frame(PC = 1:length(pca.prop),
#                                       prop = pca.prop)) +
#     labs(title="Variance contributed by each PC",
#          x = "PC", y = "Proportion of Variance")
#   plot(var.plot)
# }
# 
# pca_var_plot(PCA_data_test)
# 
# 
# #make a pca plot function
# # pca_plot_fun <- function(df, 
# #                          col_var = NULL,
# #                          shape_var = NULL,
# #                          title = "") {
# #   ggplot(df) + geom_point(aes_string(
# #     x = "PC1",
# #     y = "PC2",
# #     color = col_var,
# #     shape = shape_var),
# #     size = 5) +
# #     labs(title = title, x = "PC 1", y = "PC 2") +
# #     scale_color_manual(values = c("#003f5c", "#374c80", "#7A5195", "#BC5090", "#EF5675", "#FF764A", "#FFA600")) +
# #                        scale_shape_manual(values = c(0, 1, 2, 5, 6, 15, 16, 17, 18))
# #   
# # }
# 
# colorsInd <- c("#003f5c", "#374c80", "#7A5195", "#BC5090", "#EF5675", "#FF764A", "#FFA600")
# 
# #now we can go into plotting!
# ####make an annotation matrix####
# annot <- data.frame("samples" = colnames(fC_Matrix_Full_cpm_filter)) %>% separate_wider_delim(., cols = samples, names = c("Tx", "Time", "Ind"), delim = "_", cols_remove = FALSE) %>% unite(., col = "Tx_Time", Tx, Time, sep = "_", remove = FALSE)
# 
# #combine the prcomp matrix and annotation
# annot_PCA_matrix <- PCA_data_test$x %>% cbind(., annot)
# 
# #now you can go ahead and make PCA plots with this matrix! woohoo!
# pca_plot_fun(annot_PCA_matrix, col_var = "Ind", shape_var = "Tx_Time")
# 
# annot_PCA_matrix %>% ggplot(., aes(x=PC1, y=PC2, size=6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#   # scale_color_manual(values = c("#003f5c", "#374c80", "#7A5195", "#BC5090", "#EF5675", "#FF764A", "#FFA600")) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
#   
# annot_PCA_matrix %>% ggplot(., aes(x=PC2, y=PC3, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# annot_PCA_matrix %>% ggplot(., aes(x=PC3, y=PC4, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# #Now that I've successfully generated ggplot PCA plots with Renee's code - now I can subset the data by time
# 
# time_24 <- annot_PCA_matrix %>% filter(Time=="24")
# time_24rec <- annot_PCA_matrix %>% filter(Time=="24rec")
# time_144rec <- annot_PCA_matrix %>% filter(Time=="144rec")
# 
# ####time 24hr####
# time_24 %>% ggplot(., aes(x=PC1, y=PC2, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# time_24 %>% ggplot(., aes(x=PC2, y=PC3, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# time_24 %>% ggplot(., aes(x=PC3, y=PC4, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# ####time 24rec####
# time_24rec %>% ggplot(., aes(x=PC1, y=PC2, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# time_24rec %>% ggplot(., aes(x=PC2, y=PC3, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# time_24rec %>% ggplot(., aes(x=PC3, y=PC4, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# ####time 144rec####
# time_144rec %>% ggplot(., aes(x=PC1, y=PC2, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# time_144rec %>% ggplot(., aes(x=PC2, y=PC3, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# time_144rec %>% ggplot(., aes(x=PC3, y=PC4, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# #####now that I've broken them up by time, try by treatment####
# 
# dox_rec <- annot_PCA_matrix %>% filter(Tx=="DOX")
# fluo_rec <- annot_PCA_matrix %>% filter(Tx=="FLUO")
# dmso_rec <- annot_PCA_matrix %>% filter(Tx=="DMSO")
# 
# ####DOX####
# dox_rec %>% ggplot(., aes(x=PC1, y=PC2, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# dox_rec %>% ggplot(., aes(x=PC2, y=PC3, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# dox_rec %>% ggplot(., aes(x=PC3, y=PC4, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# ####5FU####
# fluo_rec %>% ggplot(., aes(x=PC1, y=PC2, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# fluo_rec %>% ggplot(., aes(x=PC2, y=PC3, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# fluo_rec %>% ggplot(., aes(x=PC3, y=PC4, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# ####DMSO VEHICLE####
# dmso_rec %>% ggplot(., aes(x=PC1, y=PC2, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# dmso_rec %>% ggplot(., aes(x=PC2, y=PC3, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# dmso_rec %>% ggplot(., aes(x=PC3, y=PC4, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# #####Now I want to combine them with vehicle to see any effect
# dox_veh_rec <- annot_PCA_matrix %>% filter(Tx!="FLUO")
# fluo_veh_rec <- annot_PCA_matrix %>% filter(Tx!="DOX")
# 
# ####DOX + DMSO VEHICLE####
# dox_veh_rec %>% ggplot(., aes(x=PC1, y=PC2, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# dox_veh_rec %>% ggplot(., aes(x=PC2, y=PC3, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# dox_veh_rec %>% ggplot(., aes(x=PC3, y=PC4, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# 
# ####5FU + DMSO VEHICLE####
# fluo_veh_rec %>% ggplot(., aes(x=PC1, y=PC2, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# fluo_veh_rec %>% ggplot(., aes(x=PC2, y=PC3, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")
# 
# fluo_veh_rec %>% ggplot(., aes(x=PC3, y=PC4, size =6)) +
#   geom_point(aes(color = Tx_Time, shape = Ind)) +
#                        scale_shape_manual(values = c(15, 16, 17, 18, 19, 20, 21)) +
#   scale_color_brewer(palette = "Paired")

```


After trying and failing to get all three factors on one PCA plot - I will make iterations of each of these with two factors each

```{r PCA Individual Treatment_Time, include = FALSE}

# #mutate the file so that the samples are rows and columns are genes
# fC_Matrix_Full_cpm_filter_mutate_tx_time <- data.frame(t(fC_Matrix_Full_cpm_filter))
# 
# #now let's separate out the factors we have in this experiment
# #Factor 1 - Individual
# Individual <- as.factor(c(rep("Ind1", 9), rep("Ind2", 9), rep("Ind3", 9), rep("Ind4", 9), rep("Ind5", 9), rep("Ind6", 9), rep("Ind6REP", 9)))
# 
# #Factor 2 - Treatment_Time
# tx_time_factor <- c("DOX_24", "FLUO_24", "DMSO_24", "DOX_24rec", "FLUO_24rec", "DMSO_24rec", "DOX_6drec", "FLUO_6drec", "DMSO_6drec")
# Treatment_Time <- as.factor(c(rep(tx_time_factor, 7)))
# #view(Treatment)
# 
# Full_filter_cpm_mutate_categories_txtime <- cbind(Individual, Treatment_Time, fC_Matrix_Full_cpm_filter_mutate_tx_time)
# 
# pca_plot_txtime <- prcomp(fC_Matrix_Full_cpm_filter_mutate_tx_time, scale. = TRUE)
# 
# autoplot(pca_plot_txtime, data = Full_filter_cpm_mutate_categories_txtime, color = "Treatment_Time", shape = "Individual", size = 5, x=1, y=2)
# 
# #when I try to use scale_shape_manual under ggplot - it gives the following error:
# ##Error in scale != 0 : 
#   ##comparison (!=) is possible only for atomic and list types
# #maybe try converting to an atomic vector or list?
# 
# shapes <- c(0, 1, 2, 5, 6, 15, 16, 17, 18)

#make a pca plot function
# pca_plot_fun_fill <- function(df,
#                          col_var = NULL,
#                          shape_var = NULL, 
#                          fill_var = NULL,
#                          title = "") {
#   ggplot(df) + geom_point(aes_string(
#     x = "PC1",
#     y = "PC2",
#     color = col_var,
#     shape = shape_var,
#     fill = fill_var,
#     size = 5)) +
#     labs(title = title) +
#     scale_color_manual(values = c("red", "blue", "darkgray")) +
#                        scale_shape_manual(values = c(0, 1, 2, 5, 6, 15, 16, 17, 18)) +
#     scale_fill_manual(values = c("#003f5c", "#374c80", "#7A5195", "#BC5090", "#EF5675", "#FF764A", "#FFA600"))
#   
# }


####From the below
#pca_plot_fun_fill(annot_PCA_matrix, col_var = "Tx", shape_var = "Time", fill_var = "Ind")

####PC1/PC2####
# annot_PCA_matrix %>% ggplot(., aes(x=PC1, y=PC2)) +
#   geom_point(aes(color = Ind, shape = Time, fill = Tx, size = 10)) +
#                        scale_shape_manual(values = c(21, 22, 24)) +
#   scale_fill_manual(values =  tx_col)+
#   scale_color_manual(values = fill_col_ind_dark)+
#   guides(fill=guide_legend(override.aes=list(shape=25)))+
#   guides(color=guide_legend(override.aes=list(shape=25)))+
#   guides(fill=guide_legend(override.aes=list(shape=25,fill=tx_col,color=fill_col_ind_dark)))
# 
# #success! i finally got the legend to represent the colors and shapes
# #now let's make two more for PC2/PC3 and PC3/PC4
# 
# ####PC2/PC3####
# annot_PCA_matrix %>% ggplot(., aes(x=PC2, y=PC3)) +
#   geom_point(aes(color = Tx, shape = Time, fill = Ind, size = 10)) +
#                        scale_shape_manual(values = c(21, 22, 24)) +
#   scale_fill_manual(values =  fill_col_names)+
#   scale_color_manual(values = c("black", "red", "blue"))+
#   guides(fill=guide_legend(override.aes=list(shape=21)))+
#   guides(color=guide_legend(override.aes=list(shape=21)))+
#   guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col,color=fill_col)))
# 
# ####PC3/PC4####
# annot_PCA_matrix %>% ggplot(., aes(x=PC3, y=PC4)) +
#   geom_point(aes(color = Tx, shape = Time, fill = Ind, size = 10)) +
#                        scale_shape_manual(values = c(21, 22, 24)) +
#   scale_fill_manual(values =  fill_col_names)+
#   scale_color_manual(values = c("black", "red", "blue"))+
#   guides(fill=guide_legend(override.aes=list(shape=21)))+
#   guides(color=guide_legend(override.aes=list(shape=21)))+
#   guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col,color=fill_col)))


```


Now, I want to update these PCA plots to have treatment and time as the primary variables
Individual will be a fill? Or somehow get some numbers labelled on the points.
```{r PCA Plots Updated, fig.width=8, fig.height=8, collapse=TRUE}
PCA_data <- fC_Matrix_Full_cpm_filter %>% 
  prcomp(.) %>% 
  t()

PCA_data_test <- (prcomp(t(fC_Matrix_Full_cpm_filter), scale. = TRUE))

####now make an annotation for my PCA####
annot <- data.frame("samples" = colnames(fC_Matrix_Full_cpm_filter)) %>% separate_wider_delim(., cols = samples, names = c("Tx", "Time", "Ind"), delim = "_", cols_remove = FALSE) %>% unite(., col = "Tx_Time", Tx, Time, sep = "_", remove = FALSE)

#combine the prcomp matrix and annotation
annot_PCA_matrix <- PCA_data_test$x %>% cbind(., annot)

#now I can make a graph where I have filled values for individual! I have seven colors for seven individuals
# I have three fill values for three timepoints

#using annotation matrix above as well as annotated PCA matrix (annot_PCA_matrix)

#extra info for colors in the graph (fill parameter)
fill_col_ind <- c("#66C2A5", "#FC8D62", "#1F78B4", "#E78AC3", "#A6D854", "#FFD92A", "#8B3E9B")

fill_col_ind_dark <- c("#003F5C", "#45AE91",  "#58508D", "#BC4099", "#8B3E9B", "#FF6361", "#FF2362")

fill_col_tx <- c("#499FBD", "#DCACED", "#63666D")

#Now I have to switch the fill and color parameters as it's doing the opposite of what I wanted: need individuals to be the outline color (color), and tx to be the inside color (fill)

####PC1/PC2####
annot_PCA_matrix %>% ggplot(., aes(x=PC1, y=PC2, size = 10)) +
  geom_point(aes(color = Ind, shape = Time, fill = Tx)) +
                       scale_shape_manual(values = c(21, 22, 24)) +
  scale_fill_manual(values =  fill_col_tx)+
  scale_color_manual(values = c(fill_col_ind_dark))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  guides(color=guide_legend(override.aes=list(shape=21)))+
  guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))

####PC2/PC3####
annot_PCA_matrix %>% ggplot(., aes(x=PC2, y=PC3)) +
  geom_point(aes(color = Ind, shape = Time, fill = Tx, size = 10)) +
                       scale_shape_manual(values = c(21, 22, 24)) +
  scale_fill_manual(values =  fill_col_tx)+
  scale_color_manual(values = c(fill_col_ind_dark))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  guides(color=guide_legend(override.aes=list(shape=21)))+
  guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))

####PC3/PC4####
annot_PCA_matrix %>% ggplot(., aes(x=PC3, y=PC4)) +
  geom_point(aes(color = Ind, shape = Time, fill = Tx, size = 10)) +
                       scale_shape_manual(values = c(21, 22, 24)) +
  scale_fill_manual(values =  fill_col_tx)+
  scale_color_manual(values = c(fill_col_ind_dark))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  guides(color=guide_legend(override.aes=list(shape=21)))+
  guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))



```


```{r Correlation Heatmaps, fig.width=12, fig.height=16, collapse=TRUE}

#correlation heatmap for all samples

####PEARSON FILTERED####
fC_Matrix_Full_cpm_filter_pearsoncor <-
  cor(
    fC_Matrix_Full_cpm_filter,
    y = NULL,
    use = "everything",
    method = "pearson"
  )

####SPEARMAN FILTERED####
fC_Matrix_Full_cpm_filter_spearmancor <-
  cor(
    fC_Matrix_Full_cpm_filter,
    y = NULL,
    use = "everything",
    method = "spearman"
  )



#Now let's graph these correlations

pheatmap(fC_Matrix_Full_cpm_filter_pearsoncor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 12, fontsize_number = 9)

pheatmap(fC_Matrix_Full_cpm_filter_spearmancor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 12, fontsize_number = 9)



#now annotate these heatmaps
#Factor 1 - Individual
Individual <- as.factor(c(rep("Ind1", 9), rep("Ind2", 9), rep("Ind3", 9), rep("Ind4", 9), rep("Ind5", 9), rep("Ind6", 9), rep("Ind6REP", 9)))

#Factor 2 - Treatment
tx_factor <- c("DOX", "FLUO", "DMSO")
Tx <- as.factor(c(rep(tx_factor, 21)))
#view(Treatment)

#Factor 3 - Timepoint
time_factor <- c(rep("24", 3), rep("24rec", 3), rep("144rec", 3))
Time <- as.factor(c(rep(time_factor, 7)))

####annotation for colors####
annot_col_hm = list(Tx = c(DOX = "red", FLUO = "blue", DMSO = "black"),
                             Ind = c(Ind1 = "#66C2A5", Ind2 = "#FC8D62", Ind3 = "#1F78B4", Ind4 = "#E78AC3", Ind5 = "#A6D854", Ind6 = "#FFD92A", Ind6REP = "#8B3E9B"),
                             Time = c("24" = "#046A38", "24rec" = "#0050B5", "144rec" = "#B725AD"))

####annotation for values####
annot_list_hm <- data.frame(Individual = as.factor(c(rep("Ind1", 9), rep("Ind2", 9), rep("Ind3", 9), rep("Ind4", 9), rep("Ind5", 9), rep("Ind6", 9), rep("Ind6REP", 9))),
                                               Tx = as.factor(c(rep(tx_factor, 21))), 
                                               Time = as.factor(c(rep(time_factor, 7))))

  
##add in the annotations from above into the dataframe
row.names(annot_list_hm) <- colnames(fC_Matrix_Full_cpm_filter_pearsoncor)
row.names(annot_list_hm) <- colnames(fC_Matrix_Full_cpm_filter_spearmancor)

####ANNOTATED HEATMAPS####
pheatmap(fC_Matrix_Full_cpm_filter_pearsoncor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 10, fontsize_number = 5, annotation_col = annot_list_hm, annotation_colors = annot_col_hm)

pheatmap(fC_Matrix_Full_cpm_filter_spearmancor, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 10, fontsize_number = 5, annotation_col = annot_list_hm, annotation_colors = annot_col_hm)


```

 
```{r Data Frames for Counts DE, include=FALSE}

counts_DE_raw <- read_csv("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_RNAseq/featureCounts_Concat_Matrix_AllSamples_EMP_250220.csv")
#View(counts_DE)

counts_DE <- as.data.frame(Counts_Full_df) 
#%>% 
        #dplyr::select(-"...1") %>% 
        #dplyr::select(-(contains("Ind6REP")))

#sometimes when loaded in this dataframe has the column ..1 - if needed uncomment that out

#now I've removed that individual, so I'll filter all counts by rowmeans
#first check the number of variables in the counts file
##I should have 54 samples if not including Ind6REP
dim(counts_DE)
#[1] 78932    54 - original ensembl annotation
#[1] 28395    54 - new inbuilt annotation



```


```{r Attempted Plot, include=FALSE}
# ####TOTAL NUMBER OF READS PER SAMPLE####
# tx_col <- list(Treatment = c("DMSO" = "#63666D","5FU" = "#DCACED","DOX" = "#499FBD"))
# counts_DE_raw %>% 
#   ggplot(., aes (x =samplenames, y=count, fill = drug, group_by=indv))+
#   geom_col()+
#  geom_hline(aes(yintercept=20000000))+
#  scale_fill_manual(values=drug_palc)+
#   ggtitle(expression("Total number of reads by sample"))+
#   xlab("")+
#   ylab(expression("RNA -sequencing reads"))+
#   theme_bw()+
#   theme(plot.title = element_text(size = rel(2), hjust = 0.5),
#         axis.title = element_text(size = 15, color = "black"),
#         axis.ticks = element_line(linewidth = 1.5),
#         axis.line = element_line(linewidth = 1.5),
#         axis.text.y = element_text(size =10, color = "black", angle = 0, hjust = 0.8, vjust = 0.5),
#         axis.text.x = element_text(size =10, color = "black", angle = 90, hjust = 1, vjust = 0.2),
#         #strip.text.x = element_text(size = 15, color = "black", face = "bold"),
#         strip.text.y = element_text(color = "white"))

```

Now that I've finalized my QC checks, I will go ahead and begin the differential expression analysis pipeline.

```{r Differential Expression Analysis filtering rowMeans greater than 0, collapse=TRUE}

# group <- c(rep(c(1, 2, 3, 4, 5, 6, 7, 8, 9), 6))
# group <- factor(group, levels = c("1", "2", "3", "4", "5", "6", "7", "8", "9"))


####rowMeans > 0 Filtering####
rowMeans_DE <- rowMeans(counts_DE)
counts_DE_filter <- counts_DE[rowMeans_DE > 0,]
dim(counts_DE_filter)

group_1 <- rep(c("DOX_24","FLUO_24",
                                "DMSO_24",
                                "DOX_24rec",
                                "FLUO_24rec",
                                "DMSO_24rec",
                                "DOX_144rec",
                                "FLUO_144rec",
                                "DMSO_144rec"), 6)

group_2 <- rep(c("DOX_24","FLUO_24",
                                "DMSO_24",
                                "DOX_24rec",
                                "FLUO_24rec",
                                "DMSO_24rec",
                                "DOX_144rec",
                                "FLUO_144rec",
                                "DMSO_144rec"), 7)

dge <- DGEList.data.frame(counts = counts_DE_filter, group = group_2, genes = row.names(counts_DE_filter))

#check the inital file before norm factors are calculated
#dge$samples

#calculate the normalization factors with method TMM
dge_calc <- calcNormFactors(dge, method = "TMM")

#check final file after norm calculation
#dge_calc$samples

#after calculating norm factors, that column was changed.

#View(dge_calc)

#Pull out factors
snames <- data.frame("samples" = colnames(dge_calc)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)

#snames_list <- as.list(snames)

snames_time <- snames$Time
snames_tx <- snames$Treatment
snames_ind <- snames$Individual

#Create my model matrix
mm_r <- model.matrix(~0 + group_2)

p <- voom(dge_calc$counts, mm_r, plot = TRUE)

corfit <- duplicateCorrelation(p, mm_r, block = snames_ind)

v <- voom(dge_calc$counts, mm_r, block = snames_ind, correlation = corfit$consensus)

fit <- lmFit(v, mm_r, block = snames_ind, correlation = corfit$consensus)

#make sure to check which order the columns are in - otherwise they won't match right (it was moved into alphabetical and number order)
colnames(mm_r) <- c("DMSO_144rec","DMSO_24","DMSO_24rec","DOX_144rec","DOX_24","DOX_24rec","FLUO_144rec","FLUO_24","FLUO_24rec")

cm_r <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.F24 = FLUO_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.F24r = FLUO_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        V.F144r = FLUO_144rec - DMSO_144rec,
        levels = mm_r
)

vfit_r <- lmFit(p, mm_r)
vfit_r <- contrasts.fit(vfit_r, contrasts = cm_r)

efit2 <- eBayes(vfit_r)

results = decideTests(efit2)
summary(results)
#inbuilt annotation results rowMeans >0
#         V.D24 V.F24 V.D24r V.F24r V.D144r V.F144r
# Down    4567     0   2083      0     288       0
# NotSig 14521 26445  16339  26445   25296   26445
# Up      7357     0   8023      0     861       0

## have no DE genes detected for 5FU, only DOX


####plot your voom####
voom_plot <- voom(dge_calc, mm_r, plot = TRUE)


```


The voom plot for this with rowMeans > 0 has a dip in the front, meaning that there is likely some noise left over in my data.
I am going to filter a bit stricter at rowMeans > 1 with the same process as above.

```{r Differential Expression Analysis filtering rowMeans greater than 1, collapse=TRUE}

####rowMeans >1####
rowMeans_DE <- rowMeans(counts_DE)
counts_DE_filter1 <- counts_DE[rowMeans_DE > 1,]
dim(counts_DE_filter1)


dge1 <- DGEList.data.frame(counts = counts_DE_filter1, group = group_2, genes = row.names(counts_DE_filter1))

#check the inital file before norm factors are calculated
#dge1$samples

#calculate the normalization factors with method TMM
dge1_calc <- calcNormFactors(dge1, method = "TMM")

#check final file after norm calculation
#dge1_calc$samples

#View(dge1_calc)

#Pull out factors
snames1 <- data.frame("samples" = colnames(dge1_calc)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)

#snames_list <- as.list(snames)

snames1_time <- snames1$Time
snames1_tx <- snames1$Treatment
snames1_ind <- snames1$Individual

#Create my model matrix
mm_r1 <- model.matrix(~0 + group_2)

p1 <- voom(dge1_calc$counts, mm_r1, plot = TRUE)

corfit1 <- duplicateCorrelation(p1, mm_r1, block = snames1_ind)

v1 <- voom(dge1_calc$counts, mm_r1, block = snames1_ind, correlation = corfit1$consensus)

fit1 <- lmFit(v1, mm_r1, block = snames1_ind, correlation = corfit1$consensus)

#make sure to check which order the columns are in - otherwise they won't match right (it was moved into alphabetical and number order)
colnames(mm_r1) <- c("DMSO_144rec","DMSO_24","DMSO_24rec","DOX_144rec","DOX_24","DOX_24rec","FLUO_144rec","FLUO_24","FLUO_24rec")

cm_r1 <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.F24 = FLUO_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.F24r = FLUO_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        V.F144r = FLUO_144rec - DMSO_144rec,
        levels = mm_r1
)

vfit_r1 <- lmFit(p1, mm_r1)
vfit_r1 <- contrasts.fit(vfit_r1, contrasts = cm_r1)

efit4 <- eBayes(vfit_r1)

results1 = decideTests(efit4)
summary(results1)
#inbuilt annotation results rowMeans >0
#         V.D24 V.F24 V.D24r V.F24r V.D144r V.F144r
# Down    4567     0   2083      0     288       0
# NotSig 14521 26445  16339  26445   25296   26445
# Up      7357     0   8023      0     861       0

## have no DE genes detected for 5FU, only DOX


####plot your voom####
voom_plot1 <- voom(dge1_calc, mm_r1, plot = TRUE)
```
This looks much better! No more noise in the front. 
I could potentially filter at rowMeans >0.5, but I will leave that for now.

Now I can go ahead and make toptables for safekeeping and later analysis

```{r Make a TopTable with Your New Data, results='hide',collapse=TRUE}

top.table_V.D24 <- topTable(fit = efit4, coef = "V.D24", number = nrow(dge_calc), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.D24)

top.table_V.F24 <- topTable(fit = efit4, coef = "V.F24", number = nrow(dge_calc), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.F24)

top.table_V.D24r <- topTable(fit = efit4, coef = "V.D24r", number = nrow(dge_calc), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.D24r)

top.table_V.F24r <- topTable(fit = efit4, coef = "V.F24r", number = nrow(dge_calc), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.F24r)

top.table_V.D144r <- topTable(fit = efit4, coef = "V.D144r", number = nrow(dge_calc), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.D144r)

top.table_V.F144r <- topTable(fit = efit4, coef = "V.F144r", number = nrow(dge_calc), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.F144r)

```


Volcano Plots
```{r Volcano Plots from Pairwise Gene Analysis, fig.width=8, fig.height=10, collapse=TRUE}

generate_volcano_plot <- function(toptable, title) {
  
  # Add Significance Labels
  toptable$Significance <- "Not Significant"
  toptable$Significance[toptable$logFC > 0 & toptable$adj.P.Val < 0.05] <- "Upregulated"
  toptable$Significance[toptable$logFC < 0 & toptable$adj.P.Val < 0.05] <- "Downregulated"

  # Generate Volcano Plot
  ggplot(toptable, aes(x = logFC, y = -log10(P.Value), color = Significance)) +
    geom_point(alpha = 0.4, size = 2) + 
    scale_color_manual(values = c("Downregulated" = "red", "Upregulated" = "blue", "Not Significant" = "gray")) +
    xlim(-5, 5) +
    labs(title = title, x = "log2 Fold Change", y = "-log10 P-value") +
    theme(legend.position = "none", 
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25))) +
    theme_bw()
}

#now that I've made a function, I can make volcano plots for each of my comparisons (6 total)

volcano_plots <- list(
  "V.D24" = generate_volcano_plot(top.table_V.D24, "Volcano Plot DOX 24hr (adj P-val<0.05)"),
  "V.F24" = generate_volcano_plot(top.table_V.F24, "Volcano Plot 5FU 24hr (adj P-val<0.05)"),
  "V.D24r" = generate_volcano_plot(top.table_V.D24r, "Volcano Plot DOX 24hr Recovery (adj P-val<0.05)"),
  "V.F24r" = generate_volcano_plot(top.table_V.F24r, "Volcano Plot 5FU 24hr Recovery (adj P-val<0.05)"),
  "V.D144r" = generate_volcano_plot(top.table_V.D144r, "Volcano Plot DOX 6d Recovery (adj P-val<0.05)"),
  "V.F144r" = generate_volcano_plot(top.table_V.F144r, "Volcano Plot 5FU 6d Recovery (adj P-val<0.05)")
)

# Display each volcano plot
for (plot_name in names(volcano_plots)) {
  print(volcano_plots[[plot_name]])
}


```


```{r Pull Out Genes of Interest}
fC_df_filter_cpm <- as.data.frame(fC_Matrix_Full_cpm_filter)
my_data <- rownames_to_column(fC_df_filter_cpm, var = "entrezgene_id")
col_tx_large <- rep(c("#499FBD" ,"#DCACED", "#63666D"), 21)
col_tx_large_2 <- c(rep("#499FBD" , 3), rep("#DCACED", 3), rep("#63666D", 3), 21)


##Add columns with more information to each gene I pull out##
ind_names <- c(rep("Ind1", 9), rep("Ind2", 9), rep("Ind3", 9), rep("Ind4", 9), rep("Ind5", 9), rep("Ind6", 9), rep("Ind6REP", 9))
time_names <- c(rep("24", 3), rep("24rec", 3), rep("144rec", 3))
time_names2 <- c("24", "24rec", "144rec")
time_names <- c(rep(time_names, 7))
time_names2 <- c(rep(time_names2, 7))
tx_names <- c("DOX", "FLUO", "DMSO")
tx_names <- c(rep(tx_names, 21))
tx_names2 <- c(rep("DOX", 3), rep("FLUO", 3), rep("DMSO", 3))
tx_names2 <- c(rep(tx_names2, 21))
txtime_names <- c("DOX_24", "FLUO_24", "DMSO_24", "DOX_24rec", "FLUO_24rec", "DMSO_24rec", "DOX_144rec",   "FLUO_144rec",  "DMSO_144rec")
txtime_names <- c(rep(txtime_names, 7))
txtime_names2 <- c("DOX_24", "DOX_24rec", "DOX_144rec", "FLUO_24", "FLUO_24rec", "FLUO_144rec","DMSO_24", "DMSO_24rec", "DMSO_144rec")
txtime_names2 <- c(rep(txtime_names2, 7))


my_data 
my_data_Fluo <- my_data %>% dplyr::select(c("entrezgene_id", "FLUO_24_Ind1", "FLUO_24rec_Ind1", "FLUO_144rec_Ind1", "FLUO_24_Ind2", "FLUO_24rec_Ind2", "FLUO_144rec_Ind2", "FLUO_24_Ind3", "FLUO_24rec_Ind3", "FLUO_144rec_Ind3", "FLUO_24_Ind4", "FLUO_24rec_Ind4", "FLUO_144rec_Ind4", "FLUO_24_Ind5", "FLUO_24rec_Ind5", "FLUO_144rec_Ind5", "FLUO_24_Ind6", "FLUO_24rec_Ind6", "FLUO_144rec_Ind6", "FLUO_24_Ind6REP", "FLUO_24rec_Ind6REP", "FLUO_144rec_Ind6REP"))

###MDM2 - 4193###
MDM2 <- my_data %>% filter(entrezgene_id=="4193")
MDM2_new <- as.data.frame(MDM2) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "FLUO_24_Ind1", "FLUO_24rec_Ind1", "FLUO_144rec_Ind1", "DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1","DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "FLUO_24_Ind2", "FLUO_24rec_Ind2", "FLUO_144rec_Ind2", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3", "FLUO_24_Ind3", "FLUO_24rec_Ind3", "FLUO_144rec_Ind3", "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "FLUO_24_Ind4", "FLUO_24rec_Ind4", "FLUO_144rec_Ind4", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "FLUO_24_Ind5", "FLUO_24rec_Ind5", "FLUO_144rec_Ind5", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "FLUO_24_Ind6", "FLUO_24rec_Ind6", "FLUO_144rec_Ind6", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP", "FLUO_24_Ind6REP", "FLUO_24rec_Ind6REP", "FLUO_144rec_Ind6REP", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")
MDM2_melt <- melt(MDM2, variable.name = "sample")
MDM2_melt_new <- melt(MDM2_new, variable.name = "sample")

MDM2_melt_df <- data.frame(tx = factor(tx_names, levels = unique(tx_names)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names, levels = unique(txtime_names)),
                               time = factor(time_names, levels = unique(time_names)))
MDM2_melt_df2 <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
MDM2_melt_df_all <- cbind(MDM2_melt, MDM2_melt_df)
MDM2_melt_df_all2 <- cbind(MDM2_melt_new, MDM2_melt_df2)

ggplot(MDM2_melt, aes(x=value, y=sample, fill = sample))+
  ggtitle("MDM2")+
    geom_bar(stat = "identity", width = 0.8)+
  theme(legend.position = "none")+
  scale_fill_manual(values = col_tx_large)+
    xlab("log2cpm")+
  ylab("Conditions")
###FLUO ONLY####
MDM2_fluo <- my_data_Fluo %>% filter(entrezgene_id=="4193")
MDM2_melt_fluo <- melt(MDM2_fluo, variable.name = "sample")
ggplot(MDM2_melt_fluo, aes(x=value, y=sample, fill = sample))+
  ggtitle("MDM2")+
    geom_bar(stat = "identity", width = 0.8)+
  theme(legend.position = "none")+
  scale_fill_manual(values = col_tx_large)+
    xlab("log2cpm")+
  ylab("Conditions")


####CDKN1A - 1026####
CDKN1A <- my_data %>% filter(entrezgene_id=="1026")
CDKN1A_new <- as.data.frame(CDKN1A) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "FLUO_24_Ind1", "FLUO_24rec_Ind1", "FLUO_144rec_Ind1", "DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1","DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "FLUO_24_Ind2", "FLUO_24rec_Ind2", "FLUO_144rec_Ind2", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3", "FLUO_24_Ind3", "FLUO_24rec_Ind3", "FLUO_144rec_Ind3", "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "FLUO_24_Ind4", "FLUO_24rec_Ind4", "FLUO_144rec_Ind4", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "FLUO_24_Ind5", "FLUO_24rec_Ind5", "FLUO_144rec_Ind5", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "FLUO_24_Ind6", "FLUO_24rec_Ind6", "FLUO_144rec_Ind6", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP", "FLUO_24_Ind6REP", "FLUO_24rec_Ind6REP", "FLUO_144rec_Ind6REP", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")
CDKN1A_melt <- melt(CDKN1A, variable.name = "sample")
CDKN1A_melt_new <- melt(CDKN1A_new, variable.name = "sample")

CDKN1A_melt_df <- data.frame(tx = factor(tx_names, levels = unique(tx_names)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names, levels = unique(txtime_names)),
                               time = factor(time_names, levels = unique(time_names)))
CDKN1A_melt_df2 <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
CDKN1A_melt_df_all <- cbind(CDKN1A_melt, CDKN1A_melt_df)
CDKN1A_melt_df_all2 <- cbind(CDKN1A_melt_new, CDKN1A_melt_df2)

ggplot(CDKN1A_melt, aes(x=value, y=sample, fill = sample))+
  ggtitle("CDKN1A")+
    geom_bar(stat = "identity", width = 0.8)+
  theme(legend.position = "none")+
  scale_fill_manual(values = col_tx_large)+
    xlab("log2cpm")+
  ylab("Conditions")


####SIRT1 - 23411####
SIRT1 <- my_data %>% filter(entrezgene_id=="23411")
SIRT1_new <- as.data.frame(SIRT1) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "FLUO_24_Ind1", "FLUO_24rec_Ind1", "FLUO_144rec_Ind1", "DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1","DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "FLUO_24_Ind2", "FLUO_24rec_Ind2", "FLUO_144rec_Ind2", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3", "FLUO_24_Ind3", "FLUO_24rec_Ind3", "FLUO_144rec_Ind3", "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "FLUO_24_Ind4", "FLUO_24rec_Ind4", "FLUO_144rec_Ind4", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "FLUO_24_Ind5", "FLUO_24rec_Ind5", "FLUO_144rec_Ind5", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "FLUO_24_Ind6", "FLUO_24rec_Ind6", "FLUO_144rec_Ind6", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP", "FLUO_24_Ind6REP", "FLUO_24rec_Ind6REP", "FLUO_144rec_Ind6REP", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")
SIRT1_melt <- melt(SIRT1, variable.name = "sample")
SIRT1_melt_new <- melt(SIRT1_new, variable.name = "sample")

SIRT1_melt_df <- data.frame(tx = factor(tx_names, levels = unique(tx_names)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names, levels = unique(txtime_names)),
                               time = factor(time_names, levels = unique(time_names)))
SIRT1_melt_df2 <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
SIRT1_melt_df_all <- cbind(SIRT1_melt, SIRT1_melt_df)
SIRT1_melt_df_all2 <- cbind(SIRT1_melt_new, SIRT1_melt_df2)

SIRT1_melt <- melt(SIRT1, variable.name = "sample")
ggplot(SIRT1_melt, aes(x=value, y=sample, fill = sample))+
  ggtitle("SIRT1")+
    geom_bar(stat = "identity", width = 0.8)+
  theme(legend.position = "none")+
  scale_fill_manual(values = col_tx_large)+
    xlab("log2cpm")+
  ylab("Conditions")


####Pull out genes and make them into boxplots with points for each individual####
####MDM2####
MDM2_melt_df_all2 %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "MDM2")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")

####CDKN1A####
CDKN1A_melt_df_all2 %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "CDKN1A")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")


####SIRT1####
SIRT1_melt_df_all2 %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "SIRT1")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")

```

Now that I've pulled out some select genes for each condition - let's look at some typically expressed cardiac genes to check whether they are present in this dataset

```{r Typical genes expressed in iPSC-CMs}
fC_df_filter_cpm <- as.data.frame(fC_Matrix_Full_cpm_filter)
my_data <- rownames_to_column(fC_df_filter_cpm, var = "entrezgene_id")
###now let's pull some classic cardiac genes expressed in iPSC-CMs###
genecardiccheck <- c("MYH7", "TNNT2","MYH6","ACTN2","BMP3","TNNI3","RYR2","CACNA1C","KCNQ1", "HCN1", "ADRB1", "ADRB2")

# ensembl <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
# saveRDS(ensembl, "data/ensembl_backup.RDS")
ensembl <- readRDS("data/ensembl_backup.RDS")
my_chr <- c(1:22, 'M', 'X', 'Y')  ## creates a filter for each database

my_attributes <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')

heartgenes <- getBM(attributes=my_attributes,filters ='hgnc_symbol',
                values = genecardiccheck, mart = ensembl)
write.csv(heartgenes, "data/heartgenes.csv")
heartgenes <-read.csv("data/heartgenes.csv")

fungraph <- as.data.frame(fC_df_filter_cpm[rownames(fC_df_filter_cpm) %in% heartgenes$entrezgene_id,])


fungraph %>% 
  rownames_to_column("entrezgene_id") %>% 
  pivot_longer(-entrezgene_id, names_to = "samples",values_to = "counts") %>% 
  mutate(gene = case_match(entrezgene_id,"88"~"ACTN2","153"~"ADRB1",
  "154"~"ADRB2","651"~"BMP3","775"~"CACNA1C", "100874369"~"CACNA1C","348980"~"HCN1",
                           "3784"~"KCNQ1", "4624"~"MYH6","4625"~"MYH7","6262"~"RYR2",
                           "7137"~"TNNI3","7139"~"TNNT2",.default = entrezgene_id)) %>% 
  ggplot(., aes(x=reorder(gene,counts,decreasing=TRUE), y=counts))+
  geom_boxplot()+
  ggtitle(expression("Expression of typical cardiac tissue genes"))+
  xlab("")+
  ylab(expression("log"[2]~"cpm"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))


```


Now that I've gone through and figured out the basics of DE - let's remove unwanted variation using ruv
This uses my replicates to figure out covariates in the data and remove variation occurring due to batch effects or other effects
We're seeing a batch effect from RNA extraction (all RNAseq library prep done in a single batch) occurring here in PCA and cor. heatmaps rather than individuals clustering together.

```{r Remove Unwanted Variation - RUVs, include = FALSE}
# 
# #counts need to be integer values and in a numeric matrix for ruv
# counts_DE_matrix <- as.matrix(counts_DE_filter1)
# dim(counts_DE_matrix)
# #[1] 20443    54 - new inbuilt
# #this does NOT contain the replicate samples, save for later
# 
# ##this file contains the replicates that I can pull from
# fC_Matrix_cpm_filter_ruv <- as.matrix (fC_Matrix_Full_cpm_filter)
# 
# #create a matrix specifying the replicates
# reps <- as.data.frame(fC_Matrix_Full_cpm_filter) %>%
#   dplyr::select(contains("Ind6"))
# dim(reps)
# #[1] 14225    18
# reps_mat <- as.matrix(reps)
# 
# #using the documentation do this:
# #controls: gene-by-sample numeric matrix containing counts
# controls <- rownames(fC_Matrix_Full_cpm_filter)
# 
# #I have replicates for Individual 6, which I can compare using this method
# #They are matched as following:
# ##col46:55, 47:56, 48:57, 49:58, 50:59, 51:60, 52:61, 53:62, 54:63
# ###this is the scIdx argument for reps, the cIdx argument is for genes
# #differences <- matrix(data = c(1, 10, 2, 11, 3, 12, 4, 13, 5, 14, 6, 15, 7, 16, 8, 17, 9, 18), byrow = TRUE, nrow = 9)
# differences_full <- matrix(data = c(46, 55, 47, 56, 48 , 57, 49, 58, 50 , 59, 51, 60, 52, 61, 53, 62, 54, 63), byrow = TRUE, nrow = 9)
# 
# 
# signature(x = "matrix", cIdx = "ANY", k = "numeric", scIdx = "matrix")
# 
# 
# seqRUVs <- RUVs(rep, controls, k=3, differences_full, isLog = TRUE)
# 
# pData(counts_DE)
# head(seqRUVs)
# 
# 
# ###now try putting this in your matrix?
# design <- model.matrix(~0 + group_1, seqRUVs)
# 
# 
# 
# 
# ####try it exactly like the documentation and filter afterwards?####
# genes <- rownames(fC_Matrix_Full)[grep("^"), rownames(fC_Matrix_Full)]
# spikes <- rownames()
# 
# ####leaving off here EMP 25/02/26####


```





```{r Remove Unwanted Variation - RUVs New, include = FALSE}
#omar stuff

# vignette("RUVSeq")
# library("EDASeq")
# RUVSeq::makeGroups() %>% help()
# comrade_groups <- makeGroups(annot$Tx_Time) 
# comrade_groups <- comrade_groups %>% as.matrix()
# 
# gene_names_comrade <- counts_DE_filter1 %>% rownames() %>% as.double()
# counts_DE_filter1_ES <- EDASeq::newSeqExpressionSet(counts = counts_DE_filter1)
# 
# counts_DE_filter1_ES
# 
# counts_DE_filter1_ES %>% assay() 
# set3 <- RUVSeq::RUVs(x = counts_DE_filter1_ES, cIdx = gene_names_comrade, k=1, scIdx = comrade_groups, isLog = FALSE)
# 
# #counts need to be integer values and in a numeric matrix for ruv
# counts_DE_matrix <- as.matrix(counts_DE_filter1)
# dim(counts_DE_matrix)
# counts_DE_matrix %>% head
# counts_DE_matrix %>% rownames() %>% unique() %>% length()
# 
# #[1] 20443    54 - new inbuilt
# #this does NOT contain the replicate samples, save for later
# 
# ##this file contains the replicates that I can pull from
# fC_Matrix_cpm_filter_ruv <- as.matrix (fC_Matrix_Full_cpm_filter)
# fC_Matrix_cpm_filter_ruv
# 
# #create a matrix specifying the replicates
# reps <- as.data.frame(fC_Matrix_Full_cpm_filter) %>%
#   dplyr::select(contains("Ind6"))
# dim(reps)
# #[1] 14225    18
# reps_mat <- as.matrix(reps)
# 
# #using the documentation do this:
# #controls: gene-by-sample numeric matrix containing counts
# controls <- rownames(fC_Matrix_Full_cpm_filter)
# 
# #I have replicates for Individual 6, which I can compare using this method
# #They are matched as following:
# ##col46:55, 47:56, 48:57, 49:58, 50:59, 51:60, 52:61, 53:62, 54:63
# ###this is the scIdx argument for reps, the cIdx argument is for genes
# #differences <- matrix(data = c(1, 10, 2, 11, 3, 12, 4, 13, 5, 14, 6, 15, 7, 16, 8, 17, 9, 18), byrow = TRUE, nrow = 9)
# differences_full <- matrix(data = c(46, 55, 47, 56, 48 , 57, 49, 58, 50 , 59, 51, 60, 52, 61, 53, 62, 54, 63), byrow = TRUE, nrow = 9)
# 
# 
# signature(x = "matrix", cIdx = "ANY", k = "numeric", scIdx = "matrix")
# 
# 
# seqRUVs <- RUVs(rep, controls, k=3, differences_full, isLog = TRUE)
# 
# pData(counts_DE)
# head(seqRUVs)
# 
# 
# ###now try putting this in your matrix?
# design <- model.matrix(~0 + group_1, seqRUVs)
# 
# 
# 
# 
# ####try it exactly like the documentation and filter afterwards?####
# genes <- rownames(fC_Matrix_Full)[grep("^"), rownames(fC_Matrix_Full)]
# spikes <- rownames()
# 
# ####leaving off here EMP 25/02/27####
# 

```


```{r Remove Unwanted Variation with RUVs, include = FALSE}
# #OMAR NEW CODE - EMMA EDITS
# fC_Matrix_Full_cpm_filter
# dim(fC_Matrix_Full_cpm_filter)
# 
# #now I want to un-log transform this data - square it?
# fC_Matrix_Full_new <- fC_Matrix_Full_cpm_filter^2
# dim(fC_Matrix_Full_new)
# 
# 
# #  counts need to be integer values and in a numeric matrix
# # note: the log transformation needs to be accounted for in the isLog argument in RUVs function.
# counts <- as.matrix(fC_Matrix_Full)
# 
# # Create a DataFrame for the phenoData
# phenoData <- DataFrame(annot)
# 
# # Now create the RangedSummarizedExperiment necessary for RUVs input
# # looks like it did need both the phenodata and the counts.
# set <- SummarizedExperiment(assays =  counts, metadata = phenoData)
# 
# # Generate a background matrix
# # The column "Cond" holds the comparisons that you actually want to make. DOX_24, DMSO_24,5FU_24, DOX_3,etc.
# scIdx <-RUVSeq::makeGroups(phenoData$Tx_Time)
# scIdx
# # Apply RUVs function from RUVSeq
# # note: for now just get code to work. "k" will be iteratively adjusted over time depending on your PCA.
# set <- RUVSeq::RUVs(x = counts, k =1, scIdx = scIdx, isLog = FALSE)
# 
# #DO NOT USE THESE COUNTS FOR LINEAR MODELING EMMA
# 
# 
# 
# 
# 
# # get the ruv weights to put into the linear model. n weights = k.
# RUV_df <- set$W %>% as.data.frame()
# RUV_df$Names <- rownames(RUV_df)
# 
# # Make sure the data frame names match 
# RUV_df_rm <- RUV_df[RUV_df$Names %in% annot_new$samples, ] 
# RUV_1 <-  RUV_df_rm$W_1
# 
# 
# # before ruv
# prcomp_res <- prcomp(t(counts), scale. = TRUE, center = TRUE, )
# ggplot2::autoplot(prcomp_res, data = annot, colour = "Tx_Time", shape = "Ind", size =4)+
#   theme_bw()+
#   ggtitle("No RUV")
# # after ruv
# prcomp_res <- prcomp(t(set$normalizedCounts), scale. = TRUE, center = TRUE, )
# ggplot2::autoplot(prcomp_res, data = annot, colour = "Tx_Time", shape = "Ind", size =4)+
#   theme_bw()+
#   ggtitle("RUV Corrected")
# 
# 
# #Now that I've calculated the weights for RUV correction, I want to put my raw counts data into my model
# 
# 
# counts_DE %>% head()
# counts_DE %>% dim()
# 
# raw_counts_mat <- counts_DE %>% dplyr::select(-(contains("Ind6REP"))) %>% as.matrix()
# dim(raw_counts_mat)
# 
# #I want to make sure that I only have 54 observations in this dataset, removing the biological replicate samples (doesn't matter whether you remove Ind6 or Ind6REP, but I will choose Ind6REP)
# 
# dge1 <- DGEList.data.frame(counts = raw_counts_mat, group = group_2, genes = row.names(raw_counts_mat))
# 
# #check the initial file before norm factors are calculated
# #dge1$samples
# 
# #calculate the normalization factors with method TMM
# dge1_calc <- calcNormFactors(dge1, method = "TMM")
# 
# #check final file after norm calculation
# #dge1_calc$samples
# 
# #View(dge1_calc)
# 
# #Pull out factors
# snames1 <- data.frame("samples" = colnames(dge1_calc)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)
# 
# #snames_list <- as.list(snames)
# snames1_ind <- snames1$Individual
# 
# # Make sure the names match 
# annot_new$samples == RUV_df_rm$Names
# annot_new$RUV_1 <- RUV_df_rm$W_1
# 
# 
# #Create my model matrix
# mm_r1 <- model.matrix(~0 + group_1 + RUV_1, data = annot_new)
# 
# p1 <- voom(dge1_calc$counts, mm_r1, plot = TRUE)
# 
# corfit1 <- duplicateCorrelation(p1, mm_r1, block = snames1_ind)
# 
# v1 <- voom(dge1_calc$counts, mm_r1, block = snames1_ind, correlation = corfit1$consensus)
# 
# fit1 <- lmFit(v1, mm_r1, block = snames1_ind, correlation = corfit1$consensus)
# 
# 
# #make sure to check which order the columns are in - otherwise they won't match right (it was moved into alphabetical and number order)
# 
# mm_r1_names <- str_replace(string = colnames(mm_r1), pattern = "group_1", replacement = "")
# design <- model.matrix(~ group_1 + RUV_1 , annot_new)
# colnames(mm_r1) <- mm_r1_names
# 
# cm_r1 <- makeContrasts(
#         V.D24 = DOX_24 - DMSO_24,
#         V.F24 = FLUO_24 - DMSO_24,
#         V.D24r = DOX_24rec - DMSO_24rec,
#         V.F24r = FLUO_24rec - DMSO_24rec,
#         V.D144r = DOX_144rec - DMSO_144rec,
#         V.F144r = FLUO_144rec - DMSO_144rec,
#         RUV_1_24 = RUV_1 - DMSO_24,
#         RUV_1_24r = RUV_1 - DMSO_24rec,
#         RUV_1_144r = RUV_1 - DMSO_144rec,
#         levels = mm_r1
# )
# 
# vfit_r1 <- lmFit(p1, mm_r1)
# vfit_r1 <- contrasts.fit(vfit_r1, contrasts = cm_r1)
# 
# efit4 <- eBayes(vfit_r1)
# 
# results1 = decideTests(efit4)
# summary(results1)
# 
# toptable_Dupcor_DOX <- topTable(efit4, coef = "V.D24", number = nrow(dge1_calc$counts), p.value = 1)
# toptable_Dupcor_DOXrec <- topTable(efit4, coef = "V.D24r", number = nrow(dge1_calc$counts), p.value = 1)
# toptable_Dupcor_DOX144 <- topTable(efit4, coef = "V.D144r", number = nrow(dge1_calc$counts), p.value = 1)
# 
# toptable_Dupcor_DOX$logFC %>% hist(., main = "DOX vs VEH 24hr")
# toptable_Dupcor_DOXrec$logFC %>% hist(., main = "DOX vs VEH 24hr Recovery")
# toptable_Dupcor_DOX144$logFC %>% hist(., main = "DOX vs VEH 6d Recovery")
```


```{r RUVs Correction}
fC_Matrix_Full_cpm_filter
dim(fC_Matrix_Full_cpm_filter)

#now I want to un-log transform this data - square it?
fC_Matrix_Full_new <- fC_Matrix_Full_cpm_filter^2
dim(fC_Matrix_Full_new)


#  counts need to be integer values and in a numeric matrix
# note: the log transformation needs to be accounted for in the isLog argument in RUVs function.
# however, I'll be using raw counts only for this instance
counts <- as.matrix(fC_Matrix_Full)

# Create a DataFrame for the phenoData
phenoData <- DataFrame(annot)

# Now create the RangedSummarizedExperiment necessary for RUVs input
# looks like it did need both the phenodata and the counts.
set <- SummarizedExperiment(assays =  counts, metadata = phenoData)

# Generate a background matrix
# The column "Cond" holds the comparisons that you actually want to make. DOX_24, DMSO_24,5FU_24, DOX_3,etc.
scIdx <-RUVSeq::makeGroups(phenoData$Tx_Time)
scIdx
# Apply RUVs function from RUVSeq
# note: for now just get code to work. "k" will be iteratively adjusted over time depending on your PCA.
set <- RUVSeq::RUVs(x = counts, k =1, scIdx = scIdx, isLog = TRUE)

set2 <- RUVSeq::RUVs(x = counts, k =2, scIdx = scIdx, isLog = TRUE)

set3 <- RUVSeq::RUVs(x = counts, k =3, scIdx = scIdx, isLog = TRUE)

set4 <- RUVSeq::RUVs(x = counts, k =4, scIdx = scIdx, isLog = TRUE)

set5 <- RUVSeq::RUVs(x = counts, k =5, scIdx = scIdx, isLog = TRUE)

#DO NOT USE THESE COUNTS FOR LINEAR MODELING

# get the ruv weights to put into the linear model. n weights = k.
#this one is k=1
RUV_df <- set$W %>% as.data.frame()
RUV_df$Names <- rownames(RUV_df)

#k=2
RUV_df2 <- set2$W %>% as.data.frame()
RUV_df2$Names <- rownames(RUV_df2)

#k=3
RUV_df3 <- set3$W %>% as.data.frame()
RUV_df3$Names <- rownames(RUV_df3)


#k=4
RUV_df4 <- set4$W %>% as.data.frame()
RUV_df4$Names <- rownames(RUV_df4)

#k=5
RUV_df5 <- set5$W %>% as.data.frame()
RUV_df5$Names <- rownames(RUV_df5)

# Make sure the data frame names match 
#k=1
RUV_df_rm <- RUV_df[RUV_df$Names %in% annot$samples, ] 
RUV_1 <-  RUV_df_rm$W_1

#k=2
RUV_df_rm2 <- RUV_df2[RUV_df2$Names %in% annot$samples, ] 
RUV_2 <-  RUV_df_rm2$W_2

#k=3
RUV_df_rm3 <- RUV_df3[RUV_df3$Names %in% annot$samples, ] 
RUV_3 <-  RUV_df_rm3$W_3

#k=4
RUV_df_rm4 <- RUV_df4[RUV_df4$Names %in% annot$samples, ] 
RUV_4 <-  RUV_df_rm4$W_4

#k=5
RUV_df_rm5 <- RUV_df5[RUV_df5$Names %in% annot$samples, ] 
RUV_5 <-  RUV_df_rm5$W_5


# before ruv
prcomp_res <- prcomp(t(counts), scale. = FALSE, center = TRUE)
ggplot2::autoplot(prcomp_res, data = annot, colour = "Tx_Time", shape = "Ind", size =4)+
  theme_bw()+
  ggtitle("No RUV")

# after ruv k=1
prcomp_res <- prcomp(t(set$normalizedCounts), scale. = FALSE, center = TRUE)
ggplot2::autoplot(prcomp_res, data = annot, colour = "Tx_Time", shape = "Ind", size =4)+
  theme_bw()+
  ggtitle("RUV Corrected k=1")

# after ruv k=2
prcomp_res <- prcomp(t(set2$normalizedCounts), scale. = FALSE, center = TRUE)
ggplot2::autoplot(prcomp_res, data = annot, colour = "Tx_Time", shape = "Ind", size =4)+
  theme_bw()+
  ggtitle("RUV Corrected k=2")

# after ruv k=3
prcomp_res <- prcomp(t(set3$normalizedCounts), scale. = FALSE, center = TRUE)
ggplot2::autoplot(prcomp_res, data = annot, colour = "Tx_Time", shape = "Ind", size =4)+
  theme_bw()+
  ggtitle("RUV Corrected k=3")

#after ruv k=4
prcomp_res <- prcomp(t(set4$normalizedCounts), scale. = FALSE, center = TRUE)
ggplot2::autoplot(prcomp_res, data = annot, colour = "Tx_Time", shape = "Ind", size =4)+
  theme_bw()+
  ggtitle("RUV Corrected k=4")

# after ruv k=5
prcomp_res <- prcomp(t(set5$normalizedCounts), scale. = FALSE, center = TRUE)
ggplot2::autoplot(prcomp_res, data = annot, colour = "Tx_Time", shape = "Ind", size =4)+
  theme_bw()+
  ggtitle("RUV Corrected k=5")


####new PCA plots no correction####
prcomp_res <- prcomp(t(counts), scale. = FALSE, center = TRUE)
annot_prcomp_res <- prcomp_res$x %>% cbind(., annot)

annot_prcomp_res %>% ggplot(., aes(x=PC1, y=PC2, size = 10)) +
  geom_point(aes(color = Ind, shape = Time, fill = Tx)) +
                       scale_shape_manual(values = c(21, 22, 24)) +
  scale_fill_manual(values =  fill_col_tx)+
  scale_color_manual(values = c(fill_col_ind_dark))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  guides(color=guide_legend(override.aes=list(shape=21)))+
  guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))+
  ggtitle("no RUVs")

###with RUVs correction####
#k=1
prcomp_res_1 <- prcomp(t(set$normalizedCounts), scale. = FALSE, center = TRUE)
annot_prcomp_res_1 <- prcomp_res_1$x %>% cbind(., annot)

annot_prcomp_res_1 %>% ggplot(., aes(x=PC1, y=PC2, size = 10)) +
  geom_point(aes(color = Ind, shape = Time, fill = Tx)) +
                       scale_shape_manual(values = c(21, 22, 24)) +
  scale_fill_manual(values =  fill_col_tx)+
  scale_color_manual(values = c(fill_col_ind_dark))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  guides(color=guide_legend(override.aes=list(shape=21)))+
  guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))+
  ggtitle("RUVs k=1")
  
#k=2
prcomp_res_2 <- prcomp(t(set2$normalizedCounts), scale. = FALSE, center = TRUE)
annot_prcomp_res_2 <- prcomp_res_2$x %>% cbind(., annot)

annot_prcomp_res_2 %>% ggplot(., aes(x=PC1, y=PC2, size = 10)) +
  geom_point(aes(color = Ind, shape = Time, fill = Tx)) +
                       scale_shape_manual(values = c(21, 22, 24)) +
  scale_fill_manual(values =  fill_col_tx)+
  scale_color_manual(values = c(fill_col_ind_dark))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  guides(color=guide_legend(override.aes=list(shape=21)))+
  guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))+
  ggtitle("RUVs k=2")

#k=3
prcomp_res_3 <- prcomp(t(set3$normalizedCounts), scale. = FALSE, center = TRUE)
annot_prcomp_res_3 <- prcomp_res_3$x %>% cbind(., annot)

annot_prcomp_res_3 %>% ggplot(., aes(x=PC1, y=PC2, size = 10)) +
  geom_point(aes(color = Ind, shape = Time, fill = Tx)) +
                       scale_shape_manual(values = c(21, 22, 24)) +
  scale_fill_manual(values =  fill_col_tx)+
  scale_color_manual(values = c(fill_col_ind_dark))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  guides(color=guide_legend(override.aes=list(shape=21)))+
  guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))+
  ggtitle("RUVs k=3")

#k=4
prcomp_res_4 <- prcomp(t(set4$normalizedCounts), scale. = FALSE, center = TRUE)
annot_prcomp_res_4 <- prcomp_res_4$x %>% cbind(., annot)

annot_prcomp_res_4 %>% ggplot(., aes(x=PC1, y=PC2, size = 10)) +
  geom_point(aes(color = Ind, shape = Time, fill = Tx)) +
                       scale_shape_manual(values = c(21, 22, 24)) +
  scale_fill_manual(values =  fill_col_tx)+
  scale_color_manual(values = c(fill_col_ind_dark))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  guides(color=guide_legend(override.aes=list(shape=21)))+
  guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))+
  ggtitle("RUVs k=4")

#k=5
prcomp_res_5 <- prcomp(t(set5$normalizedCounts), scale. = FALSE, center = TRUE)
annot_prcomp_res_5 <- prcomp_res_5$x %>% cbind(., annot)

annot_prcomp_res_5 %>% ggplot(., aes(x=PC1, y=PC2, size = 10)) +
  geom_point(aes(color = Ind, shape = Time, fill = Tx)) +
                       scale_shape_manual(values = c(21, 22, 24)) +
  scale_fill_manual(values =  fill_col_tx)+
  scale_color_manual(values = c(fill_col_ind_dark))+
  guides(fill=guide_legend(override.aes=list(shape=21)))+
  guides(color=guide_legend(override.aes=list(shape=21)))+
  guides(fill=guide_legend(override.aes=list(shape=21,fill=fill_col_tx,color=fill_col_tx)))+
  ggtitle("RUVs k=5")

group_2 <- rep(c("DOX_24","FLUO_24",
                                "DMSO_24",
                                "DOX_24rec",
                                "FLUO_24rec",
                                "DMSO_24rec",
                                "DOX_144rec",
                                "FLUO_144rec",
                                "DMSO_144rec"), 7)


dge1 <- DGEList.data.frame(counts = fC_Matrix_Full, group = group_2, genes = row.names(fC_Matrix_Full))

#check the initial file before norm factors are calculated
#dge1$samples

#calculate the normalization factors with method TMM
dge1_calc <- calcNormFactors(dge1, method = "TMM")

#check final file after norm calculation
#dge1_calc$samples

#View(dge1_calc)

#Pull out factors
snames1 <- data.frame("samples" = colnames(dge1_calc)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)

#snames_list <- as.list(snames)
snames1_ind <- snames1$Individual

# Make sure the names match 
#k=1
annot$samples == RUV_df_rm$Names
annot$RUV_1 <- RUV_df_rm$W_1

#k=2
annot$samples == RUV_df_rm2$Names
annot$RUV_2 <- RUV_df_rm2$W_2

#k=3
annot$samples == RUV_df_rm3$Names
annot$RUV_3 <- RUV_df_rm3$W_3

#k=4
annot$samples == RUV_df_rm4$Names
annot$RUV_4 <- RUV_df_rm4$W_4

#k=5
annot$samples == RUV_df_rm5$Names
annot$RUV_5 <- RUV_df_rm5$W_5


#Create my model matrix 
#k=1
mm_r1 <- model.matrix(~0 + group_2 + RUV_1, data = annot)

p1 <- voom(dge1_calc$counts, mm_r1, plot = TRUE)

corfit1 <- duplicateCorrelation(p1, mm_r1, block = snames1_ind)

v1 <- voom(dge1_calc$counts, mm_r1, block = snames1_ind, correlation = corfit1$consensus)

fit1 <- lmFit(v1, mm_r1, block = snames1_ind, correlation = corfit1$consensus)

#k=2
mm_r2 <- model.matrix(~0 + group_2 + RUV_2, data = annot)

p2 <- voom(dge1_calc$counts, mm_r2, plot = TRUE)

corfit2 <- duplicateCorrelation(p2, mm_r2, block = snames1_ind)

v2 <- voom(dge1_calc$counts, mm_r2, block = snames1_ind, correlation = corfit2$consensus)

fit2 <- lmFit(v2, mm_r2, block = snames1_ind, correlation = corfit2$consensus)

#k=3
mm_r3 <- model.matrix(~0 + group_2 + RUV_3, data = annot)

p3 <- voom(dge1_calc$counts, mm_r3, plot = TRUE)

corfit3 <- duplicateCorrelation(p3, mm_r3, block = snames1_ind)

v3 <- voom(dge1_calc$counts, mm_r3, block = snames1_ind, correlation = corfit3$consensus)

fit3 <- lmFit(v3, mm_r3, block = snames1_ind, correlation = corfit3$consensus)

#k=4
mm_r4 <- model.matrix(~0 + group_2 + RUV_4, data = annot)

p4 <- voom(dge1_calc$counts, mm_r4, plot = TRUE)

corfit4 <- duplicateCorrelation(p4, mm_r4, block = snames1_ind)

v4 <- voom(dge1_calc$counts, mm_r4, block = snames1_ind, correlation = corfit4$consensus)

fit4 <- lmFit(v4, mm_r4, block = snames1_ind, correlation = corfit4$consensus)

#k=5
mm_r5 <- model.matrix(~0 + group_2 + RUV_5, data = annot)

p5 <- voom(dge1_calc$counts, mm_r5, plot = TRUE)

corfit5 <- duplicateCorrelation(p5, mm_r5, block = snames1_ind)

v5 <- voom(dge1_calc$counts, mm_r5, block = snames1_ind, correlation = corfit5$consensus)

fit5 <- lmFit(v5, mm_r5, block = snames1_ind, correlation = corfit5$consensus)

#make sure to check which order the columns are in - otherwise they won't match right (it was moved into alphabetical and number order)
#k=1
mm_r1_names <- str_replace(string = colnames(mm_r1), pattern = "group_2", replacement = "")
design <- model.matrix(~ group_2 + RUV_1 , annot)
colnames(mm_r1) <- mm_r1_names
#k=2
mm_r2_names <- str_replace(string = colnames(mm_r2), pattern = "group_2", replacement = "")
design <- model.matrix(~ group_2 + RUV_2 , annot)
colnames(mm_r2) <- mm_r2_names
#k=3
mm_r3_names <- str_replace(string = colnames(mm_r3), pattern = "group_2", replacement = "")
design <- model.matrix(~ group_2 + RUV_3 , annot)
colnames(mm_r3) <- mm_r3_names
#k=4
mm_r4_names <- str_replace(string = colnames(mm_r4), pattern = "group_2", replacement = "")
design <- model.matrix(~ group_2 + RUV_4 , annot)
colnames(mm_r4) <- mm_r4_names
#k=5
mm_r5_names <- str_replace(string = colnames(mm_r5), pattern = "group_2", replacement = "")
design <- model.matrix(~ group_2 + RUV_5 , annot)
colnames(mm_r5) <- mm_r5_names

#k=1
cm_r1 <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.F24 = FLUO_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.F24r = FLUO_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        V.F144r = FLUO_144rec - DMSO_144rec,
        RUV_1_24 = RUV_1 - DMSO_24,
        RUV_1_24r = RUV_1 - DMSO_24rec,
        RUV_1_144r = RUV_1 - DMSO_144rec,
        levels = mm_r1
)

#k=2
cm_r2 <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.F24 = FLUO_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.F24r = FLUO_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        V.F144r = FLUO_144rec - DMSO_144rec,
        RUV_2_24 = RUV_2 - DMSO_24,
        RUV_2_24r = RUV_2 - DMSO_24rec,
        RUV_2_144r = RUV_2 - DMSO_144rec,
        levels = mm_r2
)
#k=3
cm_r3 <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.F24 = FLUO_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.F24r = FLUO_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        V.F144r = FLUO_144rec - DMSO_144rec,
        RUV_3_24 = RUV_3 - DMSO_24,
        RUV_3_24r = RUV_3 - DMSO_24rec,
        RUV_3_144r = RUV_3 - DMSO_144rec,
        levels = mm_r3
)

#k=4
cm_r4 <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.F24 = FLUO_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.F24r = FLUO_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        V.F144r = FLUO_144rec - DMSO_144rec,
        RUV_4_24 = RUV_4 - DMSO_24,
        RUV_4_24r = RUV_4 - DMSO_24rec,
        RUV_4_144r = RUV_4 - DMSO_144rec,
        levels = mm_r4
)

#k=5
cm_r5 <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.F24 = FLUO_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.F24r = FLUO_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        V.F144r = FLUO_144rec - DMSO_144rec,
        RUV_5_24 = RUV_5 - DMSO_24,
        RUV_5_24r = RUV_5 - DMSO_24rec,
        RUV_5_144r = RUV_5 - DMSO_144rec,
        levels = mm_r5
)


#k=1
vfit_r1 <- lmFit(p1, mm_r1)
vfit_r1 <- contrasts.fit(vfit_r1, contrasts = cm_r1)
#k=2
vfit_r2 <- lmFit(p2, mm_r2)
vfit_r2 <- contrasts.fit(vfit_r2, contrasts = cm_r2)
#k=3
vfit_r3 <- lmFit(p3, mm_r3)
vfit_r3 <- contrasts.fit(vfit_r3, contrasts = cm_r3)
#k=4
vfit_r4 <- lmFit(p4, mm_r4)
vfit_r4 <- contrasts.fit(vfit_r4, contrasts = cm_r4)
#k=5
vfit_r5 <- lmFit(p5, mm_r5)
vfit_r5 <- contrasts.fit(vfit_r5, contrasts = cm_r5)


#k=1
efit1 <- eBayes(vfit_r1)
#k=2
efit2 <- eBayes(vfit_r2)
#k=3
efit3 <- eBayes(vfit_r3)
#k=4
efit4 <- eBayes(vfit_r4)
#k=5
efit5 <- eBayes(vfit_r5)

#k=1
results1 = decideTests(efit1)
summary(results1)
#k=2
results2 = decideTests(efit2)
summary(results2)
#k=3
results3 = decideTests(efit3)
summary(results3)
#k=4
results4 = decideTests(efit4)
summary(results4)
#k=5
results5 = decideTests(efit5)
summary(results5)

#k=1
toptable_Dupcor_DOX <- topTable(efit1, coef = "V.D24", number = nrow(dge1_calc$counts), p.value = 1)
toptable_Dupcor_DOXrec <- topTable(efit1, coef = "V.D24r", number = nrow(dge1_calc$counts), p.value = 1)
toptable_Dupcor_DOX144 <- topTable(efit1, coef = "V.D144r", number = nrow(dge1_calc$counts), p.value = 1)
#k=1 plots
toptable_Dupcor_DOX$logFC %>% hist(, main= "RUVs k=1 DOX 24hr Toptable")
toptable_Dupcor_DOXrec$logFC %>% hist(, main = "RUVs k=1 DOX 24Rec Toptable")
toptable_Dupcor_DOX144$logFC %>% hist(, main = "RUVs k=1 DOX 144Rec Toptable")

#k=2
toptable2_Dupcor_DOX <- topTable(efit2, coef = "V.D24", number = nrow(dge1_calc$counts), p.value = 1)
toptable2_Dupcor_DOXrec <- topTable(efit2, coef = "V.D24r", number = nrow(dge1_calc$counts), p.value = 1)
toptable2_Dupcor_DOX144 <- topTable(efit2, coef = "V.D144r", number = nrow(dge1_calc$counts), p.value = 1)
#k=2 plots
toptable2_Dupcor_DOX$logFC %>% hist(, main= "RUVs k=2 DOX 24hr Toptable")
toptable2_Dupcor_DOXrec$logFC %>% hist(, main = "RUVs k=2 DOX 24Rec Toptable")
toptable2_Dupcor_DOX144$logFC %>% hist(, main = "RUVs k=2 DOX 144Rec Toptable")

#k=3
toptable3_Dupcor_DOX <- topTable(efit3, coef = "V.D24", number = nrow(dge1_calc$counts), p.value = 1)
toptable3_Dupcor_DOXrec <- topTable(efit3, coef = "V.D24r", number = nrow(dge1_calc$counts), p.value = 1)
toptable3_Dupcor_DOX144 <- topTable(efit3, coef = "V.D144r", number = nrow(dge1_calc$counts), p.value = 1)
#k=3 plots
toptable3_Dupcor_DOX$logFC %>% hist(, main= "RUVs k=3 DOX 24hr Toptable")
toptable3_Dupcor_DOXrec$logFC %>% hist(, main = "RUVs k=3 DOX 24Rec Toptable")
toptable3_Dupcor_DOX144$logFC %>% hist(, main = "RUVs k=3 DOX 144Rec Toptable")

#k=4
toptable4_Dupcor_DOX <- topTable(efit4, coef = "V.D24", number = nrow(dge1_calc$counts), p.value = 1)
toptable4_Dupcor_DOXrec <- topTable(efit4, coef = "V.D24r", number = nrow(dge1_calc$counts), p.value = 1)
toptable4_Dupcor_DOX144 <- topTable(efit4, coef = "V.D144r", number = nrow(dge1_calc$counts), p.value = 1)
#k=4 plots
toptable4_Dupcor_DOX$logFC %>% hist(, main= "RUVs k=4 DOX 24hr Toptable")
toptable4_Dupcor_DOXrec$logFC %>% hist(, main = "RUVs k=4 DOX 24Rec Toptable")
toptable4_Dupcor_DOX144$logFC %>% hist(, main = "RUVs k=4 DOX 144Rec Toptable")

#k=5
toptable5_Dupcor_DOX <- topTable(efit5, coef = "V.D24", number = nrow(dge1_calc$counts), p.value = 1)
toptable5_Dupcor_DOXrec <- topTable(efit5, coef = "V.D24r", number = nrow(dge1_calc$counts), p.value = 1)
toptable5_Dupcor_DOX144 <- topTable(efit5, coef = "V.D144r", number = nrow(dge1_calc$counts), p.value = 1)
#k=5 plots
toptable5_Dupcor_DOX$logFC %>% hist(, main= "RUVs k=5 DOX 24hr Toptable")
toptable5_Dupcor_DOXrec$logFC %>% hist(, main = "RUVs k=5 DOX 24Rec Toptable")
toptable5_Dupcor_DOX144$logFC %>% hist(, main = "RUVs k=5 DOX 144Rec Toptable")

```


```{r pvalue and adj pvalue plots of DE, include=FALSE}

####DOX####



```




Now I want to attempt to use Cormotif to look at motifs within my conditions
```{r Cormotif, include=FALSE}



```

