---
title: "Recovery RNAseq DOX Data"
author: "Emma M Pfortmiller"
date: "2025-02-10"
site: workflowr::wflow_site
output:
  workflowr::wflow_html:
    toc: true
    toc_depth: 2
    toc_float: true
    theme: journal
editor_options: 
  chunk_output_type: inline
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{css, echo=FALSE}
pre {
  max-height: 400px;
  overflow-y: auto;
}

pre[class] {
  max-height: 200px;
}
```

```{r Libraries, collapse=TRUE, include=FALSE, results='hide'}
library(ggplot2)
library(ggrepel)
library(reshape2)
library(hrbrthemes)
library(tibble)
library(tidyverse)
library(edgebundleR)
library(readxl)
library(readr)
library(edgeR)
library(pheatmap)
library(ggfortify)
library(PCAtools)
library(Cormotif)
library(RColorBrewer)
library(biomaRt)
library(RUVSeq)
library(Rfast)
library(cowplot)
library(gprofiler2)
```

I want to separate out my dataset for this to only include the DOX and DMSO vehicle samples
```{r Dataset DOX}

fC_Matrix_Full_cpm_filter <- readRDS("data/fC_Matrix_Full_cpm_filter.RDS")

fC_Matrix_Full_cpm_filter_dox <- as.data.frame(fC_Matrix_Full_cpm_filter) %>% dplyr::select(-(contains("FLUO")))
#this dataset is the original log2cpm transformed and rowMeans filtered dataset which I need for heatmaps

counts_DE_df <- readRDS("data/counts_DE_df.RDS")
#this dataframe is pre-filtered and only contains genes (rows) that were present after log2cpm transformation and filtering by rowMeans > 0

DOX_counts_DE_df <- counts_DE_df %>% dplyr::select(-(contains("FLUO")))
dim(DOX_counts_DE_df)
#this should have 42 variables if we remove the 21 variables from 5FU

```

```{r Pull out Genes of Interest}
col_tx_large <- rep(c("#499FBD" , "#63666D"), 21)
col_tx_large_2 <- c(rep("#499FBD" , 3), rep("#63666D", 3), 21)


ind_col <- c("#003F5C", "#45AE91",  "#58508D", "#BC4099", "#8B3E9B", "#FF6361", "#FF2362")

tx_col <- c("#499FBD","#63666D")

time_col <- c("#fbb4b9", "#f768a1", "#ae017e")

##Add columns with more information to each gene I pull out##
ind_names <- c(rep("Ind1", 6), rep("Ind2", 6), rep("Ind3", 6), rep("Ind4", 6), rep("Ind5", 6), rep("Ind6", 6), rep("Ind6REP", 6))
time_names <- c(rep("24", 2), rep("24rec", 2), rep("144rec", 2))
time_names2 <- c("24", "24rec", "144rec")
time_names <- c(rep(time_names, 7))
time_names2 <- c(rep(time_names2, 7))
tx_names <- c("DOX", "DMSO")
tx_names <- c(rep(tx_names, 21))
tx_names2 <- c(rep("DOX", 3), rep("DMSO", 3))
tx_names2 <- c(rep(tx_names2, 21))
txtime_names <- c("DOX_24", "DMSO_24", "DOX_24rec", "DMSO_24rec", "DOX_144rec", "DMSO_144rec")
txtime_names <- c(rep(txtime_names, 7))
txtime_names2 <- c("DOX_24", "DOX_24rec", "DOX_144rec", "DMSO_24", "DMSO_24rec", "DMSO_144rec")
txtime_names2 <- c(rep(txtime_names2, 7))


genematrix_dox <- fC_Matrix_Full_cpm_filter_dox %>% rownames_to_column(var = "entrezgene_id")

#saveRDS(genematrix_dox, "data/genematrix_dox.RDS")

###CDKN1A - 1026###
CDKN1A_d <- genematrix_dox %>% filter(entrezgene_id=="1026")
CDKN1A_new_d <- as.data.frame(CDKN1A_d) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1","DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3",  "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")
CDKN1A_melt_d <- melt(CDKN1A_d, variable.name = "sample")
CDKN1A_melt_new_d <- melt(CDKN1A_new_d, variable.name = "sample")

CDKN1A_melt_df_d <- data.frame(tx = factor(tx_names, levels = unique(tx_names)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names, levels = unique(txtime_names)),
                               time = factor(time_names, levels = unique(time_names)))
CDKN1A_melt_df2_d <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
CDKN1A_melt_df_all_d <- cbind(CDKN1A_melt_d, CDKN1A_melt_df_d)
CDKN1A_melt_df_all2_d <- cbind(CDKN1A_melt_new_d, CDKN1A_melt_df2_d)


####CDKN1A####
CDKN1A_melt_df_all2_d %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "CDKN1A")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  theme(plot.title = element_text(face = "italic"))

#saveRDS(CDKN1A_geneplot_Dox, "data/CDKN1A_geneplot_Dox.RDS")

###MDM2 - 4193###
MDM2_d <- genematrix_dox %>% filter(entrezgene_id=="4193")
MDM2_new_d <- as.data.frame(MDM2_d) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1","DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3",  "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")
MDM2_melt_d <- melt(MDM2_d, variable.name = "sample")
MDM2_melt_new_d <- melt(MDM2_new_d, variable.name = "sample")

MDM2_melt_df_d <- data.frame(tx = factor(tx_names, levels = unique(tx_names)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names, levels = unique(txtime_names)),
                               time = factor(time_names, levels = unique(time_names)))
MDM2_melt_df2_d <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
MDM2_melt_df_all_d <- cbind(MDM2_melt_d, MDM2_melt_df_d)
MDM2_melt_df_all2_d <- cbind(MDM2_melt_new_d, MDM2_melt_df2_d)

####MDM2####
MDM2_melt_df_all2_d %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "MDM2")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  theme(plot.title = element_text(face = "italic"))

#saveRDS(MDM2_geneplot_Dox, "data/MDM2_geneplot_Dox.RDS")


###SIRT1 - 23411###
SIRT1_d <- genematrix_dox %>% filter(entrezgene_id=="23411")
SIRT1_new_d <- as.data.frame(SIRT1_d) %>% dplyr::select("entrezgene_id", "DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1","DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3",  "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")
SIRT1_melt_d <- melt(SIRT1_d, variable.name = "sample")
SIRT1_melt_new_d <- melt(SIRT1_new_d, variable.name = "sample")

SIRT1_melt_df_d <- data.frame(tx = factor(tx_names, levels = unique(tx_names)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names, levels = unique(txtime_names)),
                               time = factor(time_names, levels = unique(time_names)))
SIRT1_melt_df2_d <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                               ind = factor(ind_names, levels = unique(ind_names)),
                               txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                               time = factor(time_names2, levels = unique(time_names2)))
SIRT1_melt_df_all_d <- cbind(SIRT1_melt_d, SIRT1_melt_df_d)
SIRT1_melt_df_all2_d <- cbind(SIRT1_melt_new_d, SIRT1_melt_df2_d)

####SIRT1####
SIRT1_vertplot <- SIRT1_melt_df_all2_d %>% ggplot(aes(x = txtime, y = value))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "SIRT1")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  theme(plot.title = element_text(face = "italic"))

# get_legend_func <- function(plot) {
#   legends <- cowplot::get_plot_component(plot, "guide-box", return_all = TRUE)
#   nonzero <- vapply(plot_legend_d, \(x) !inherits(x, "zeroGrob"), TRUE)
#   idx <- which(nonzero)
# #this will return the first nonzero legend since it's not in the standard spot

# if (length(idx) >0) {
#   return(legends[[idx[1]]])  
# } else {
#   return(legends[[1]])
# }
# }

# plot_leg_d_hor <- get_legend_func(plot = SIRT1_vertplot + guides(color = guide_legend(nrow=1))+
#     theme(legend.position = "bottom")
# )


plot_leg_d_ver <- cowplot::get_legend(SIRT1_vertplot)

# ggdraw(plot_leg_d_hor)
ggdraw(plot_leg_d_ver)

#saveRDS(plot_leg_d_hor, "data/plot_leg_d_horizontal.RDS")
#saveRDS(plot_leg_d_ver, "data/plot_leg_d_vertical.RDS")

#saveRDS(SIRT1_geneplot_Dox, "data/SIRT1_geneplot_Dox.RDS")
#I saved these with no legend

#plot all of these genes together in one set

CDKN1A_geneplot_Dox <- readRDS("data/CDKN1A_geneplot_Dox.RDS")
MDM2_geneplot_Dox <- readRDS("data/MDM2_geneplot_Dox.RDS")
SIRT1_geneplot_Dox <- readRDS("data/SIRT1_geneplot_Dox.RDS")
plot_legend_d_ver <- readRDS("data/plot_leg_d_vertical.RDS")
plot_legend_d_hor <- readRDS("data/plot_leg_d_horizontal.RDS")

DOXgeneplots <- plot_grid(CDKN1A_geneplot_Dox,
                          MDM2_geneplot_Dox, 
                          SIRT1_geneplot_Dox, nrow = 1, ncol = 3)
DOXgeneplots

#saveRDS(DOXgeneplots, "data/DOXgeneplots.RDS")

```


```{r Pull Out Some Cardiac Genes}
genes_dox <- as.data.frame(fC_Matrix_Full_cpm_filter_dox)
genes_dox_df <- rownames_to_column(fC_Matrix_Full_cpm_filter_dox, var = "entrezgene_id")
###now let's pull some classic cardiac genes expressed in iPSC-CMs###
genecardiccheck_dox <- c("MYH7", "TNNT2","MYH6","ACTN2","BMP3","TNNI3","RYR2","CACNA1C","KCNQ1", "HCN1", "ADRB1", "ADRB2")

#ensembl_dox <- useMart("ensembl", dataset="hsapiens_gene_ensembl")
#saveRDS(ensembl_dox, "data/ensembl_backup_dox.RDS")
ensembl_dox <- readRDS("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/ensembl_backup_dox.RDS")
my_chr_dox <- c(1:22, 'M', 'X', 'Y')  ## creates a filter for each database

my_attributes_dox <- c('entrezgene_id', 'ensembl_gene_id', 'hgnc_symbol')

heartgenes_dox <- getBM(attributes=my_attributes_dox,filters ='hgnc_symbol',
                values = genecardiccheck_dox, mart = ensembl_dox)
write.csv(heartgenes_dox, "data/heartgenes_dox.csv")
heartgenes_dox <-read.csv("data/heartgenes_dox.csv")

fungraph_dox <- as.data.frame(fC_Matrix_Full_cpm_filter_dox[rownames(fC_Matrix_Full_cpm_filter_dox) %in% heartgenes_dox$entrezgene_id,])


fungraph_dox %>% 
  rownames_to_column("entrezgene_id") %>% 
  pivot_longer(-entrezgene_id, names_to = "samples",values_to = "counts") %>% 
  mutate(gene = case_match(entrezgene_id,"88"~"ACTN2","153"~"ADRB1",
  "154"~"ADRB2","651"~"BMP3","775"~"CACNA1C", "100874369"~"CACNA1C","348980"~"HCN1",
                           "3784"~"KCNQ1", "4624"~"MYH6","4625"~"MYH7","6262"~"RYR2",
                           "7137"~"TNNI3","7139"~"TNNT2",.default = entrezgene_id)) %>% 
  ggplot(., aes(x=reorder(gene,counts,decreasing=TRUE), y=counts))+
  geom_boxplot()+
  ggtitle(expression("Expression of typical cardiac tissue genes"))+
  xlab("")+
  ylim(c(0,20))+
  ylab(expression("log"[2]~"cpm"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))


#Now I want to plot these with DOX vs Veh to see if there are any changes

#I'll choose a subset of these genes - TNN13, TNNT2, NPPB, MYH7, ACTN2

fungraph_dox_only <- fungraph_dox %>% dplyr::select("DOX_24_Ind1", "DOX_24rec_Ind1", "DOX_144rec_Ind1", "DOX_24_Ind2", "DOX_24rec_Ind2", "DOX_144rec_Ind2", "DOX_24_Ind3", "DOX_24rec_Ind3", "DOX_144rec_Ind3", "DOX_24_Ind4", "DOX_24rec_Ind4", "DOX_144rec_Ind4", "DOX_24_Ind5", "DOX_24rec_Ind5", "DOX_144rec_Ind5", "DOX_24_Ind6", "DOX_24rec_Ind6", "DOX_144rec_Ind6", "DOX_24_Ind6REP", "DOX_24rec_Ind6REP", "DOX_144rec_Ind6REP")

fungraph_dmso_only <- fungraph_dox %>% dplyr::select("DMSO_24_Ind1", "DMSO_24rec_Ind1", "DMSO_144rec_Ind1", "DMSO_24_Ind2", "DMSO_24rec_Ind2", "DMSO_144rec_Ind2", "DMSO_24_Ind3", "DMSO_24rec_Ind3", "DMSO_144rec_Ind3", "DMSO_24_Ind4", "DMSO_24rec_Ind4", "DMSO_144rec_Ind4", "DMSO_24_Ind5", "DMSO_24rec_Ind5", "DMSO_144rec_Ind5", "DMSO_24_Ind6", "DMSO_24rec_Ind6", "DMSO_144rec_Ind6", "DMSO_24_Ind6REP", "DMSO_24rec_Ind6REP", "DMSO_144rec_Ind6REP")

fungraph_dox_only %>% 
  rownames_to_column("entrezgene_id") %>% 
  pivot_longer(-entrezgene_id, names_to = "samples",values_to = "counts") %>% 
  mutate(gene = case_match(entrezgene_id,"88"~"ACTN2","153"~"ADRB1",
  "154"~"ADRB2","651"~"BMP3","775"~"CACNA1C", "100874369"~"CACNA1C","348980"~"HCN1",
                           "3784"~"KCNQ1", "4624"~"MYH6","4625"~"MYH7","6262"~"RYR2",
                           "7137"~"TNNI3","7139"~"TNNT2",.default = entrezgene_id)) %>% 
  ggplot(., aes(x=reorder(gene,counts,decreasing=TRUE), y=counts))+
  geom_boxplot()+
  ggtitle(expression("Expression Cardiac Genes DOX Only"))+
  xlab("")+
  ylim(c(0,20))+
  ylab(expression("log"[2]~"cpm"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))

fungraph_dmso_only %>% 
  rownames_to_column("entrezgene_id") %>% 
  pivot_longer(-entrezgene_id, names_to = "samples",values_to = "counts") %>% 
  mutate(gene = case_match(entrezgene_id,"88"~"ACTN2","153"~"ADRB1",
  "154"~"ADRB2","651"~"BMP3","775"~"CACNA1C", "100874369"~"CACNA1C","348980"~"HCN1",
                           "3784"~"KCNQ1", "4624"~"MYH6","4625"~"MYH7","6262"~"RYR2",
                           "7137"~"TNNI3","7139"~"TNNT2",.default = entrezgene_id)) %>% 
  ggplot(., aes(x=reorder(gene,counts,decreasing=TRUE), y=counts))+
  geom_boxplot()+
  ggtitle(expression("Expression Cardiac Genes Vehicle Only"))+
  xlab("")+
  ylim(c(0,20))+
  ylab(expression("log"[2]~"cpm"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))

#I want to just plot TNNI3 now

fungraph_dox_only_TNNI3 <- fungraph_dox_only %>% 
  rownames_to_column("entrezgene_id") %>% 
  pivot_longer(-entrezgene_id, names_to = "samples",values_to = "counts") %>% 
  filter(entrezgene_id == "7137") %>% 
  mutate(gene = case_match(entrezgene_id,"7137"~"TNNI3",.default = entrezgene_id)) %>% 
  ggplot(., aes(x=reorder(gene,counts,decreasing=TRUE), y=counts))+
  geom_boxplot()+
  ggtitle(expression("TNNI3 DOX"))+
  xlab("")+
  ylim(c(0,10))+
  ylab(expression("log"[2]~"cpm"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))

#saveRDS(fungraph_dox_only_TNNI3, "data/DOX_TNN13_plot.RDS")

fungraph_dmso_only_TNNI3 <- fungraph_dmso_only %>% 
  rownames_to_column("entrezgene_id") %>% 
  pivot_longer(-entrezgene_id, names_to = "samples",values_to = "counts") %>% 
  filter(entrezgene_id == "7137") %>% 
  mutate(gene = case_match(entrezgene_id,"7137"~"TNNI3",.default = entrezgene_id)) %>% 
  ggplot(., aes(x=reorder(gene,counts,decreasing=TRUE), y=counts))+
  geom_boxplot()+
  ggtitle(expression("TNNI3 VEH"))+
  xlab("")+
  ylim(c(0,10))+
  ylab(expression("log"[2]~"cpm"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(2), hjust = 0.5),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))

#saveRDS(fungraph_dmso_only_TNNI3, "data/DMSO_TNN13_plot.RDS")

fungraph_dox_only_TNNI3
fungraph_dmso_only_TNNI3

plot_grid(fungraph_dox_only_TNNI3,
                          fungraph_dmso_only_TNNI3, 
                          nrow = 1, ncol = 2)


```

```{r Spearman Heatmap}

####SPEARMAN FILTERED####
fC_Matrix_Full_cpm_filter_spearmancor_dox <-
  cor(
    fC_Matrix_Full_cpm_filter_dox,
    y = NULL,
    use = "everything",
    method = "spearman"
  )


Individual <- as.factor(c(rep("Ind1", 6), rep("Ind2", 6), rep("Ind3", 6), rep("Ind4", 6), rep("Ind5", 6), rep("Ind6", 6), rep("Ind6REP", 6)))

#Factor 2 - Treatment
tx_factor <- c("DOX", "DMSO")
Tx <- as.factor(c(rep(tx_factor, 21)))
#view(Treatment)

#Factor 3 - Timepoint
time_factor <- c(rep("24", 2), rep("24rec", 2), rep("144rec", 2))
Time <- as.factor(c(rep(time_factor, 7)))

####annotation for colors####
annot_col_hm = list(Tx = c(DOX = "blue", DMSO = "black"),
                             Ind = c(Ind1 = "#66E2A5", Ind2 = "#FC8D62", Ind3 = "#1F78B4", Ind4 = "#EFEDA3", Ind5 = "#A6D854", Ind6 = "#FFD92A", Ind6REP = "#8B3E9B"),
                             Time = c("24" = "#096F38", "24rec" = "#0050B5", "144rec" = "#B725AD"))

####annotation for values####
annot_list_hm <- data.frame(Individual = as.factor(c(rep("Ind1", 6), rep("Ind2", 6), rep("Ind3", 6), rep("Ind4", 6), rep("Ind5", 6), rep("Ind6", 6), rep("Ind6REP", 6))),
                                               Tx = as.factor(c(rep(tx_factor, 21))), 
                                               Time = as.factor(c(rep(time_factor, 7))))

  
##add in the annotations from above into the dataframe
row.names(annot_list_hm) <- colnames(fC_Matrix_Full_cpm_filter_spearmancor_dox)

####ANNOTATED HEATMAPS####

pheatmap(fC_Matrix_Full_cpm_filter_spearmancor_dox, border_color = "black", legend = TRUE, angle_col = 90, display_numbers = FALSE, number_color = "black", fontsize = 10, fontsize_number = 5, annotation_col = annot_list_hm, annotation_colors = annot_col_hm)



```

Now that I've put together my data, let's begin DE!

```{r Differential Expression Analysis, collapse=TRUE}

group_1d <- rep(c("DOX_24",
                  "DMSO_24",
                  "DOX_24rec", 
                  "DMSO_24rec",
                  "DOX_144rec",
                  "DMSO_144rec"), 6)

group_2d <- rep(c("DOX_24",
                  "DMSO_24",
                  "DOX_24rec",
                  "DMSO_24rec",
                  "DOX_144rec",
                  "DMSO_144rec"), 7)

dge_d <- DGEList.data.frame(counts = DOX_counts_DE_df, group = group_2d, genes = row.names(DOX_counts_DE_df))


#calculate the normalization factors with method TMM
dged_calc <- calcNormFactors(dge_d, method = "TMM")


#Pull out factors
snames_d <- data.frame("samples" = colnames(dged_calc)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)

snames_time_d <- snames_d$Time
snames_tx_d <- snames_d$Treatment
snames_ind_d <- snames_d$Individual

#Create my model matrix
mm_r_d <- model.matrix(~0 + group_2d)

p_d <- voom(dged_calc$counts, mm_r_d, plot = TRUE)

corfit_d <- duplicateCorrelation(p_d, mm_r_d, block = snames_ind_d)

v_d <- voom(dged_calc$counts, mm_r_d, block = snames_ind_d, correlation = corfit_d$consensus)

fit_d <- lmFit(v_d, mm_r_d, block = snames_ind_d, correlation = corfit_d$consensus)

#make sure to check which order the columns are in - otherwise they won't match right (it was moved into alphabetical and number order)
colnames(mm_r_d) <- c("DMSO_144rec","DMSO_24","DMSO_24rec","DOX_144rec","DOX_24","DOX_24rec")

cm_r_d <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        levels = mm_r_d
)

vfit_r_d <- lmFit(p_d, mm_r_d)
vfit_r_d <- contrasts.fit(vfit_r_d, contrasts = cm_r_d)

efit2_d <- eBayes(vfit_r_d)

results_d = decideTests(efit2_d)
summary(results_d)
# #       V.D24 V.D24r V.D144r
# Down    9709   6869      31
# NotSig  3276   6379   13680
# Up      1185    922     459

####plot your voom####
voom_plot_d <- voom(dged_calc, mm_r_d, plot = TRUE)

```

```{r Make Toptables Of Each Condition}

top.table_V.D24_dox <- topTable(fit = efit2_d, coef = "V.D24", number = nrow(dged_calc), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.D24_dox)

#saveRDS(top.table_V.D24_dox, "data/top.table_V.D24_dox.RDS")

top.table_V.D24r_dox <- topTable(fit = efit2_d, coef = "V.D24r", number = nrow(dged_calc), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.D24r_dox)

#saveRDS(top.table_V.D24r_dox, "data/top.table_V.D24r_dox.RDS")

top.table_V.D144r_dox <- topTable(fit = efit2_d, coef = "V.D144r", number = nrow(dged_calc), adjust.method = "BH", p.value = 1, sort.by = "none")
head(top.table_V.D144r_dox)

#saveRDS(top.table_V.D144r_dox, "data/top.table_V.D144r_dox.RDS")


#plot the top 5 most DE genes for each condition
#i sorted the toptable by the top 5 most DE genes for each + found the corresponding gene name
#####24hr Tx####
topDEG_dox24 <- top.table_V.D24_dox %>% 
  dplyr::slice_min(., n=5, order_by=P.Value) %>% 
  rownames_to_column(var = "GeneID")

Gene_Name <- c("FAM163A","SLC26A8","THBS-AS1","FOXD3-AS1","GRIK3")

topDEG_dox24 %>% mutate(GeneID = factor(GeneID, levels = c("148753", "116369", "101929523", "100996301", "2899"))) %>% 
  cbind(Gene_Name)
  

#top 5 DE genes based on p-value and adj p value 
ggplot(topDEG_dox24, aes(x=Gene_Name, y=logFC))+
  geom_boxplot(aes(colour = adj.P.Val))+
  ggtitle(expression("Top 5 DE Genes DOX 24"))+
  xlab("")+
  ylim(c(-10, 10))+
  ylab(expression("logFC"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5)),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))

####24hr Recovery####
topDEG_dox24R <- top.table_V.D24r_dox %>% 
  dplyr::slice_min(., n=5, order_by=P.Value) %>% 
  rownames_to_column(var = "GeneID")

Gene_Name1 <- c("KCNK10", "CSTA", "LCE1B", "OTP", "KRTAP19-1")

topDEG_dox24R %>% mutate(GeneID = factor(GeneID, levels = c("54207", "1475", "353132", "23440", "337882"))) %>% 
  cbind(Gene_Name1)
  

ggplot(topDEG_dox24R, aes(x=Gene_Name1, y=logFC))+
  geom_boxplot(aes(colour = adj.P.Val))+
  ggtitle(expression("Top 5 DE Genes DOX 24 Recovery"))+
  xlab("")+
  ylim(c(-10,10))+
  ylab(expression("logFC"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5)),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))

####144R####
topDEG_dox144R <- top.table_V.D144r_dox %>% 
  dplyr::slice_min(., n=5, order_by=P.Value) %>% 
  rownames_to_column(var = "GeneID")

Gene_Name2 <- c("LINC03025", "SLC4A10", "VIT", "PSMA8", "MYO3B")

topDEG_dox144R %>% mutate(GeneID = factor(GeneID, levels = c("440896", "57282", "5212", "143471", "140469"))) %>% 
  cbind(Gene_Name2)
  

ggplot(topDEG_dox144R, aes(x=Gene_Name2, y=logFC))+
  geom_boxplot(aes(colour = adj.P.Val))+
  ggtitle(expression("Top 5 DE Genes DOX 144 Recovery"))+
  xlab("")+
  ylim(c(-10,10))+
  ylab(expression("logFC"))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5)),
        axis.title = element_text(size = 15, color = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size =10, color = "black", angle = 0),
        strip.text.y = element_text(color = "white"))

```



```{r Volcano Plots from Pairwise Gene Analysis, fig.width=6, fig.height=8}

generate_volcano_plot <- function(toptable, title) {
  
  #make significance labels
  toptable$Significance <- "Not Significant"
  toptable$Significance[toptable$logFC > 0 & toptable$adj.P.Val < 0.05] <- "Upregulated"
  toptable$Significance[toptable$logFC < 0 & toptable$adj.P.Val < 0.05] <- "Downregulated"
  
  #add number of genes for each significance label
  upgenes <- toptable %>% filter(Significance == "Upregulated") %>% nrow()
  downgenes <- toptable %>% filter(Significance == "Downregulated") %>% nrow()
  nsgenes <- toptable %>% filter(Significance == "Not Significant") %>% nrow()

  #make legend labels for no of genes
  legend_lab <- c(
    str_c('Upregulated: ', upgenes),
    str_c('Not Significant: ', nsgenes),
    str_c('Downregulated: ', downgenes)
  )

  #generate volcano plot w/ legend
  ggplot(toptable, aes(x = logFC, y = -log10(P.Value), color = Significance)) +
    geom_point(alpha = 0.4, size = 2) + 
    scale_color_manual(values = c("Upregulated" = "blue", "Downregulated" = "red", "Not Significant" = "gray"), labels = legend_lab) +
    xlim(-5, 5) +
    labs(title = title, x = expression(x = "log"[2]*"FC"), y = expression(y = "-log"[10]*"P-value")) +
    theme(legend.position = "none", 
          plot.title = element_text(size = rel(1.5), hjust = 0.5),
          axis.title = element_text(size = rel(1.25))) +
    theme_bw()
}

#now that I've made a function, I can make volcano plots for each of my comparisons (6 total)

volcano_plots <- list(
  "V.D24" = generate_volcano_plot(top.table_V.D24_dox, "Volcano Plot DOX 24hr (adj P-val<0.05)"),
  "V.D24r" = generate_volcano_plot(top.table_V.D24r_dox, "Volcano Plot DOX 24hr Recovery (adj P-val<0.05)"),
  "V.D144r" = generate_volcano_plot(top.table_V.D144r_dox, "Volcano Plot DOX 144hr Recovery (adj P-val<0.05)")
)

# Display each volcano plot
for (plot_name in names(volcano_plots)) {
  print(volcano_plots[[plot_name]])
}


#I want to add the number of DEG onto this for each
#make each volcano plot into an object so I can see how many rows it has

# Alternative function from Renee to combine them all into one image
# volcanosig <- function(df, psig.lvl,topg) {
#     df <- df %>% 
#     mutate(threshold = ifelse(adj.P.Val > psig.lvl, "A", ifelse(adj.P.Val <= psig.lvl & logFC<=0,"B","C")))
#     
#   ggplot(df, aes(x=logFC, y=-log10(adj.P.Val))) + 
#     geom_point(aes(color=threshold))+
#     xlab(expression("Log"[2]*" FC"))+
#     ylim(0,30)+
#     ylab(expression("-log"[10]*"P Value"))+
#     scale_color_manual(values = c("black", "red","blue"))+
#     theme_cowplot()+
#     theme(legend.position = "none",
#           plot.title = element_text(size = rel(0.8), hjust = 0.5),
#           axis.title = element_text(size = rel(0.8))) 
# }
# 
# vol1 <- volcanosig(top.table_V.D24_dox, 0.01,0)+ ggtitle("Doxorubicin \n 24 hour")
# vol2 <- volcanosig(top.table_V.D24r_dox, 0.01,0)+ ggtitle("Doxorubicin \n 24 hour Recovery")
# vol3 <- volcanosig(top.table_V.D144r_dox, 0.01,0)+ ggtitle("Doxorubicin\n 144 hour Recovery")+ylab("")
# 
# 
# Volcanoplots <- plot_grid(vol1,vol2,vol3, nrow = 1, ncol = 3)
# Volcanoplots


```

```{r Boxplots Across Conditions}
#now like the hypoxia paper, I'd like to put together comparisons between samples and conditions to check if they are more similar than others




```


```{r DOX only PCA plots no RUVs}
PCA_data_d <- fC_Matrix_Full_cpm_filter_dox %>% 
  prcomp(.) %>% 
  t()

PCA_data_test_d <- (prcomp(t(fC_Matrix_Full_cpm_filter_dox), scale. = TRUE))


#ind_num_dox <- c("1", "1", "1", "1", "1", "1", "2", "2", "2", "2", "2", "2", "3", "3", "3", "3", "3", "3", "4", "4", "4", "4", "4", "4","5", "5", "5", "5", "5", "5", "6", "6", "6", "6", "6", "6", "6R", "6R", "6R", "6R", "6R", "6R")

#saveRDS(ind_num_dox, "data/ind_num_dox.RDS")

ind_num_dox <- readRDS("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/ind_num_dox.RDS")

####now make an annotation for my PCA####
annot_d <- data.frame("samples" = colnames(fC_Matrix_Full_cpm_filter_dox)) %>% separate_wider_delim(., cols = samples, names = c("Tx", "Time", "Ind"), delim = "_", cols_remove = FALSE) %>% unite(., col = "Tx_Time", Tx, Time, sep = "_", remove = FALSE) %>% cbind(., ind_num_dox)

#saveRDS(annot_d, "C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/annot_dox.RDS")


#combine the prcomp matrix and annotation
annot_PCA_matrix_d <- PCA_data_test_d$x %>% cbind(., annot_d)


#now I can make a graph where I have filled values for individual! I have seven colors for seven individuals
# I have three fill values for three timepoints

#using annotation matrix above as well as annotated PCA matrix #(annot_PCA_matrix)

#extra info for colors in the graph (fill parameter)
fill_col_ind <- c("#66C2A5", "#FC8D62", "#1F78B4", "#E78AC3", "#A6D854", "#FFD92A", "#8B3E9B")

fill_col_ind_dark <- c("#003F5C", "#45AE91",  "#58508D", "#BC4099", "#8B3E9B", "#FF6361", "#FF2362")

fill_col_tx_dox <- c("#63666D", "#499FBD")

fill_col_txtime_d <- c("#45AE91",  "#58508D", "#BC4099", "#FF2362", "#A6D854", "#FC8D62")

#I want the color parameter to be treatment, the shape as time, and individual as number

####PC1/PC2####
annot_PCA_matrix_d %>% ggplot(., aes(x=PC1, y=PC2, size=10)) +
  geom_point(aes(color = Tx, shape = Time)) +
  scale_color_manual(values = c(fill_col_tx_dox))+
  scale_shape_manual(values = c(15, 19, 17))+
  geom_text_repel(aes(label = ind_num_dox))+
  theme_bw(base_size = 10)+
  ggtitle(expression("PCA of filtered log"[2]*"cpm"))+
  guides(size="none")


####PC1/PC2 autoplot####
prcomp_res_d <- prcomp(t(fC_Matrix_Full_cpm_filter_dox), scale. = FALSE, center = TRUE)
annot_prcomp_res_d <- prcomp_res_d$x %>% cbind(., annot_d)

ggplot2::autoplot(prcomp_res_d, data = annot_d, colour = "Tx_Time", size =4)+
  theme_bw()+
  scale_color_manual(values = c(fill_col_txtime_d))+
  ggrepel::geom_text_repel(label= ind_num_dox)+
  ggtitle("PCA of DOX filtered log2cpm")

####PC2/PC3####
annot_PCA_matrix_d %>% ggplot(., aes(x=PC2, y=PC3, size=10)) +
  geom_point(aes(color = Tx, shape = Time)) +
  scale_color_manual(values = c(fill_col_tx_dox))+
  scale_shape_manual(values = c(15, 19, 17))+
  geom_text_repel(aes(label = ind_num_dox))+
  theme_bw(base_size = 10)+
  ggtitle(expression("PCA of filtered log"[2]*"cpm"))+
  guides(size="none")

####PC3/PC4####
annot_PCA_matrix_d %>% ggplot(., aes(x=PC3, y=PC4, size=10)) +
  geom_point(aes(color = Tx, shape = Time)) +
  scale_color_manual(values = c(fill_col_tx_dox))+
  scale_shape_manual(values = c(15, 19, 17))+
  geom_text_repel(aes(label = ind_num_dox))+
  theme_bw(base_size = 10)+
  ggtitle(expression("PCA of filtered log"[2]*"cpm"))+
  guides(size="none")
```

```{r RUVs Correction Start DOX Only Set}

fC_Matrix_Full <- readRDS("data/fC_Matrix_Full.RDS")

fC_Matrix_Full_dox <- as.data.frame(fC_Matrix_Full) %>% dplyr::select(-(contains("FLUO")))

Counts_Full_df_dox <- data.frame(fC_Matrix_Full_dox)

#filter this dataframe by the filtered gene list I have
filt_gene_list_dox <- row.names(fC_Matrix_Full_cpm_filter_dox)
#saveRDS(filt_gene_list_dox, "data/filt_gene_list_dox.RDS")

counts_DE_df_dox <- Counts_Full_df_dox[! (rownames(Counts_Full_df_dox) %in% filt_gene_list_dox), ]

#saveRDS(counts_DE_df_dox, "data/counts_DE_df_dox.RDS")

RUV_filt_counts_dox <- fC_Matrix_Full_dox %>% 
  as.data.frame() %>% 
  dplyr::filter(., row.names(.)%in% filt_gene_list_dox)

#add in the annotation files
ind_num_dox <- readRDS("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/ind_num_dox.RDS")
annot_d <- readRDS("C:/Users/emmap/RDirectory/Recovery_RNAseq/Recovery_5FU/data/annot_dox.RDS")


#  counts need to be integer values and in a numeric matrix
# note: the log transformation needs to be accounted for in the isLog argument in RUVs function.
counts_d <- as.matrix(RUV_filt_counts_dox)
dim(counts_d)
#14225 genes
# Create a DataFrame for the phenoData
phenoData_d <- DataFrame(annot_d)

# Now create the RangedSummarizedExperiment necessary for RUVs input
# looks like it did need both the phenodata and the counts.
set_dox <- SummarizedExperiment(assays =  counts_d, metadata = phenoData_d)

# Generate a background matrix
# The column "Cond" holds the comparisons that you actually want to make. DOX_24, DMSO_24,5FU_24, DOX_3,etc.
scIdx_d <-RUVSeq::makeGroups(phenoData_d$Tx_Time)
scIdx_d

#now I've made all of the data I need for this - they are located in each section for k values

#DO NOT USE THESE COUNTS FOR LINEAR MODELING

#colors for all of the plots
fill_col_ind <- c("#66C2A5", "#FC8D62", "#1F78B4", "#E78AC3", "#A6D854", "#FFD92A", "#8B3E9B")

fill_col_ind_dark <- c("#003F5C", "#45AE91",  "#58508D", "#BC4099", "#8B3E9B", "#FF6361", "#FF2362")

fill_col_tx_dox <- c("#63666D", "#499FBD")

fill_col_txtime_dox <- c("#45AE91",  "#58508D", "#BC4099", "#FF2362", "#A6D854", "#FC8D62")

# before ruv
prcomp_res_d <- prcomp(t(counts_d), scale. = FALSE, center = TRUE)
ggplot2::autoplot(prcomp_res_d, data = annot_d, colour = "Tx_Time", size =4)+
  theme_bw()+
  scale_color_manual(values = c(fill_col_txtime_dox))+
  ggrepel::geom_text_repel(label= ind_num_dox)+
  ggtitle("No RUV")



####new PCA plots no correction####
#PCA plots for each value of k attached in each section
prcomp_res_d <- prcomp(t(counts_d), scale. = FALSE, center = TRUE)
annot_prcomp_res_d <- prcomp_res_d$x %>% cbind(., annot_d)

group_2d <- rep(c("DOX_24",
                  "DMSO_24",
                  "DOX_24rec",
                  "DMSO_24rec",
                  "DOX_144rec",
                  "DMSO_144rec"), 7)


dge1_d <- DGEList.data.frame(counts = fC_Matrix_Full_dox, group = group_2d, genes = row.names(fC_Matrix_Full_dox))

#calculate the normalization factors with method TMM
dge1_calc_d <- calcNormFactors(dge1_d, method = "TMM")

#Pull out factors
snames1_d <- data.frame("samples" = colnames(dge1_calc_d)) %>% separate_wider_delim(., cols = samples, names = c("Treatment", "Time", "Individual"), delim = "_", cols_remove = FALSE)

snames1_ind_d <- snames1_d$Individual

```

```{r RUVs k1 DOX only set}
#Apply RUVs function from RUVSeq
#"k" will be iteratively adjusted over time depending on your PCA.
set_d <- RUVSeq::RUVs(x = counts_d, k =1, scIdx = scIdx_d, isLog = FALSE)

#get the ruv weights to put into the linear model. n weights = k.
#k=1
RUV_df_d <- set_d$W %>% as.data.frame()
RUV_df_d$Names <- rownames(RUV_df_d)


#Check that the names match
#k=1
RUV_df_rm_d <- RUV_df_d[RUV_df_d$Names %in% annot_d$samples, ] 
RUV_1_d <-  RUV_df_rm_d$W_1

# after ruv k=1

#PCA checks
#k=1
prcomp_res_1_d <- prcomp(t(set_d$normalizedCounts), scale. = FALSE, center = TRUE)
annot_prcomp_res_1_d <- prcomp_res_1_d$x %>% cbind(., annot_d)

ggplot2::autoplot(prcomp_res_1_d, data = annot_d, colour = "Tx_Time", size =4)+
  theme_bw()+
  scale_color_manual(values = c(fill_col_txtime_dox))+
  ggrepel::geom_text_repel(label= ind_num_dox)+
  ggtitle("RUVs Correction k=1")


#k=1
annot_d$samples == RUV_df_rm_d$Names
annot_d$RUV_1_d <- RUV_df_rm_d$W_1

#Create my model matrix
#k=1
mm_r1_d <- model.matrix(~0 + group_2d + RUV_1_d, data = annot_d)

p1_d <- voom(dge1_calc_d$counts, mm_r1_d, plot = TRUE)

corfit1_d <- duplicateCorrelation(p1_d, mm_r1_d, block = snames1_ind_d)

v1_d <- voom(dge1_calc_d$counts, mm_r1_d, block = snames1_ind_d, correlation = corfit1_d$consensus)

fit1_d <- lmFit(v1_d, mm_r1_d, block = snames1_ind_d, correlation = corfit1_d$consensus)

#k=1
mm_r1_names_d <- str_replace(string = colnames(mm_r1_d), pattern = "group_2d", replacement = "")
design_d <- model.matrix(~ group_2d + RUV_1_d , annot_d)
colnames(mm_r1_d) <- mm_r1_names_d

#k=1
cm_r1_d <- makeContrasts(
        V.D24 = DOX_24 - DMSO_24,
        V.D24r = DOX_24rec - DMSO_24rec,
        V.D144r = DOX_144rec - DMSO_144rec,
        RUV_1_24 = RUV_1_d - DMSO_24,
        RUV_1_24r= RUV_1_d - DMSO_24rec,
        RUV_1_144r = RUV_1_d - DMSO_144rec,
        levels = mm_r1_d
)

#k=1
vfit_r1_d <- lmFit(p1_d, mm_r1_d)
vfit_r1_d <- contrasts.fit(vfit_r1_d, contrasts = cm_r1_d)

#k=1
efit1_d <- eBayes(vfit_r1_d)

#k=1
results1_d = decideTests(efit1_d)
summary(results1_d)
#         V.D24 V.D24r V.D144r RUV_1_24 RUV_1_24r RUV_1_144r
# Down    5222   2650     486    13532     13539      13491
# NotSig 14886  16841   26921     1645      1641       1721
# Up      8287   8904     988    13218     13215      13183

#k=1
toptable_Dupcor_DOX_d <- topTable(efit1_d, coef = "V.D24", number = nrow(dge1_calc_d$counts), p.value = 1)
toptable_Dupcor_DOXrec_d <- topTable(efit1_d, coef = "V.D24r", number = nrow(dge1_calc_d$counts), p.value = 1)
toptable_Dupcor_DOX144_d <- topTable(efit1_d, coef = "V.D144r", number = nrow(dge1_calc_d$counts), p.value = 1)
#k=1 plots
toptable_Dupcor_DOX_d$logFC %>% hist(, main= "RUVs k=1 DOX 24hr Toptable")
toptable_Dupcor_DOXrec_d$logFC %>% hist(, main = "RUVs k=1 DOX 24Rec Toptable")
toptable_Dupcor_DOX144_d$logFC %>% hist(, main = "RUVs k=1 DOX 144Rec Toptable")

```

```{r Cormotif Data}
#function is on the previous
annot_data_d <- annot_d %>% dplyr::select("Tx_Time", "Time", "Tx", "Ind", "samples")

Cormotif_d <- counts_DE_df_dox %>% cpm(., log = TRUE)
Cormotif_df_d <- as.data.frame(Cormotif_d)

groupid_d <- rep(c(1, 2, 3, 4, 5, 6), 7)

compid_d <- data.frame(c1 = c(1, 3, 5), c2 = c(2, 4, 6))


```

```{r Cormotif Run}

#set.seed(12345)
# initial_cormotif_dox <- cormotiffit(exprs = Cormotif_d,
#                                  groupid = groupid_d,
#                                  compid = compid_d,
#                                  K=1:8  , max.iter = 500, runtype = "logCPM")

#only need to run this once! 
#save this to an object so I can retrieve it as needed

#saveRDS(initial_cormotif_dox, "data/initial_cormotif_dox.RDS")

initial_cormotif_dox <- readRDS("data/initial_cormotif_dox.RDS")


```

```{r Plot AIC and BIC}

plotIC(initial_cormotif_dox)

```

```{r Plot Motifs DOX Only}

plotMotif(initial_cormotif_dox)

myColors <-  rev(c("#FFFFFF", "#E6E6E6" ,"#CCCCCC", "#B3B3B3", "#999999", "#808080", "#666666","#4C4C4C", "#333333", "#191919","#000000"))

plot.new()
legend('bottomleft',fill=myColors, legend =rev(c("0", "0.1", "0.2", "0.3", "0.4",  "0.5", "0.6", "0.7", "0.8","0.9", "1")), box.col="white",title = "Probability\nlegend", horiz=FALSE,title.cex=.8)

```

```{r Cormotif Cluster Probabilities}

topgenelist_d <- generank(initial_cormotif_dox$bestmotif$p.post)
rownames(topgenelist_d) <- rownames(Cormotif_df_d)


motif_prob_d <- initial_cormotif_dox$bestmotif$clustlike
rownames(motif_prob_d) <- rownames(topgenelist_d)
#saveRDS(motif_prob_d, "data/Cormotif_prob_gene_list_doxonly.RDS")


#Define the gene probability groups - I have 4
clust1_d <- motif_prob_d %>% 
  as.data.frame() %>% 
  filter(V1>0.57) %>% 
  rownames

#using a filter of >0.57 I get 11130 genes compared to 11156 above

length(clust1_d)
#11482 > 0.5
#11130 > 0.57
#is this clust1 a non-response?

clust1_d_df <- as.data.frame(clust1_d)

#example gene - UID 4681 - NBL1
NBL1_motif1 <- Cormotif_df_d %>% 
  rownames_to_column(var = "entrezgene_id") %>% 
  dplyr::filter(entrezgene_id == "4681")

NBL1_motif1_long <- melt(NBL1_motif1,
                         id.vars = c("entrezgene_id"),
                         variable.name = "Sample",
                         value.name = "log2cpm")

#now add in my factors like time, tx, tx_time, and ind by breaking up the Sample column
NBL1_motif1_long_df <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                                  ind = factor(ind_names, levels = unique(ind_names)),
                                  txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                                  time = factor(time_names2, levels = unique(time_names2)))

NBL1_motif1_long_factors <- cbind(NBL1_motif1_long_df, NBL1_motif1_long)

NBL1_motif1_long_factors %>% ggplot(aes(x = txtime, y = log2cpm))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "NBL1")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(0,10)+
  theme(plot.title = element_text(face = "italic"))

#saveRDS(CDKN1A_geneplot_Dox, "data/CDKN1A_geneplot_Dox.RDS")

clust2_d <- motif_prob_d %>% 
  as.data.frame() %>% 
  filter(V2>0.132) %>% 
  rownames

#using a filter of >0.132 I get the closest - 326 original vs 327 genes here

length(clust2_d)
#0 > 0.5
#326 > 0.132
#is clust2 a dox early response?

clust2_d_df <- as.data.frame(clust2_d)

#example gene - 597 - BCL2A1
BCL2A1_motif1 <- Cormotif_df_d %>% 
  rownames_to_column(var = "entrezgene_id") %>% 
  dplyr::filter(entrezgene_id == "597")

BCL2A1_motif1_long <- melt(BCL2A1_motif1,
                         id.vars = c("entrezgene_id"),
                         variable.name = "Sample",
                         value.name = "log2cpm")

#now add in my factors like time, tx, tx_time, and ind by breaking up the Sample column
BCL2A1_motif1_long_df <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                                  ind = factor(ind_names, levels = unique(ind_names)),
                                  txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                                  time = factor(time_names2, levels = unique(time_names2)))

BCL2A1_motif1_long_factors <- cbind(BCL2A1_motif1_long_df, BCL2A1_motif1_long)

BCL2A1_motif1_long_factors %>% ggplot(aes(x = txtime, y = log2cpm))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "BCL2A1")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(0,10)+
  theme(plot.title = element_text(face = "italic"))


clust3_d <- motif_prob_d %>% 
  as.data.frame() %>% 
  filter(V3>0.3) %>% 
  rownames

#using a filter of >0.3 I get 740 genes vs original 739 genes

length(clust3_d)
#538 > 0.5
#740 > 0.3
#is clust 3 a late response?

clust3_d_df <- as.data.frame(clust3_d)

#example gene - 8856 - NR1I2
NR1I2_motif1 <- Cormotif_df_d %>% 
  rownames_to_column(var = "entrezgene_id") %>% 
  dplyr::filter(entrezgene_id == "8856")

NR1I2_motif1_long <- melt(NR1I2_motif1,
                         id.vars = c("entrezgene_id"),
                         variable.name = "Sample",
                         value.name = "log2cpm")

#now add in my factors like time, tx, tx_time, and ind by breaking up the Sample column
NR1I2_motif1_long_df <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                                  ind = factor(ind_names, levels = unique(ind_names)),
                                  txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                                  time = factor(time_names2, levels = unique(time_names2)))

NR1I2_motif1_long_factors <- cbind(NR1I2_motif1_long_df, NR1I2_motif1_long)

NR1I2_motif1_long_factors %>% ggplot(aes(x = txtime, y = log2cpm))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "NR1I2")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(0,10)+
  theme(plot.title = element_text(face = "italic"))

#cluster 4
clust4_d <- motif_prob_d %>% 
  as.data.frame() %>% 
  filter(V4>0.4) %>% 
  rownames

#using >0.4 as my cutoff, I get 1938 genes as compared to original 1948 genes

length(clust4_d)
#1576 > 0.5
#1938 > 0.4
#is clust4 a DOX late response?

clust4_d_df <- as.data.frame(clust4_d)

#example gene - 6725 - SRMS 
SRMS_motif1 <- Cormotif_df_d %>% 
  rownames_to_column(var = "entrezgene_id") %>% 
  dplyr::filter(entrezgene_id == "6725")

SRMS_motif1_long <- melt(SRMS_motif1,
                         id.vars = c("entrezgene_id"),
                         variable.name = "Sample",
                         value.name = "log2cpm")

#now add in my factors like time, tx, tx_time, and ind by breaking up the Sample column
SRMS_motif1_long_df <- data.frame(tx = factor(tx_names2, levels = unique(tx_names2)),
                                  ind = factor(ind_names, levels = unique(ind_names)),
                                  txtime = factor(txtime_names2, levels = unique(txtime_names2)),
                                  time = factor(time_names2, levels = unique(time_names2)))

SRMS_motif1_long_factors <- cbind(SRMS_motif1_long_df, SRMS_motif1_long)

SRMS_motif1_long_factors %>% ggplot(aes(x = txtime, y = log2cpm))+
  geom_boxplot(aes(fill = tx))+
  geom_point(aes(color = ind)) +
  labs(title = "SRMS")+
  theme_bw(base_size = 16)+
  scale_fill_manual(values = c(tx_col))+
  scale_color_manual(values = c(ind_col))+
  xlab("Conditions")+
  ylab("log2cpm")+
  ylim(0,10)+
  theme(plot.title = element_text(face = "italic"))

#now let's make a pie chart with the above number of genes for each motif
clusterdata_d_adj <- data.frame(
  Category = c("No Response","Acute Response", "Late Response", "Early Sustained Response"), 
  Value = c(11130, 327, 740, 1938)
)

piecolors <- c("No Response" = "#007896", 
               "Acute Response" = "#079255",
               "Late Response" = "#58508D", 
               "Early Sustained Response" = "#BC5090")

#make a piechart of these distributions
clusterdata_d_adj %>% ggplot(aes(x = "", y = Value, fill = Category))+
  geom_bar(width = 1, stat = "identity")+
  coord_polar("y", start = 0)+
  geom_text(aes(label = Value),
            position = position_stack(vjust = 0.5),
            size = 4, color = "black")+
  labs(title = "Distribution of Gene Clusters Identified By Cormotif", x = NULL, y = NULL)+
  theme_void()+
  scale_fill_manual(values = piecolors)



```

```{r By Study Cormotif Posterior Probability}
#now let's look at each of these motifs by study with the posterior probability
#changing the cutoffs based on the look of each motif - if it's very dark = > 0.5
#if white < 0.5 
#if grey try a less stringent cutoff > 0.1
#ideally each of these are similar to the number of genes in the above plot

#gene_postprob_motif <- initial_cormotif_dox$bestmotif$p.post
#rownames(gene_postprob_motif) <- rownames(Cormotif_df_d)

#saveRDS(gene_postprob_motif, "data/gene_postprob_motif.RDS")


gene_postprob_motif <- readRDS("data/gene_postprob_motif.RDS")

gene_postprob_motif_df <- as.data.frame(gene_postprob_motif)


#motif 1 no response p.prob
prob_motif_1 <- rownames(gene_postprob_motif_df[(gene_postprob_motif_df[,1] < 0.62 
                                              & gene_postprob_motif_df[,2] < 0.62 
                                              & gene_postprob_motif_df[,3] < 0.62),])
length(prob_motif_1)
#10903 genes post prob

#gene example:  943 - TNFRSF8
TNFRF8_gene <- 

#10903 genes post prob - close to value of 11156 on cormotif pattern graph

#gene example:  11009 - IL24
IL24_gene_motif1 <- 


#motif 2 acute response
prob_motif_2 <- rownames(gene_postprob_motif_df[(gene_postprob_motif_df[,1] > 0.99 
                                              & gene_postprob_motif_df[,2] < 0.5 
                                              & gene_postprob_motif_df[,3] < 0.5),])
length(prob_motif_2)
#2586 genes post prob

#motif 3 late response
prob_motif_3 <- rownames(gene_postprob_motif_df[(gene_postprob_motif_df[,1] > 0.1 
                                              & gene_postprob_motif_df[,2] > 0.5 
                                              & gene_postprob_motif_df[,3] > 0.5),])
length(prob_motif_3)
#451 genes post prob

#motif 4 early sustained response
prob_motif_4 <- rownames(gene_postprob_motif_df[(gene_postprob_motif_df[,1] > 0.5 
                                              & gene_postprob_motif_df[,2] > 0.1 
                                              & gene_postprob_motif_df[,3] < 0.5),])
length(prob_motif_4)
#2600 genes post prob

#this totals 16540

```


```{r Assess Gene Distribution Post Prob Cormotif}

clusterdata_postprob <- data.frame(
  Category = c("No Response", "Acute Response", "Late Response", "Early Sustained Response"), 
  Value = c(10903, 2586, 451, 2600)
)

piecolors_2 <- c("No Response" = "#007896",
               "Acute Response" = "#58508D",
               "Late Response" = "#ABCC59", 
               "Early Sustained Response" = "#BC5090")

#make a piechart of these distributions
clusterdata_postprob %>% ggplot(aes(x = "", y = Value, fill = Category))+
  geom_bar(width = 1, stat = "identity")+
  coord_polar("y", start = 0)+
  geom_text(aes(label = Value),
            position = position_stack(vjust = 0.5),
            size = 4, color = "black")+
  labs(title = "Distribution of Gene Clusters Identified By Cormotif", x = NULL, y = NULL)+
  theme_void()+
  scale_fill_manual(values = piecolors_2)

```

Now we will use Gene Ontology analysis to discover biological relevance to the genes that are assigned to this cluster
This set is for the No Response motif (motif 1) with clustlike
```{r GO analysis NR Dox only}
motif_NR_d <- clust1_d

NRmotif_genes_d <- gost(query = motif_NR_d,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

cormotifNFclust_d <- gostplot(NRmotif_genes_d, capped = FALSE, interactive = TRUE)
cormotifNFclust_d

tableNR_d <- NRmotif_genes_d$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

tableNR_d %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(tableNR, "output/table_NRmotif.csv")

#GO:BP
tableNR_GOBP_d <- tableNR_d %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(tableNR_GOBP, "data/tableNR_GOBP.RDS")

tableNR_GOBP_d %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
tableNR_KEGG_d <- tableNR_d %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

tableNR_KEGG_d %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

```

Now let's do the same thing but with the posterior probability rather than the clustlike
```{r No Response Motif Genes Post Prob}
motif_NR_pp <- prob_motif_1

NRmotif_genes_pp <- gost(query = motif_NR_pp,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

cormotifNFclust_pp <- gostplot(NRmotif_genes_pp, capped = FALSE, interactive = TRUE)
cormotifNFclust_pp

tableNR_pp <- NRmotif_genes_pp$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

tableNR_pp %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(tableNR, "output/table_NRmotif.csv")

#GO:BP
tableNR_GOBP_pp <- tableNR_pp %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(tableNR_GOBP_pp, "data/tableNR_GOBP_postprob.RDS")

tableNR_GOBP_pp %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
tableNR_KEGG_pp <- tableNR_pp %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

tableNR_KEGG_pp %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

```

The Acute Response had no genes pop up for clustlike at the cutoff of > 0.5
However, when we changed the cutoff and used post prob we got some genes, so let's use them for GO analysis
This is from motif 2, where there is a light gray box in the first study (DOX 24hr)
```{r Acute Response GO Analysis KEGG terms}
motif_AR_pp <- prob_motif_2

ARmotif_genes_pp <- gost(query = motif_AR_pp,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

cormotifARclust_pp <- gostplot(ARmotif_genes_pp, capped = FALSE, interactive = TRUE)
cormotifARclust_pp

tableAR_pp <- ARmotif_genes_pp$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

tableAR_pp %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(tableNR, "output/table_NRmotif.csv")

#GO:BP
tableNR_GOBP_pp <- tableNR_pp %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(tableNR_GOBP_pp, "data/tableNR_GOBP_postprob.RDS")

tableNR_GOBP_pp %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
tableNR_KEGG_pp <- tableNR_pp %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

tableNR_KEGG_pp %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))




```



Next, we'll look at the Late Response (motif 3)
```{r Late DOX Response Motif Genes}

motif_LD_d <- clust3_d


LDmotif_genes_d <- gost(query = motif_LD_d,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

cormotifLDclust_d <- gostplot(LDmotif_genes_d, capped = FALSE, interactive = TRUE)
cormotifLDclust_d

tableLD_d <- LDmotif_genes_d$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

tableLD_d %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(tableLD, "output/table_LateDOXmotif.csv")

#GO:BP
tableLD_GOBP_d <- tableLD_d %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(tableLD_GOBP, "data/tableLD_GOBP.RDS")

tableLD_GOBP_d %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("Late Response Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
tableLD_KEGG_d <- tableLD_d %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

tableLD_KEGG_d %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("Late Response Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))


```

Now let's look at this motif 3 (Late Response) with the posterior probability
```{r Late Response Motif 3 Posterior Probability}
motif_LR_pp <- prob_motif_3

LRmotif_genes_pp <- gost(query = motif_LR_pp,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

cormotifLRclust_pp <- gostplot(LRmotif_genes_pp, capped = FALSE, interactive = TRUE)
cormotifLRclust_pp

tableLR_pp <- LRmotif_genes_pp$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

tableLR_pp %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(tableLR_pp, "output/table_LRmotif.csv")

#GO:BP
tableLR_GOBP_pp <- tableLR_pp %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(tableLR_GOBP_pp, "data/tableLR_GOBP_postprob.RDS")

tableLR_GOBP_pp %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
tableLR_KEGG_pp <- tableLR_pp %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

tableLR_KEGG_pp %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))




```
Now let's look at motif 4 - Early Sustained Response

```{r Early Sustained DOX Response Motif Genes}

motif_ED_d <- clust4_d

EDmotif_genes_d <- gost(query = motif_ED_d,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

cormotifEDclust_d <- gostplot(EDmotif_genes_d, capped = FALSE, interactive = TRUE)
cormotifEDclust_d

tableED_d <- EDmotif_genes_d$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

tableED_d %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

#write.csv(tableED, "output/table_EarlyDOXmotif.csv")


#GO:BP
tableED_GOBP_d <- tableED_d %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(tableED_GOBP, "data/tableED_GOBP.RDS")

tableED_GOBP_d %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("Early Sustained Response Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
tableED_KEGG_d <- tableED_d %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

tableED_KEGG_d %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("Early Sustained Response Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

```

Now let's see this with the posterior probability for the Early Sustained Response motif 4
```{r Early Sustained Response Posterior Probability}
motif_ESR_pp <- prob_motif_4

ESRmotif_genes_pp <- gost(query = motif_ESR_pp,
                      organism = "hsapiens",
                      ordered_query = FALSE,
                      measure_underrepresentation = FALSE,
                      evcodes = FALSE,
                      user_threshold = 0.05,
                      correction_method = c("fdr"),
                      sources = c("GO:BP", "KEGG"))

cormotifESRclust_pp <- gostplot(ESRmotif_genes_pp, capped = FALSE, interactive = TRUE)
cormotifESRclust_pp

tableESR_pp <- LRmotif_genes_pp$result %>% 
  dplyr::select(c(source, term_id, term_name, intersection_size, term_size, p_value))

tableESR_pp %>% 
  mutate_at(.vars = 6, .funs = scales::label_scientific(digits=4)) %>% 
  kableExtra::kable(.,) %>% 
  kableExtra::kable_paper("striped", full_width = FALSE) %>% 
  kableExtra::kable_styling(full_width = FALSE, position = "left", bootstrap_options = c("striped", "hover")) %>% 
  kableExtra::scroll_box(width = "100%", height = "400px")

write.csv(tableESR_pp, "output/table_ESRmotif.csv")

#GO:BP
tableESR_GOBP_pp <- tableESR_pp %>% 
  dplyr::filter(source=="GO:BP") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

#saveRDS(tableESR_GOBP_pp, "data/tableESR_GOBP_postprob.RDS")

tableESR_GOBP_pp %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched GO:BP Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("GO:BP term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))

#KEGG
tableESR_KEGG_pp <- tableESR_pp %>% 
  dplyr::filter(source=="KEGG") %>% 
  dplyr::select(p_value, term_name, intersection_size) %>% 
  dplyr::slice_min(., n=10, order_by=p_value) %>% 
  mutate(log_val = -log10(p_value))

tableESR_KEGG_pp %>% ggplot(., aes(x = log_val, y = reorder(term_name, p_value), col= intersection_size)) +
  geom_point(aes(size = intersection_size)) +
  ggtitle("No Response Enriched KEGG Terms")+
  xlab(expression("-log"[10]~"p-value"))+
  guides(col="none", size= guide_legend(title = "# of intersected \n terms"))+
  ylab("KEGG term")+
  scale_y_discrete(labels = scales::label_wrap(30))+
  theme_bw()+
  theme(plot.title = element_text(size = rel(1.5), hjust = 0.5),
        axis.title = element_text(size = 15, colour = "black"),
        axis.ticks = element_line(linewidth = 1.5),
        axis.line = element_line(linewidth = 1.5),
        axis.text = element_text(size = 10, colour = "black", angle = 0),
        strip.text = element_text(size = 15, colour = "black", face = "bold"))






```

Currently for the posterior probability code - motif 3 and motif 4 are showing up identically

